<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.553">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>The APOE isoforms differentially shape the transcriptomic and epigenomic landscapes of human microglia in a xenotransplantation model of Alzheimer’s disease</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="manuscript_figures_files/libs/clipboard/clipboard.min.js"></script>
<script src="manuscript_figures_files/libs/quarto-html/quarto.js"></script>
<script src="manuscript_figures_files/libs/quarto-html/popper.min.js"></script>
<script src="manuscript_figures_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="manuscript_figures_files/libs/quarto-html/anchor.min.js"></script>
<link href="manuscript_figures_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="manuscript_figures_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="manuscript_figures_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="manuscript_figures_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="manuscript_figures_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">The APOE isoforms differentially shape the transcriptomic and epigenomic landscapes of human microglia in a xenotransplantation model of Alzheimer’s disease</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>This markdown contains all the code to generate the main and supplementary figures in:</p>
<p>Murphy, KB., Hu D., Wolfes, L., Mancuso, R., De Strooper, B., Marzi, SJ. The APOE isoforms differentially shape the transcriptomic and epigenomic landscapes of human microglia in a xenotransplantation model of Alzheimer’s disease.</p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Differential expression analysis using DeSeq2 </span><span class="al">###</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Define samples and groups </span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span>(<span class="st">"E2-"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">sep=</span><span class="st">""</span>), </span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span>(<span class="st">"E3-"</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">sep=</span><span class="st">""</span>), </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span>(<span class="st">"E4-"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">sep=</span><span class="st">""</span>), </span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">paste</span>(<span class="st">"KO-"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">sep=</span><span class="st">""</span>)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">c</span>(</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"E2"</span>, <span class="at">times =</span> <span class="dv">5</span>), </span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"E3"</span>, <span class="at">times =</span> <span class="dv">4</span>), </span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"E4"</span>, <span class="at">times =</span> <span class="dv">5</span>), </span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rep</span>(<span class="st">"KO"</span>, <span class="at">times =</span> <span class="dv">5</span>)</span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">samples =</span> samples, <span class="at">group =</span> group)</span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="do">## Hashed out code is to read in the Salmon quant.sf files and create a DeSeqDataset object ##</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="co"># make tx2gene from gtf file </span></span>
<span id="cb1-20"><a href="#cb1-20" aria-hidden="true" tabindex="-1"></a><span class="co"># txdb &lt;- TxDb.Hsapiens.UCSC.hg38.knownGene</span></span>
<span id="cb1-21"><a href="#cb1-21" aria-hidden="true" tabindex="-1"></a><span class="co"># key &lt;- keys(txdb, keytype = "GENEID")</span></span>
<span id="cb1-22"><a href="#cb1-22" aria-hidden="true" tabindex="-1"></a><span class="co"># tx2gene &lt;- select(txdb, keys = key, keytype = "GENEID", columns = "TXNAME")</span></span>
<span id="cb1-23"><a href="#cb1-23" aria-hidden="true" tabindex="-1"></a><span class="co"># tx2gene &lt;- tx2gene[, c(2, 1)]</span></span>
<span id="cb1-24"><a href="#cb1-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-25"><a href="#cb1-25" aria-hidden="true" tabindex="-1"></a><span class="co"># import salmon quant files </span></span>
<span id="cb1-26"><a href="#cb1-26" aria-hidden="true" tabindex="-1"></a><span class="co"># files &lt;- file.path(rna_dir, "salmon", samples, "quant.sf")</span></span>
<span id="cb1-27"><a href="#cb1-27" aria-hidden="true" tabindex="-1"></a><span class="co"># txi &lt;- tximport(files, type = "salmon", tx2gene = tx2gene)</span></span>
<span id="cb1-28"><a href="#cb1-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-29"><a href="#cb1-29" aria-hidden="true" tabindex="-1"></a><span class="co"># make DeSeqDataset object </span></span>
<span id="cb1-30"><a href="#cb1-30" aria-hidden="true" tabindex="-1"></a><span class="co"># ddsTxi &lt;- DESeqDataSetFromTximport(txi,</span></span>
<span id="cb1-31"><a href="#cb1-31" aria-hidden="true" tabindex="-1"></a><span class="co">#                                    colData = colData,</span></span>
<span id="cb1-32"><a href="#cb1-32" aria-hidden="true" tabindex="-1"></a><span class="co">#                                    design = ~ group) </span></span>
<span id="cb1-33"><a href="#cb1-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-34"><a href="#cb1-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in DESeqDataset object </span></span>
<span id="cb1-35"><a href="#cb1-35" aria-hidden="true" tabindex="-1"></a>ddsTxi <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">"data/APOE_microglia_RNAseq_DESeqDataset_obj.rds"</span>)</span>
<span id="cb1-36"><a href="#cb1-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-37"><a href="#cb1-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove samples with high APOE expression in the KO</span></span>
<span id="cb1-38"><a href="#cb1-38" aria-hidden="true" tabindex="-1"></a>indices_to_remove <span class="ot">&lt;-</span> <span class="fu">which</span>(ddsTxi<span class="sc">$</span>samples <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"KO-1"</span>, <span class="st">"KO-2"</span>))</span>
<span id="cb1-39"><a href="#cb1-39" aria-hidden="true" tabindex="-1"></a>ddsTxi_subset <span class="ot">&lt;-</span> ddsTxi[, <span class="sc">-</span>indices_to_remove]</span>
<span id="cb1-40"><a href="#cb1-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-41"><a href="#cb1-41" aria-hidden="true" tabindex="-1"></a><span class="co"># Run differential expression analysis </span></span>
<span id="cb1-42"><a href="#cb1-42" aria-hidden="true" tabindex="-1"></a>dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(ddsTxi_subset)  </span>
<span id="cb1-43"><a href="#cb1-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-44"><a href="#cb1-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Run contrasts</span></span>
<span id="cb1-45"><a href="#cb1-45" aria-hidden="true" tabindex="-1"></a>contrasts <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb1-46"><a href="#cb1-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"E2"</span>, <span class="st">"E3"</span>),</span>
<span id="cb1-47"><a href="#cb1-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"E4"</span>, <span class="st">"E3"</span>),</span>
<span id="cb1-48"><a href="#cb1-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"E4"</span>, <span class="st">"E2"</span>),</span>
<span id="cb1-49"><a href="#cb1-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"KO"</span>, <span class="st">"E3"</span>)</span>
<span id="cb1-50"><a href="#cb1-50" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add more contrasts as needed</span></span>
<span id="cb1-51"><a href="#cb1-51" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb1-52"><a href="#cb1-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-53"><a href="#cb1-53" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">lapply</span>(contrasts, <span class="cf">function</span>(con) <span class="fu">results</span>(dds, <span class="at">contrast =</span> con))</span>
<span id="cb1-54"><a href="#cb1-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-55"><a href="#cb1-55" aria-hidden="true" tabindex="-1"></a><span class="co"># Get gene symbols corresponding to entrez ids</span></span>
<span id="cb1-56"><a href="#cb1-56" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="at">biomart =</span> <span class="st">"genes"</span>, <span class="at">dataset =</span> <span class="st">"hsapiens_gene_ensembl"</span>)</span>
<span id="cb1-57"><a href="#cb1-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-58"><a href="#cb1-58" aria-hidden="true" tabindex="-1"></a>convertEntrezToGeneSymbol <span class="ot">&lt;-</span> <span class="cf">function</span>(res) {</span>
<span id="cb1-59"><a href="#cb1-59" aria-hidden="true" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(res)</span>
<span id="cb1-60"><a href="#cb1-60" aria-hidden="true" tabindex="-1"></a>  entrez_ids <span class="ot">&lt;-</span> <span class="fu">rownames</span>(res_df)</span>
<span id="cb1-61"><a href="#cb1-61" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb1-62"><a href="#cb1-62" aria-hidden="true" tabindex="-1"></a>  gene_symbols <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb1-63"><a href="#cb1-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">'hgnc_symbol'</span>, <span class="st">'entrezgene_id'</span>),</span>
<span id="cb1-64"><a href="#cb1-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> entrez_ids,</span>
<span id="cb1-65"><a href="#cb1-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">mart =</span> ensembl</span>
<span id="cb1-66"><a href="#cb1-66" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-67"><a href="#cb1-67" aria-hidden="true" tabindex="-1"></a>  res_df_with_symbols <span class="ot">&lt;-</span> <span class="fu">merge</span>(</span>
<span id="cb1-68"><a href="#cb1-68" aria-hidden="true" tabindex="-1"></a>    gene_symbols, </span>
<span id="cb1-69"><a href="#cb1-69" aria-hidden="true" tabindex="-1"></a>    res_df, </span>
<span id="cb1-70"><a href="#cb1-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">by.x =</span> <span class="st">"entrezgene_id"</span>, </span>
<span id="cb1-71"><a href="#cb1-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">by.y =</span> <span class="st">"row.names"</span></span>
<span id="cb1-72"><a href="#cb1-72" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb1-73"><a href="#cb1-73" aria-hidden="true" tabindex="-1"></a>  res_df_with_symbols <span class="ot">&lt;-</span> res_df_with_symbols[<span class="fu">order</span>(res_df_with_symbols<span class="sc">$</span>padj), ]</span>
<span id="cb1-74"><a href="#cb1-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res_df_with_symbols)</span>
<span id="cb1-75"><a href="#cb1-75" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-76"><a href="#cb1-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-77"><a href="#cb1-77" aria-hidden="true" tabindex="-1"></a>res_annot <span class="ot">&lt;-</span> <span class="fu">lapply</span>(res, convertEntrezToGeneSymbol)</span>
<span id="cb1-78"><a href="#cb1-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-79"><a href="#cb1-79" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-80"><a href="#cb1-80" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter to remove genes with NA logFC and perform FDR correction</span></span>
<span id="cb1-81"><a href="#cb1-81" aria-hidden="true" tabindex="-1"></a>filter_adjust <span class="ot">&lt;-</span> <span class="cf">function</span>(x) { </span>
<span id="cb1-82"><a href="#cb1-82" aria-hidden="true" tabindex="-1"></a>  res_df <span class="ot">&lt;-</span> x[<span class="fu">complete.cases</span>(x<span class="sc">$</span>log2FoldChange), ]</span>
<span id="cb1-83"><a href="#cb1-83" aria-hidden="true" tabindex="-1"></a>  res_df<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(res_df<span class="sc">$</span>pvalue, <span class="at">method =</span> <span class="st">"fdr"</span>)</span>
<span id="cb1-84"><a href="#cb1-84" aria-hidden="true" tabindex="-1"></a>  res_df<span class="sc">$</span>padj <span class="ot">&lt;-</span> <span class="cn">NULL</span></span>
<span id="cb1-85"><a href="#cb1-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(res_df)</span>
<span id="cb1-86"><a href="#cb1-86" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-87"><a href="#cb1-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-88"><a href="#cb1-88" aria-hidden="true" tabindex="-1"></a>res_annot_filt <span class="ot">&lt;-</span> <span class="fu">lapply</span>(res_annot, filter_adjust)</span>
<span id="cb1-89"><a href="#cb1-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-90"><a href="#cb1-90" aria-hidden="true" tabindex="-1"></a><span class="co"># Get results df for each contrast</span></span>
<span id="cb1-91"><a href="#cb1-91" aria-hidden="true" tabindex="-1"></a><span class="co"># Define a function to process each dataset</span></span>
<span id="cb1-92"><a href="#cb1-92" aria-hidden="true" tabindex="-1"></a>get_deg_res <span class="ot">&lt;-</span> <span class="cf">function</span>(data) {</span>
<span id="cb1-93"><a href="#cb1-93" aria-hidden="true" tabindex="-1"></a>  data <span class="sc">%&gt;%</span></span>
<span id="cb1-94"><a href="#cb1-94" aria-hidden="true" tabindex="-1"></a>    <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb1-95"><a href="#cb1-95" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">distinct</span>(entrezgene_id, <span class="at">.keep_all =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb1-96"><a href="#cb1-96" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(<span class="fu">complete.cases</span>(log2FoldChange))</span>
<span id="cb1-97"><a href="#cb1-97" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb1-98"><a href="#cb1-98" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-99"><a href="#cb1-99" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the function to each dataset</span></span>
<span id="cb1-100"><a href="#cb1-100" aria-hidden="true" tabindex="-1"></a>E2vsE3_rna <span class="ot">&lt;-</span> <span class="fu">get_deg_res</span>(res_annot_filt[[<span class="dv">1</span>]])</span>
<span id="cb1-101"><a href="#cb1-101" aria-hidden="true" tabindex="-1"></a>E4vsE3_rna <span class="ot">&lt;-</span> <span class="fu">get_deg_res</span>(res_annot_filt[[<span class="dv">2</span>]])</span>
<span id="cb1-102"><a href="#cb1-102" aria-hidden="true" tabindex="-1"></a>E4vsE2_rna <span class="ot">&lt;-</span> <span class="fu">get_deg_res</span>(res_annot_filt[[<span class="dv">3</span>]])</span>
<span id="cb1-103"><a href="#cb1-103" aria-hidden="true" tabindex="-1"></a>KOvsE3_rna <span class="ot">&lt;-</span> <span class="fu">get_deg_res</span>(res_annot_filt[[<span class="dv">4</span>]])</span>
<span id="cb1-104"><a href="#cb1-104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-105"><a href="#cb1-105" aria-hidden="true" tabindex="-1"></a>E2vsE3_rna_sig <span class="ot">&lt;-</span> E2vsE3_rna[E2vsE3_rna<span class="sc">$</span>fdr<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span>
<span id="cb1-106"><a href="#cb1-106" aria-hidden="true" tabindex="-1"></a>E4vsE3_rna_sig <span class="ot">&lt;-</span> E4vsE3_rna[E4vsE3_rna<span class="sc">$</span>fdr<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span>
<span id="cb1-107"><a href="#cb1-107" aria-hidden="true" tabindex="-1"></a>E4vsE2_rna_sig <span class="ot">&lt;-</span> E4vsE2_rna[E4vsE2_rna<span class="sc">$</span>fdr<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span>
<span id="cb1-108"><a href="#cb1-108" aria-hidden="true" tabindex="-1"></a>KOvsE3_rna_sig <span class="ot">&lt;-</span> KOvsE3_rna[KOvsE3_rna<span class="sc">$</span>fdr<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="do">### Differential chromatin accessibility analysis using DeSeq2 </span><span class="al">###</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in ATAC-seq counts</span></span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Peaks were called using MACS3 with a p-value cut off of 0.01</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>atac_counts <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/APOE_microglia_ATACseq_counts.csv"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(atac_counts) <span class="ot">&lt;-</span> atac_counts<span class="sc">$</span>Peak</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>atac_f_counts <span class="ot">&lt;-</span> atac_counts[,<span class="sc">-</span><span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">6</span>)]</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove KO samples with high APOE expression</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>atac_f2_counts <span class="ot">&lt;-</span> atac_f_counts[, <span class="sc">!</span><span class="fu">names</span>(atac_f_counts) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">"KO_1"</span>, <span class="st">"KO_2"</span>)]</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Get peaks</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>atac_peaks <span class="ot">&lt;-</span> atac_counts[, <span class="fu">c</span>(<span class="st">"Chr"</span>, <span class="st">"Start"</span>, <span class="st">"End"</span>, <span class="st">"Peak"</span>)]</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Set up design matrix for differential accessibility analysis</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"E3"</span>, <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"KO"</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">"E4"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"E2"</span>, <span class="dv">5</span>)))</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> group)</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">levels</span>(group)</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Keep peaks with counts that are meaningful for statistical analysis</span></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>keep <span class="ot">&lt;-</span> <span class="fu">filterByExpr</span>(atac_f2_counts, design)</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>atac_f3_counts <span class="ot">&lt;-</span> atac_f2_counts[keep,]</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a><span class="co"># DESeq2 analysis</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>meta <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">samples =</span> <span class="fu">colnames</span>(atac_f3_counts), group)</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>atac_dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(atac_f3_counts, meta, <span class="sc">~</span>group)</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>atac_dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(atac_dds)</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to generate differential accessibility results</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>get_dar_res <span class="ot">&lt;-</span> <span class="cf">function</span>(contrast) {</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>  dar <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">results</span>(atac_dds, <span class="at">contrast =</span> contrast))</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>  dar<span class="sc">$</span>Peak <span class="ot">&lt;-</span> <span class="fu">row.names</span>(dar)</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  dar <span class="ot">&lt;-</span> <span class="fu">merge</span>(atac_peaks, dar, <span class="at">by =</span> <span class="st">"Peak"</span>)</span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  dar <span class="ot">&lt;-</span> dar[<span class="fu">order</span>(dar<span class="sc">$</span>pvalue, <span class="at">decreasing =</span> <span class="cn">FALSE</span>), ]</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>  dar<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(dar<span class="sc">$</span>pvalue, <span class="at">method =</span> <span class="st">"fdr"</span>)</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>  dar<span class="sc">$</span>diffAccessible <span class="ot">&lt;-</span> <span class="st">"Not sig"</span></span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>  dar<span class="sc">$</span>diffAccessible[dar<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> dar<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"Up"</span></span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>  dar<span class="sc">$</span>diffAccessible[dar<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> dar<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"Down"</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(dar)</span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-41"><a href="#cb2-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-42"><a href="#cb2-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate DAR results for each contrast</span></span>
<span id="cb2-43"><a href="#cb2-43" aria-hidden="true" tabindex="-1"></a>E2vsE3_dar <span class="ot">&lt;-</span> <span class="fu">get_dar_res</span>(<span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"E2"</span>, <span class="st">"E3"</span>))</span>
<span id="cb2-44"><a href="#cb2-44" aria-hidden="true" tabindex="-1"></a>E4vsE2_dar <span class="ot">&lt;-</span> <span class="fu">get_dar_res</span>(<span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"E4"</span>, <span class="st">"E2"</span>))</span>
<span id="cb2-45"><a href="#cb2-45" aria-hidden="true" tabindex="-1"></a>E4vsE3_dar <span class="ot">&lt;-</span> <span class="fu">get_dar_res</span>(<span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"E4"</span>, <span class="st">"E3"</span>))</span>
<span id="cb2-46"><a href="#cb2-46" aria-hidden="true" tabindex="-1"></a>KOvsE3_dar <span class="ot">&lt;-</span> <span class="fu">get_dar_res</span>(<span class="fu">c</span>(<span class="st">"group"</span>, <span class="st">"KO"</span>, <span class="st">"E3"</span>))</span>
<span id="cb2-47"><a href="#cb2-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-48"><a href="#cb2-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotate peaks</span></span>
<span id="cb2-49"><a href="#cb2-49" aria-hidden="true" tabindex="-1"></a>txdb <span class="ot">&lt;-</span> TxDb.Hsapiens.UCSC.hg38.knownGene</span>
<span id="cb2-50"><a href="#cb2-50" aria-hidden="true" tabindex="-1"></a>peaks_annot <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">as.data.frame</span>(<span class="fu">annotatePeak</span>(</span>
<span id="cb2-51"><a href="#cb2-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">makeGRangesFromDataFrame</span>(atac_peaks[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]), </span>
<span id="cb2-52"><a href="#cb2-52" aria-hidden="true" tabindex="-1"></a>  <span class="at">TxDb =</span> txdb, </span>
<span id="cb2-53"><a href="#cb2-53" aria-hidden="true" tabindex="-1"></a>  <span class="at">tssRegion =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">3000</span>, <span class="dv">3000</span>), </span>
<span id="cb2-54"><a href="#cb2-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">annoDb =</span> <span class="st">"org.Hs.eg.db"</span></span>
<span id="cb2-55"><a href="#cb2-55" aria-hidden="true" tabindex="-1"></a>)))</span>
<span id="cb2-56"><a href="#cb2-56" aria-hidden="true" tabindex="-1"></a>peaks_annot<span class="sc">$</span>peak <span class="ot">&lt;-</span> <span class="fu">row.names</span>(atac_peaks)</span>
<span id="cb2-57"><a href="#cb2-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-58"><a href="#cb2-58" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to merge DAR with annotation and add contrast information</span></span>
<span id="cb2-59"><a href="#cb2-59" aria-hidden="true" tabindex="-1"></a>annotate_dar <span class="ot">&lt;-</span> <span class="cf">function</span>(dar, contrast_name) {</span>
<span id="cb2-60"><a href="#cb2-60" aria-hidden="true" tabindex="-1"></a>  dar_annot <span class="ot">&lt;-</span> <span class="fu">merge</span>(</span>
<span id="cb2-61"><a href="#cb2-61" aria-hidden="true" tabindex="-1"></a>    dar, </span>
<span id="cb2-62"><a href="#cb2-62" aria-hidden="true" tabindex="-1"></a>    peaks_annot, </span>
<span id="cb2-63"><a href="#cb2-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">by.x =</span> <span class="fu">c</span>(<span class="st">"Chr"</span>, <span class="st">"Start"</span>, <span class="st">"End"</span>, <span class="st">"Peak"</span>), </span>
<span id="cb2-64"><a href="#cb2-64" aria-hidden="true" tabindex="-1"></a>    <span class="at">by.y =</span> <span class="fu">c</span>(<span class="st">"seqnames"</span>, <span class="st">"start"</span>, <span class="st">"end"</span>, <span class="st">"peak"</span>)</span>
<span id="cb2-65"><a href="#cb2-65" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-66"><a href="#cb2-66" aria-hidden="true" tabindex="-1"></a>  dar_annot<span class="sc">$</span>contrast <span class="ot">&lt;-</span> contrast_name</span>
<span id="cb2-67"><a href="#cb2-67" aria-hidden="true" tabindex="-1"></a>  dar_annot <span class="ot">&lt;-</span> dar_annot[<span class="fu">order</span>(dar_annot<span class="sc">$</span>pvalue, <span class="at">decreasing =</span> <span class="cn">FALSE</span>), ]</span>
<span id="cb2-68"><a href="#cb2-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(dar_annot)</span>
<span id="cb2-69"><a href="#cb2-69" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-70"><a href="#cb2-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-71"><a href="#cb2-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Annotate DAR results for each contrast</span></span>
<span id="cb2-72"><a href="#cb2-72" aria-hidden="true" tabindex="-1"></a>E2vsE3_atac_annot <span class="ot">&lt;-</span> <span class="fu">annotate_dar</span>(E2vsE3_dar, <span class="st">"E2vsE3"</span>)</span>
<span id="cb2-73"><a href="#cb2-73" aria-hidden="true" tabindex="-1"></a>E4vsE3_atac_annot <span class="ot">&lt;-</span> <span class="fu">annotate_dar</span>(E4vsE3_dar, <span class="st">"E4vsE3"</span>)</span>
<span id="cb2-74"><a href="#cb2-74" aria-hidden="true" tabindex="-1"></a>E4vsE2_atac_annot <span class="ot">&lt;-</span> <span class="fu">annotate_dar</span>(E4vsE2_dar, <span class="st">"E4vsE2"</span>)</span>
<span id="cb2-75"><a href="#cb2-75" aria-hidden="true" tabindex="-1"></a>KOvsE3_atac_annot <span class="ot">&lt;-</span> <span class="fu">annotate_dar</span>(KOvsE3_dar, <span class="st">"KOvsE3"</span>)</span>
<span id="cb2-76"><a href="#cb2-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-77"><a href="#cb2-77" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter significant DAR results</span></span>
<span id="cb2-78"><a href="#cb2-78" aria-hidden="true" tabindex="-1"></a>E2vsE3_atac_sig <span class="ot">&lt;-</span> E2vsE3_atac_annot[E2vsE3_atac_annot<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb2-79"><a href="#cb2-79" aria-hidden="true" tabindex="-1"></a>E4vsE3_atac_sig <span class="ot">&lt;-</span> E4vsE3_atac_annot[E4vsE3_atac_annot<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb2-80"><a href="#cb2-80" aria-hidden="true" tabindex="-1"></a>E4vsE2_atac_sig <span class="ot">&lt;-</span> E4vsE2_atac_annot[E4vsE2_atac_annot<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb2-81"><a href="#cb2-81" aria-hidden="true" tabindex="-1"></a>KOvsE3_atac_sig <span class="ot">&lt;-</span> KOvsE3_atac_annot[KOvsE3_atac_annot<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Figure 1: Transcriptomic and epigenomic profiling of xenotransplanted microglia reveals changes to their regulation in Alzheimer’s disease across the different <em>APOE</em> isoforms.</strong></p>
<p><img src="figures/chapter5_workflow.png" class="img-fluid"></p>
<p><strong>Figure 1a) Experimental design for xenotransplantation of iPSC-derived human microglia into the brains of AppNL-G-F mice (<em>APOE2</em> = 5, <em>APOE3</em> = 5, E4 = 4, <em>APOE</em>-KO = 2) and high-quality transcriptomic data across 17 mice (<em>APOE2</em> = 5, <em>APOE4</em> = 4, <em>APOE4</em> = 5, <em>APOE</em>-KO = 3), followed by transcriptomic and chromatin accessibility profiling at 12 months.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 1b: apoe expression ##</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>apoe <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">"348"</span>, <span class="at">intgroup=</span><span class="st">"group"</span>, <span class="at">returnData =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="fu">c</span>(<span class="st">"E4"</span>, <span class="st">"KO"</span>), <span class="fu">c</span>(<span class="st">"E3"</span>, <span class="st">"KO"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>,<span class="st">"KO"</span>), <span class="fu">c</span>(<span class="st">"E4"</span>, <span class="st">"E3"</span>))</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>apoe_exp <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(apoe, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>count, <span class="at">fill=</span>group)) <span class="sc">+</span> </span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Counts"</span>) <span class="sc">+</span> </span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"APOE"</span>) <span class="sc">+</span> </span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face=</span><span class="fu">c</span>(<span class="st">"italic"</span>))) <span class="sc">+</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons, <span class="at">method=</span><span class="st">"t.test"</span>, <span class="at">size=</span><span class="dv">4</span>) <span class="sc">+</span> </span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>() <span class="sc">+</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill=</span><span class="cn">NULL</span>)</span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(apoe_exp)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_1b-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 1b) Boxplot of expression profiles of <em>APOE</em>, confirming the knockout.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 1c: chromatin accessibility around APOE locus ##</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Get TSS of APOE and plot datatrack  </span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>hg38_genes <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">genes</span>(TxDb.Hsapiens.UCSC.hg38.knownGene))</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>start <span class="ot">&lt;-</span> <span class="fu">start</span>(hg38_genes[hg38_genes<span class="sc">$</span>gene_id <span class="sc">%in%</span> <span class="st">"348"</span>,]) <span class="sc">-</span> <span class="dv">2000</span> </span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>end <span class="ot">&lt;-</span> <span class="fu">end</span>(hg38_genes[hg38_genes<span class="sc">$</span>gene_id <span class="sc">%in%</span> <span class="st">"348"</span>,]) <span class="sc">+</span> <span class="dv">2000</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot data tracks </span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>track_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">E3 =</span> <span class="st">"data/E3-1.bw"</span>, <span class="at">E4 =</span> <span class="st">"data/E4-1.bw"</span>, <span class="at">E2 =</span> <span class="st">"data/E2-1.bw"</span>, <span class="at">KO =</span> <span class="st">"data/KO-1.bw"</span>)</span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a><span class="co"># Set apoe track region</span></span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>apoe_region <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">toGRanges</span>(<span class="fu">data.frame</span>(<span class="at">chr=</span><span class="st">"chr19"</span>, <span class="at">start=</span>start, <span class="at">end=</span>end)))</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Set plotting parameters</span></span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>pp <span class="ot">&lt;-</span> <span class="fu">getDefaultPlotParams</span>(<span class="at">plot.type=</span><span class="dv">1</span>)</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>leftmargin <span class="ot">&lt;-</span> <span class="fl">0.15</span></span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>topmargin <span class="ot">&lt;-</span> <span class="dv">15</span></span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>bottommargin <span class="ot">&lt;-</span> <span class="dv">15</span></span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>ideogramheight <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>data1inmargin <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">zoom =</span> apoe_region, <span class="at">cex=</span><span class="fl">1.7</span>, <span class="at">plot.params =</span> pp)</span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAddBaseNumbers</span>(kp, <span class="at">tick.dist =</span> <span class="dv">3000</span>, <span class="at">minor.tick.dist =</span> <span class="dv">1500</span>, <span class="at">add.units =</span> <span class="cn">TRUE</span>, <span class="at">cex=</span><span class="fl">1.5</span>, <span class="at">digits =</span> <span class="dv">6</span>)</span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>apoe_data <span class="ot">&lt;-</span> <span class="fu">suppressMessages</span>(<span class="fu">makeGenesDataFromTxDb</span>(TxDb.Hsapiens.UCSC.hg38.knownGene, <span class="at">karyoplot=</span>kp, <span class="at">plot.transcripts =</span> <span class="cn">TRUE</span>, <span class="at">plot.transcripts.structure =</span> <span class="cn">TRUE</span>))</span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>apoe_data<span class="sc">$</span>genes<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="st">"APOE"</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting tracks</span></span>
<span id="cb4-31"><a href="#cb4-31" aria-hidden="true" tabindex="-1"></a>plot_track <span class="ot">&lt;-</span> <span class="cf">function</span>(track_name, track_position, track_colour, r1_value) {</span>
<span id="cb4-32"><a href="#cb4-32" aria-hidden="true" tabindex="-1"></a>  bigwig.file <span class="ot">&lt;-</span> track_list[[track_name]]</span>
<span id="cb4-33"><a href="#cb4-33" aria-hidden="true" tabindex="-1"></a>  track_margin <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb4-34"><a href="#cb4-34" aria-hidden="true" tabindex="-1"></a>  genes_track_at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="at">r0=</span><span class="dv">0</span>, <span class="at">r1=</span>track_margin)</span>
<span id="cb4-35"><a href="#cb4-35" aria-hidden="true" tabindex="-1"></a>  at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(track_position, <span class="fu">length</span>(track_list), <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r1 <span class="sc">+</span> track_margin, <span class="at">margin =</span> <span class="fl">0.2</span>)</span>
<span id="cb4-36"><a href="#cb4-36" aria-hidden="true" tabindex="-1"></a>  kp <span class="ot">&lt;-</span> <span class="fu">kpPlotGenes</span>(kp, <span class="at">data=</span>apoe_data, <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r0, <span class="at">r1=</span>genes_track_at<span class="sc">$</span>r1, <span class="at">gene.name.cex=</span><span class="dv">2</span>)</span>
<span id="cb4-37"><a href="#cb4-37" aria-hidden="true" tabindex="-1"></a>  kp <span class="ot">&lt;-</span> <span class="fu">kpPlotBigWig</span>(kp, <span class="at">data=</span>bigwig.file, <span class="at">ymax=</span><span class="st">"visible.region"</span>, <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>r1_value, <span class="at">col =</span> track_colour)</span>
<span id="cb4-38"><a href="#cb4-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpAxis</span>(kp, <span class="at">ymin=</span><span class="dv">0</span>, <span class="at">ymax=</span><span class="dv">116</span>, <span class="at">numticks =</span> <span class="dv">2</span>, <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, <span class="at">cex =</span> <span class="fl">1.5</span>)</span>
<span id="cb4-39"><a href="#cb4-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kpAddLabels</span>(kp, <span class="at">labels =</span> track_name, <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, <span class="at">cex=</span><span class="dv">2</span>, <span class="at">label.margin =</span> <span class="fl">0.035</span>, <span class="at">col =</span> track_colour)</span>
<span id="cb4-40"><a href="#cb4-40" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb4-41"><a href="#cb4-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-42"><a href="#cb4-42" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_track</span>(<span class="st">"KO"</span>, <span class="dv">1</span>, <span class="st">"#4D6D93FF"</span>, <span class="fl">0.26745</span>)</span>
<span id="cb4-43"><a href="#cb4-43" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_track</span>(<span class="st">"E4"</span>, <span class="dv">2</span>, <span class="st">"#6FB382FF"</span>, <span class="fl">0.52827</span>)</span>
<span id="cb4-44"><a href="#cb4-44" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_track</span>(<span class="st">"E3"</span>, <span class="dv">3</span>, <span class="st">"#DCCA2CFF"</span>, <span class="fl">0.73274</span>)</span>
<span id="cb4-45"><a href="#cb4-45" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_track</span>(<span class="st">"E2"</span>, <span class="dv">4</span>, <span class="st">"#92BBD9FF"</span>, <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_1c-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 1c) Genome tracks showing chromatin accessibility signals of all APOE groups around the <em>APOE</em> locus.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 1d: no of DEGs ##</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>KOvsE3_rna<span class="sc">$</span>comparison <span class="ot">&lt;-</span> <span class="st">"KOvsE3"</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>E4vsE2_rna<span class="sc">$</span>comparison <span class="ot">&lt;-</span> <span class="st">"E4vsE2"</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>E4vsE3_rna<span class="sc">$</span>comparison <span class="ot">&lt;-</span> <span class="st">"E4vsE3"</span></span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>E2vsE3_rna<span class="sc">$</span>comparison <span class="ot">&lt;-</span> <span class="st">"E2vsE3"</span></span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>comparisons <span class="ot">&lt;-</span> <span class="fu">rbind</span>(E2vsE3_rna, E4vsE3_rna, E4vsE2_rna, KOvsE3_rna)</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>comparisons<span class="sc">$</span>diffExpressed <span class="ot">&lt;-</span> <span class="st">"Not sig"</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>comparisons<span class="sc">$</span>diffExpressed[comparisons<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> comparisons<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"Up"</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>comparisons<span class="sc">$</span>diffExpressed[comparisons<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">&amp;</span> comparisons<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"Down"</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>comparisons<span class="sc">$</span>diffExpressed <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparisons<span class="sc">$</span>diffExpressed, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Up"</span>, <span class="st">"Down"</span>, <span class="st">"Not sig"</span>))</span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-13"><a href="#cb5-13" aria-hidden="true" tabindex="-1"></a>degs_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(comparisons[<span class="sc">!</span>comparisons<span class="sc">$</span>diffExpressed <span class="sc">==</span> <span class="st">"Not sig"</span>,], <span class="fu">aes</span>(<span class="at">x=</span>comparison, <span class="at">fill=</span>diffExpressed)) <span class="sc">+</span> </span>
<span id="cb5-14"><a href="#cb5-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"count"</span>) <span class="sc">+</span> </span>
<span id="cb5-15"><a href="#cb5-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb5-16"><a href="#cb5-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Number of DEGs"</span>) <span class="sc">+</span></span>
<span id="cb5-17"><a href="#cb5-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">ghibli_palette</span>(<span class="st">"YesterdayLight"</span>, <span class="at">direction=</span><span class="sc">-</span><span class="dv">1</span>), <span class="at">labels=</span><span class="fu">c</span>(<span class="st">'Up'</span>, <span class="st">'Down'</span>), <span class="at">name=</span><span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb5-18"><a href="#cb5-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb5-19"><a href="#cb5-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">stat =</span> <span class="st">'count'</span>, <span class="fu">aes</span>(<span class="at">label=</span><span class="fu">after_stat</span>(count), <span class="at">y=</span><span class="fu">after_stat</span>(count) <span class="sc">+</span> <span class="dv">1</span>), <span class="at">position=</span><span class="fu">position_stack</span>(<span class="at">vjust=</span><span class="fl">0.5</span>), <span class="at">size=</span><span class="fl">6.5</span>, <span class="at">colour=</span><span class="st">"black"</span>) <span class="sc">+</span> </span>
<span id="cb5-20"><a href="#cb5-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">20</span>), <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>))</span>
<span id="cb5-21"><a href="#cb5-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-22"><a href="#cb5-22" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(degs_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_1d-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 1d) Stacked barplot of the number of differentially expressed genes (DEGs; FDR &lt; 0.05) identified through pairwise comparisons across the experimental groups.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 1e: no of DARs with DEG overlap ## </span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to get data gene names annotated to significant peaks</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>create_gene_df <span class="ot">&lt;-</span> <span class="cf">function</span>(sig_data, symbol_col, diff_col, contrast_col, assay_name) {</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  gene_df <span class="ot">&lt;-</span> sig_data[, <span class="fu">c</span>(symbol_col, diff_col, contrast_col)]</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>  gene_df<span class="sc">$</span>assay <span class="ot">&lt;-</span> assay_name</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(gene_df) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"gene"</span>, <span class="st">"direction"</span>, <span class="st">"contrast"</span>, <span class="st">"assay"</span>)</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(gene_df)</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Create ATAC-seq gene data frames</span></span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>E2vsE3_atac_genes <span class="ot">&lt;-</span> <span class="fu">create_gene_df</span>(E2vsE3_atac_sig, <span class="st">"SYMBOL"</span>, <span class="st">"diffAccessible"</span>, <span class="st">"contrast"</span>, <span class="st">"ATAC-seq"</span>)</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>E4vsE3_atac_genes <span class="ot">&lt;-</span> <span class="fu">create_gene_df</span>(E4vsE3_atac_sig, <span class="st">"SYMBOL"</span>, <span class="st">"diffAccessible"</span>, <span class="st">"contrast"</span>, <span class="st">"ATAC-seq"</span>)</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>E4vsE2_atac_genes <span class="ot">&lt;-</span> <span class="fu">create_gene_df</span>(E4vsE2_atac_sig, <span class="st">"SYMBOL"</span>, <span class="st">"diffAccessible"</span>, <span class="st">"contrast"</span>, <span class="st">"ATAC-seq"</span>)</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>KOvsE3_atac_genes <span class="ot">&lt;-</span> <span class="fu">create_gene_df</span>(KOvsE3_atac_sig, <span class="st">"SYMBOL"</span>, <span class="st">"diffAccessible"</span>, <span class="st">"contrast"</span>, <span class="st">"ATAC-seq"</span>)</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine ATAC-seq gene data frames</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>atac_genes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(E2vsE3_atac_genes, E4vsE3_atac_genes, E4vsE2_atac_genes, KOvsE3_atac_genes)</span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Create RNA-seq gene data frames</span></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>E2vsE3_rna_genes <span class="ot">&lt;-</span> <span class="fu">create_gene_df</span>(comparisons[comparisons<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> comparisons<span class="sc">$</span>comparison <span class="sc">==</span> <span class="st">"E2vsE3"</span>,], <span class="st">"hgnc_symbol"</span>, <span class="st">"diffExpressed"</span>, <span class="st">"comparison"</span>, <span class="st">"RNA-seq"</span>)</span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>E4vsE3_rna_genes <span class="ot">&lt;-</span> <span class="fu">create_gene_df</span>(comparisons[comparisons<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> comparisons<span class="sc">$</span>comparison <span class="sc">==</span> <span class="st">"E4vsE3"</span>,], <span class="st">"hgnc_symbol"</span>, <span class="st">"diffExpressed"</span>, <span class="st">"comparison"</span>, <span class="st">"RNA-seq"</span>)</span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>E4vsE2_rna_genes <span class="ot">&lt;-</span> <span class="fu">create_gene_df</span>(comparisons[comparisons<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> comparisons<span class="sc">$</span>comparison <span class="sc">==</span> <span class="st">"E4vsE2"</span>,], <span class="st">"hgnc_symbol"</span>, <span class="st">"diffExpressed"</span>, <span class="st">"comparison"</span>, <span class="st">"RNA-seq"</span>)</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>KOvsE3_rna_genes <span class="ot">&lt;-</span> <span class="fu">create_gene_df</span>(comparisons[comparisons<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> comparisons<span class="sc">$</span>comparison <span class="sc">==</span> <span class="st">"KOvsE3"</span>,], <span class="st">"hgnc_symbol"</span>, <span class="st">"diffExpressed"</span>, <span class="st">"comparison"</span>, <span class="st">"RNA-seq"</span>)</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine RNA-seq gene data frames</span></span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>rna_genes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(E2vsE3_rna_genes, E4vsE3_rna_genes, E4vsE2_rna_genes, KOvsE3_rna_genes)</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-29"><a href="#cb6-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine RNA-seq and ATAC-seq gene data frames</span></span>
<span id="cb6-30"><a href="#cb6-30" aria-hidden="true" tabindex="-1"></a>rna_atac_genes <span class="ot">&lt;-</span> <span class="fu">rbind</span>(atac_genes, rna_genes)</span>
<span id="cb6-31"><a href="#cb6-31" aria-hidden="true" tabindex="-1"></a>rna_atac_genes<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Upregulated"</span>, <span class="st">"Up"</span>, rna_atac_genes<span class="sc">$</span>direction)</span>
<span id="cb6-32"><a href="#cb6-32" aria-hidden="true" tabindex="-1"></a>rna_atac_genes<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"Downregulated"</span>, <span class="st">"Down"</span>, rna_atac_genes<span class="sc">$</span>direction)</span>
<span id="cb6-33"><a href="#cb6-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-34"><a href="#cb6-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Summarise gene counts</span></span>
<span id="cb6-35"><a href="#cb6-35" aria-hidden="true" tabindex="-1"></a>gene_counts <span class="ot">&lt;-</span> rna_atac_genes <span class="sc">%&gt;%</span></span>
<span id="cb6-36"><a href="#cb6-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">!=</span> <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-37"><a href="#cb6-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(contrast, assay) <span class="sc">%&gt;%</span></span>
<span id="cb6-38"><a href="#cb6-38" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">'drop'</span>)</span>
<span id="cb6-39"><a href="#cb6-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-40"><a href="#cb6-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Count the number of overlapping genes</span></span>
<span id="cb6-41"><a href="#cb6-41" aria-hidden="true" tabindex="-1"></a>overlap_counts <span class="ot">&lt;-</span> rna_atac_genes <span class="sc">%&gt;%</span></span>
<span id="cb6-42"><a href="#cb6-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(gene <span class="sc">!=</span> <span class="st">""</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-43"><a href="#cb6-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(contrast, gene) <span class="sc">%&gt;%</span></span>
<span id="cb6-44"><a href="#cb6-44" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">'drop'</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-45"><a href="#cb6-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(count <span class="sc">&gt;</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-46"><a href="#cb6-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(contrast) <span class="sc">%&gt;%</span></span>
<span id="cb6-47"><a href="#cb6-47" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">overlap_count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">'drop'</span>)</span>
<span id="cb6-48"><a href="#cb6-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-49"><a href="#cb6-49" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge gene counts with overlap counts</span></span>
<span id="cb6-50"><a href="#cb6-50" aria-hidden="true" tabindex="-1"></a>gene_counts <span class="ot">&lt;-</span> dplyr<span class="sc">::</span><span class="fu">left_join</span>(gene_counts, overlap_counts, <span class="at">by =</span> <span class="st">"contrast"</span>)</span>
<span id="cb6-51"><a href="#cb6-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-52"><a href="#cb6-52" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot the results</span></span>
<span id="cb6-53"><a href="#cb6-53" aria-hidden="true" tabindex="-1"></a>dars_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(gene_counts[gene_counts<span class="sc">$</span>assay <span class="sc">==</span> <span class="st">"ATAC-seq"</span>, ], <span class="fu">aes</span>(<span class="at">x =</span> contrast, <span class="at">y =</span> count, <span class="at">fill =</span> assay)) <span class="sc">+</span></span>
<span id="cb6-54"><a href="#cb6-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb6-55"><a href="#cb6-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="fu">aes</span>(<span class="at">y =</span> overlap_count, <span class="at">fill =</span> <span class="st">"Overlap"</span>), <span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">position =</span> <span class="st">"dodge"</span>) <span class="sc">+</span></span>
<span id="cb6-56"><a href="#cb6-56" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">paste</span>(overlap_count, <span class="st">"/"</span>, count)), <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>), <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">0.5</span>, <span class="at">size =</span> <span class="fl">4.5</span>) <span class="sc">+</span></span>
<span id="cb6-57"><a href="#cb6-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">""</span>, <span class="at">y =</span> <span class="st">"Number of DARs"</span>, <span class="at">fill =</span> <span class="st">"Assay"</span>) <span class="sc">+</span></span>
<span id="cb6-58"><a href="#cb6-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="dv">0</span>, <span class="dv">80</span>) <span class="sc">+</span></span>
<span id="cb6-59"><a href="#cb6-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"MarnieLight2"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">'ATAC only'</span>, <span class="st">'ATAC and RNA'</span>), <span class="at">name =</span> <span class="st">""</span>) <span class="sc">+</span></span>
<span id="cb6-60"><a href="#cb6-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span></span>
<span id="cb6-61"><a href="#cb6-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>))</span>
<span id="cb6-62"><a href="#cb6-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-63"><a href="#cb6-63" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(dars_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_1e-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 1e) Stacked barplot of the number of differentially accessible regions (DARs) and how many of these overlap with the DEGs based on peak-to-gene annotation.</strong></p>
<p><strong>Figure 2: Microglia-expressing the different APOE isoforms exhibit widespread differences in gene regulation.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 2s: E2vsE3 DEGs volcano plot ##</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a><span class="co"># set factor level </span></span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>comparisons<span class="sc">$</span>diffExpressed <span class="ot">&lt;-</span> <span class="fu">factor</span>(comparisons<span class="sc">$</span>diffExpressed, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Up"</span>, <span class="st">"Down"</span>, <span class="st">"Not sig"</span>))</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="co"># fill in missing gene symbol for one of the top sig genes</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>comparisons<span class="sc">$</span>hgnc_symbol[comparisons<span class="sc">$</span>entrezgene_id<span class="sc">==</span><span class="st">"102724316"</span>] <span class="ot">&lt;-</span> <span class="st">"SVIL-AS1"</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a><span class="co"># get top 20 genes for labelling </span></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">head</span>(comparisons[comparisons<span class="sc">$</span>comparison <span class="sc">==</span> <span class="st">"E2vsE3"</span>, ], <span class="dv">10</span>)</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>E2vsE3_volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(comparisons[comparisons<span class="sc">$</span>comparison<span class="sc">==</span><span class="st">"E2vsE3"</span>,], </span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">aes</span>(<span class="at">x=</span>log2FoldChange, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">colour=</span>diffExpressed)) <span class="sc">+</span> </span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="dv">10</span>) <span class="sc">+</span> </span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#92BBD9FF"</span>, <span class="st">"#92BBD9FF"</span>, <span class="st">"#C3DAEAFF"</span>)) <span class="sc">+</span> </span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb7-18"><a href="#cb7-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"-log10(FDR)"</span>) <span class="sc">+</span> </span>
<span id="cb7-19"><a href="#cb7-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour=</span><span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb7-20"><a href="#cb7-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb7-21"><a href="#cb7-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"E2 vs E3"</span>) <span class="sc">+</span> </span>
<span id="cb7-22"><a href="#cb7-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="at">data =</span> top_genes, <span class="fu">aes</span>(<span class="at">label =</span> hgnc_symbol), <span class="at">box.padding =</span> <span class="fl">0.5</span>, <span class="at">colour=</span><span class="st">"#506777FF"</span>, <span class="at">size=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb7-23"><a href="#cb7-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"slategrey"</span>) <span class="sc">+</span></span>
<span id="cb7-24"><a href="#cb7-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"slategrey"</span>)</span>
<span id="cb7-25"><a href="#cb7-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-26"><a href="#cb7-26" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(E2vsE3_volcano)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_2a-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 2a) Differentially expressed genes in <em>APOE2</em> vs <em>APOE3</em>-expressing microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 2b: E4vsE3 DEGs volcano plot ##</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="co"># get top 20 genes for labelling </span></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">head</span>(comparisons[comparisons<span class="sc">$</span>comparison <span class="sc">==</span> <span class="st">"E4vsE3"</span>, ], <span class="dv">10</span>)</span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>E4vsE3_volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(comparisons[comparisons<span class="sc">$</span>comparison<span class="sc">==</span><span class="st">"E4vsE3"</span>,], </span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">aes</span>(<span class="at">x=</span>log2FoldChange, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">colour=</span>diffExpressed)) <span class="sc">+</span> </span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="dv">10</span>) <span class="sc">+</span> </span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#DCCA2CFF"</span>, <span class="st">"#DCCA2CFF"</span>, <span class="st">"#ECE28BFF"</span>)) <span class="sc">+</span> </span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"-log10(FDR)"</span>) <span class="sc">+</span> </span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour=</span><span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"E4 vs E3"</span>) <span class="sc">+</span> </span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="at">data =</span> top_genes, <span class="fu">aes</span>(<span class="at">label =</span> hgnc_symbol), <span class="at">box.padding =</span> <span class="fl">0.5</span>, <span class="at">colour=</span><span class="st">"#796F18FF"</span>, <span class="at">size=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb8-17"><a href="#cb8-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"slategrey"</span>) <span class="sc">+</span></span>
<span id="cb8-18"><a href="#cb8-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"slategrey"</span>)</span>
<span id="cb8-19"><a href="#cb8-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-20"><a href="#cb8-20" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(E4vsE3_volcano)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_2b-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 2b) Differentially expressed genes in <em>APOE4</em> vs <em>APOE3</em>-expressing microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 2c: E4vsE2 DEGs volcano plot ##</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># get top 20 genes for labelling </span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">head</span>(comparisons[comparisons<span class="sc">$</span>comparison <span class="sc">==</span> <span class="st">"E4vsE2"</span>, ], <span class="dv">10</span>)</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>E4vsE2_volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(comparisons[comparisons<span class="sc">$</span>comparison<span class="sc">==</span><span class="st">"E4vsE2"</span>,], </span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>                         <span class="fu">aes</span>(<span class="at">x=</span>log2FoldChange, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">colour=</span>diffExpressed)) <span class="sc">+</span> </span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="dv">10</span>) <span class="sc">+</span> </span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#6FB382FF"</span>, <span class="st">"#6FB382FF"</span>, <span class="st">"#B1D5BBFF"</span>)) <span class="sc">+</span> </span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"-log10(FDR)"</span>) <span class="sc">+</span> </span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">colour=</span><span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb9-15"><a href="#cb9-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"E4 vs E2"</span>) <span class="sc">+</span> </span>
<span id="cb9-16"><a href="#cb9-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_label_repel</span>(<span class="at">data =</span> top_genes, <span class="fu">aes</span>(<span class="at">label =</span> hgnc_symbol), <span class="at">box.padding =</span> <span class="fl">0.5</span>, <span class="at">colour=</span><span class="st">"#3E6248FF"</span>, <span class="at">size=</span><span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb9-17"><a href="#cb9-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"slategrey"</span>) <span class="sc">+</span></span>
<span id="cb9-18"><a href="#cb9-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>, <span class="at">color =</span> <span class="st">"slategrey"</span>)</span>
<span id="cb9-19"><a href="#cb9-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb9-20"><a href="#cb9-20" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(E4vsE2_volcano)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_2c-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 2c) Differentially expressed genes in <em>APOE4</em> vs <em>APOE2</em>-expressing microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 2d: RRHO E2 and KO ##</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">1234</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Set rrho p-value for KO vs E3</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>KOvsE3_rrho <span class="ot">&lt;-</span> KOvsE3_rna[, <span class="fu">c</span>(<span class="st">"entrezgene_id"</span>, <span class="st">"pvalue"</span>, <span class="st">"log2FoldChange"</span>)]</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>KOvsE3_rrho<span class="sc">$</span>rrho_pvalue <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>up_indices <span class="ot">&lt;-</span> KOvsE3_rrho<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>down_indices <span class="ot">&lt;-</span> KOvsE3_rrho<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span></span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>KOvsE3_rrho<span class="sc">$</span>rrho_pvalue[up_indices] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(KOvsE3_rrho<span class="sc">$</span>pvalue[up_indices]) <span class="sc">*</span> KOvsE3_rrho<span class="sc">$</span>log2FoldChange[up_indices]</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>KOvsE3_rrho<span class="sc">$</span>rrho_pvalue[down_indices] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(KOvsE3_rrho<span class="sc">$</span>pvalue[down_indices]) <span class="sc">*</span> KOvsE3_rrho<span class="sc">$</span>log2FoldChange[down_indices]</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>KOvsE3_rrho <span class="ot">&lt;-</span> KOvsE3_rrho[,<span class="fu">c</span>(<span class="st">"entrezgene_id"</span>, <span class="st">"rrho_pvalue"</span>)]</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a><span class="co"># Set rrho p-value for E2 vs E3</span></span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a>E2vsE3_rrho <span class="ot">&lt;-</span> E2vsE3_rna[, <span class="fu">c</span>(<span class="st">"entrezgene_id"</span>, <span class="st">"pvalue"</span>, <span class="st">"log2FoldChange"</span>)]</span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>E2vsE3_rrho<span class="sc">$</span>rrho_pvalue <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a>up_indices <span class="ot">&lt;-</span> E2vsE3_rrho<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>down_indices <span class="ot">&lt;-</span> E2vsE3_rrho<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span></span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>E2vsE3_rrho<span class="sc">$</span>rrho_pvalue[up_indices] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(E2vsE3_rrho<span class="sc">$</span>pvalue[up_indices]) <span class="sc">*</span> E2vsE3_rrho<span class="sc">$</span>log2FoldChange[up_indices]</span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>E2vsE3_rrho<span class="sc">$</span>rrho_pvalue[down_indices] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(E2vsE3_rrho<span class="sc">$</span>pvalue[down_indices]) <span class="sc">*</span> E2vsE3_rrho<span class="sc">$</span>log2FoldChange[down_indices]</span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a>E2vsE3_rrho <span class="ot">&lt;-</span> E2vsE3_rrho[,<span class="fu">c</span>(<span class="st">"entrezgene_id"</span>, <span class="st">"rrho_pvalue"</span>)]</span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a>E2_KO_rrho <span class="ot">&lt;-</span>  <span class="fu">RRHO2_initialize</span>(<span class="fu">na.omit</span>(E2vsE3_rrho), <span class="fu">na.omit</span>(KOvsE3_rrho), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"KO"</span>), <span class="at">log10.ind=</span><span class="cn">TRUE</span>)</span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>E2_KO_heatmap <span class="ot">&lt;-</span> <span class="fu">RRHO2_heatmap</span>(E2_KO_rrho, <span class="at">colorGradient =</span> <span class="fu">wes_palette</span>(<span class="st">"Zissou1"</span>, <span class="dv">100</span>, <span class="at">type =</span> <span class="st">"continuous"</span>), <span class="at">maximum =</span> <span class="dv">350</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_2d-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 2d) Rank-Rank Hypergeometric Overlap (RRHO) heatmap comparing expression signatures between <em>APOE</em>-KO vs <em>APOE3</em> and <em>APOE2</em> vs <em>APOE3</em>.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 2e: RRHO E4 and KO heatmap ##</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Set rrho p-value for E4 vs E3</span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>E4vsE3_rrho <span class="ot">&lt;-</span> E4vsE3_rna[, <span class="fu">c</span>(<span class="st">"entrezgene_id"</span>, <span class="st">"pvalue"</span>, <span class="st">"log2FoldChange"</span>)]</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>E4vsE3_rrho<span class="sc">$</span>rrho_pvalue <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>up_indices <span class="ot">&lt;-</span> E4vsE3_rrho<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>down_indices <span class="ot">&lt;-</span> E4vsE3_rrho<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>E4vsE3_rrho<span class="sc">$</span>rrho_pvalue[up_indices] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(E4vsE3_rrho<span class="sc">$</span>pvalue[up_indices]) <span class="sc">*</span> E4vsE3_rrho<span class="sc">$</span>log2FoldChange[up_indices]</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>E4vsE3_rrho<span class="sc">$</span>rrho_pvalue[down_indices] <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(E4vsE3_rrho<span class="sc">$</span>pvalue[down_indices]) <span class="sc">*</span> E4vsE3_rrho<span class="sc">$</span>log2FoldChange[down_indices]</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>E4vsE3_rrho <span class="ot">&lt;-</span> E4vsE3_rrho[,<span class="fu">c</span>(<span class="st">"entrezgene_id"</span>, <span class="st">"rrho_pvalue"</span>)]</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>E4_KO_rrho <span class="ot">&lt;-</span>  <span class="fu">RRHO2_initialize</span>(<span class="fu">na.omit</span>(E4vsE3_rrho), <span class="fu">na.omit</span>(KOvsE3_rrho), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"E4"</span>, <span class="st">"KO"</span>), <span class="at">log10.ind=</span><span class="cn">TRUE</span>)</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>E4_KO_heatmap <span class="ot">&lt;-</span> <span class="fu">RRHO2_heatmap</span>(E4_KO_rrho, <span class="at">colorGradient =</span> <span class="fu">wes_palette</span>(<span class="st">"Zissou1"</span>, <span class="dv">100</span>, <span class="at">type =</span> <span class="st">"continuous"</span>), <span class="at">maximum =</span> <span class="dv">350</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_2e-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 2e) Rank-Rank Hypergeometric Overlap (RRHO) heatmap comparing expression signatures between <em>APOE</em>-KO vs <em>APOE3</em> and <em>APOE4</em> vs <em>APOE3</em>.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>E4_uu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>E4_KO_rrho[[<span class="st">"genelist_uu"</span>]][[<span class="st">"gene_list_overlap_uu"</span>]])</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>E4_uu<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="st">"E4_uu"</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>E4_uu<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="st">"E4"</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>E4_uu <span class="ot">&lt;-</span> <span class="fu">merge</span>(E4_uu, E4vsE3_rna, <span class="at">by.x=</span><span class="st">"gene"</span>, <span class="at">by.y=</span><span class="st">"entrezgene_id"</span>)</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(E4_uu) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"log2FoldChange"</span>, <span class="st">"log2FoldChange_E4"</span>, <span class="fu">names</span>(E4_uu))</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>E2_uu <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>E2_KO_rrho[[<span class="st">"genelist_uu"</span>]][[<span class="st">"gene_list_overlap_uu"</span>]])</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>E2_uu<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="st">"E2_uu"</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>E2_uu<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="st">"E2"</span></span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>E2_uu <span class="ot">&lt;-</span> <span class="fu">merge</span>(E2_uu, E2vsE3_rna, <span class="at">by.x=</span><span class="st">"gene"</span>, <span class="at">by.y=</span><span class="st">"entrezgene_id"</span>)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(E2_uu) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"log2FoldChange"</span>, <span class="st">"log2FoldChange_E2"</span>, <span class="fu">names</span>(E2_uu))</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>E4_dd <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>E4_KO_rrho[[<span class="st">"genelist_dd"</span>]][[<span class="st">"gene_list_overlap_dd"</span>]])</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a>E4_dd<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="st">"E4_dd"</span></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>E4_dd<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="st">"E4"</span></span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>E4_dd <span class="ot">&lt;-</span> <span class="fu">merge</span>(E4_dd, E4vsE3_rna, <span class="at">by.x=</span><span class="st">"gene"</span>, <span class="at">by.y=</span><span class="st">"entrezgene_id"</span>)</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(E4_dd) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"log2FoldChange"</span>, <span class="st">"log2FoldChange_E4"</span>, <span class="fu">names</span>(E4_dd))</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>E2_dd <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">gene=</span>E2_KO_rrho[[<span class="st">"genelist_dd"</span>]][[<span class="st">"gene_list_overlap_dd"</span>]])</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>E2_dd<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="st">"E2_dd"</span></span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>E2_dd<span class="sc">$</span>genotype <span class="ot">&lt;-</span> <span class="st">"E2"</span></span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>E2_dd <span class="ot">&lt;-</span> <span class="fu">merge</span>(E2_dd, E2vsE3_rna, <span class="at">by.x=</span><span class="st">"gene"</span>, <span class="at">by.y=</span><span class="st">"entrezgene_id"</span>)</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(E2_dd) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"log2FoldChange"</span>, <span class="st">"log2FoldChange_E2"</span>, <span class="fu">names</span>(E2_dd))</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a><span class="co"># get genes which are shared across the comparisons </span></span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>uu <span class="ot">&lt;-</span> <span class="fu">merge</span>(E2_uu, E4_uu, <span class="at">by=</span><span class="st">"gene"</span>)</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>dd <span class="ot">&lt;-</span> <span class="fu">merge</span>(E2_dd, E4_dd, <span class="at">by=</span><span class="st">"gene"</span>)</span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>KO_overlap <span class="ot">&lt;-</span> <span class="fu">rbind</span>(uu, dd)</span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-32"><a href="#cb12-32" aria-hidden="true" tabindex="-1"></a>KO_overlap<span class="sc">$</span>Significant <span class="ot">&lt;-</span> <span class="st">"Not significant"</span></span>
<span id="cb12-33"><a href="#cb12-33" aria-hidden="true" tabindex="-1"></a>KO_overlap<span class="sc">$</span>Significant[KO_overlap<span class="sc">$</span>fdr.x<span class="sc">&lt;</span><span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"FDR &lt; 0.05 in APOE2"</span></span>
<span id="cb12-34"><a href="#cb12-34" aria-hidden="true" tabindex="-1"></a>KO_overlap<span class="sc">$</span>Significant[KO_overlap<span class="sc">$</span>fdr.y<span class="sc">&lt;</span><span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"FDR &lt; 0.05 in APOE4"</span></span>
<span id="cb12-35"><a href="#cb12-35" aria-hidden="true" tabindex="-1"></a>KO_overlap<span class="sc">$</span>Significant[KO_overlap<span class="sc">$</span>fdr.x<span class="sc">&lt;</span><span class="fl">0.05</span> <span class="sc">&amp;</span> KO_overlap<span class="sc">$</span>fdr.y <span class="sc">&lt;</span> <span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"FDR &lt; 0.05 in both"</span></span>
<span id="cb12-36"><a href="#cb12-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-37"><a href="#cb12-37" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(KO_overlap, <span class="fu">aes</span>(log2FoldChange_E2, log2FoldChange_E4, <span class="at">colour=</span>Significant)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span> </span>
<span id="cb12-38"><a href="#cb12-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"logFC of genes with similar expression profiles across </span><span class="sc">\n</span><span class="st"> APOE2 vs APOE3 and APOE-KO vs APOE3"</span>) <span class="sc">+</span> </span>
<span id="cb12-39"><a href="#cb12-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"logFC of genes with similar expression profiles across </span><span class="sc">\n</span><span class="st"> APOE4 vs APOE3 and APOE-KO vs APOE3"</span>) <span class="sc">+</span> </span>
<span id="cb12-40"><a href="#cb12-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">6</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">6</span>) <span class="sc">+</span> </span>
<span id="cb12-41"><a href="#cb12-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#DCCA2CFF"</span>, <span class="st">"#92BBD9FF"</span>, <span class="st">"peru"</span>, <span class="st">"moccasin"</span>)) <span class="sc">+</span> </span>
<span id="cb12-42"><a href="#cb12-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">14</span>)) <span class="sc">+</span></span>
<span id="cb12-43"><a href="#cb12-43" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"darkgrey"</span>) <span class="sc">+</span> </span>
<span id="cb12-44"><a href="#cb12-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"darkgrey"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_2f-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="768"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 2f: MAGMA gene set analysis with AD GWAS ##</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>magma_res <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/MAGMA_DEGs_AD_GWAS_res.csv"</span>)</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>magma_res<span class="sc">$</span>VARIABLE <span class="ot">&lt;-</span> <span class="fu">factor</span>(magma_res<span class="sc">$</span>VARIABLE, <span class="at">levels=</span><span class="fu">c</span>(<span class="st">"Up in E2vsE3"</span>, <span class="st">"Down in E2vsE3"</span>, <span class="st">"Up in E4vsE3"</span>, <span class="st">"Down in E4vsE3"</span>, <span class="st">"Up in E4vsE2"</span>, <span class="st">"Down in E4vsE2"</span>, <span class="st">"Up in KOvsE3"</span>, <span class="st">"Down in KOvsE3"</span>))</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>magma_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(magma_res, <span class="fu">aes</span>(<span class="at">x=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">y=</span>VARIABLE, <span class="at">fill=</span>VARIABLE)) <span class="sc">+</span> </span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">colour=</span><span class="st">"black"</span>) <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">"-log10(FDR)"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"DEGs"</span>) <span class="sc">+</span> </span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">14</span>)) <span class="sc">+</span> </span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>, <span class="at">hjust=</span><span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.05</span>), <span class="at">linetype=</span><span class="st">'dashed'</span>, <span class="at">col =</span> <span class="st">'slategrey'</span>) <span class="sc">+</span> </span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"MAGMA gene set analysis using AD GWAS"</span>) <span class="sc">+</span> <span class="fu">facet_wrap</span>(<span class="sc">~</span>GWAS) <span class="sc">+</span> </span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">limits=</span>rev) <span class="sc">+</span> </span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#C3DAEAFF"</span>, <span class="st">"#92BBD9FF"</span>, <span class="st">"#ECE28BFF"</span>, <span class="st">"#DCCA2CFF"</span>, <span class="st">"#B1D5BBFF"</span>, <span class="st">"#6FB382FF"</span>, <span class="st">"#9DAFC3FF"</span>, <span class="st">"#4D6D93FF"</span>))</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(magma_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_2g-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 2g) MAGMA gene set analysis using the differentially expressed genes across the <em>APOE</em> groups with three independent AD GWAS.</strong></p>
<p><strong>Figure 3: Differentially expressed genes exhibit differential chromatin accessibility in their vicinity.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 3a: E2vsE3 DAR heatmap with RNA-seq genes ## </span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co"># normalise counts </span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>counts_norm <span class="ot">&lt;-</span> <span class="fu">normTransform</span>(atac_dds)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>counts_norm_assay <span class="ot">&lt;-</span> <span class="fu">assay</span>(counts_norm)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>counts_z <span class="ot">&lt;-</span> <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(counts_norm_assay)))</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>counts_z <span class="ot">&lt;-</span> counts_z[,<span class="fu">c</span>(<span class="st">"E2_1"</span>, <span class="st">"E2_2"</span>, <span class="st">"E2_3"</span>, <span class="st">"E2_4"</span>, <span class="st">"E2_5"</span>, <span class="st">"E3_1"</span>, <span class="st">"E3_2"</span>, <span class="st">"E3_3"</span>, <span class="st">"E3_4"</span>, <span class="st">"E3_5"</span>, <span class="st">"E4_1"</span>, <span class="st">"E4_2"</span>, <span class="st">"E4_3"</span>, <span class="st">"E4_5"</span>, <span class="st">"KO_3"</span>, <span class="st">"KO_5"</span>)]</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>indices <span class="ot">&lt;-</span> <span class="fu">match</span>(E2vsE3_atac_sig<span class="sc">$</span>Peak, <span class="fu">rownames</span>(counts_z))</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>E2vsE3_counts <span class="ot">&lt;-</span> counts_z[indices, ]</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a>indices <span class="ot">&lt;-</span> <span class="fu">match</span>(E4vsE3_atac_sig<span class="sc">$</span>Peak, <span class="fu">rownames</span>(counts_z))</span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a>E4vsE3_counts <span class="ot">&lt;-</span> counts_z[indices, ]</span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a>indices <span class="ot">&lt;-</span> <span class="fu">match</span>(KOvsE3_atac_sig<span class="sc">$</span>Peak, <span class="fu">rownames</span>(counts_z))</span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a>KOvsE3_counts <span class="ot">&lt;-</span> counts_z[indices, ]</span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a>indices <span class="ot">&lt;-</span> <span class="fu">match</span>(E4vsE2_atac_sig<span class="sc">$</span>Peak, <span class="fu">rownames</span>(counts_z))</span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a>E4vsE2_counts <span class="ot">&lt;-</span> counts_z[indices, ]</span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a>mat  <span class="ot">&lt;-</span> E2vsE3_counts</span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> E2vsE3_atac_genes<span class="sc">$</span>SYMBOL</span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a><span class="co"># overlapping genes between atac-seq and rna-seq </span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a>overlap_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mat)[E2vsE3_atac_genes<span class="sc">$</span>SYMBOL <span class="sc">%in%</span> E2vsE3_rna_genes<span class="sc">$</span>hgnc_symbol]</span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a><span class="co"># function to make selected rownames bold</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a>make_bold_names <span class="ot">&lt;-</span> <span class="cf">function</span>(mat, rc_fun, rc_names) {</span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a>  bold_names <span class="ot">&lt;-</span> <span class="fu">rc_fun</span>(mat)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a>  ids <span class="ot">&lt;-</span> rc_names <span class="sc">%&gt;%</span> <span class="fu">match</span>(bold_names)</span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a>  ids <span class="sc">%&gt;%</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a>    purrr<span class="sc">::</span><span class="fu">walk</span>(</span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a>      <span class="cf">function</span>(i)</span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a>        bold_names[i] <span class="ot">&lt;&lt;-</span></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>        <span class="fu">bquote</span>(<span class="fu">bold</span>(.(<span class="fu">rc_fun</span>(mat)[i]))) <span class="sc">%&gt;%</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>        <span class="fu">as.expression</span>()</span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>  bold_names</span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>mat_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(mat)</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>mat_df<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mat_df<span class="sc">$</span>E2_1<span class="sc">&gt;</span><span class="dv">0</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span>)</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a>mat_df<span class="sc">$</span>label_colour<span class="ot">=</span><span class="fu">ifelse</span>(mat_df<span class="sc">$</span>direction<span class="sc">==</span><span class="st">"Up"</span>,<span class="st">"#B2182B"</span>,<span class="st">"#4D4D4D"</span>)</span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a><span class="co"># group replicates </span></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>group_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"E2"</span>,<span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"E3"</span>,<span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"E4"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"KO"</span>,<span class="dv">2</span>)))</span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(group_col) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(mat)</span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>group_colour <span class="ot">=</span> <span class="fu">list</span>(<span class="at">Group =</span> <span class="fu">c</span>(<span class="at">E2 =</span> <span class="st">"#92BBD9FF"</span>, </span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>                              <span class="at">E3 =</span> <span class="st">"#ECE28BFF"</span>,</span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a>                              <span class="at">E4 =</span> <span class="st">"#6FB382FF"</span>, </span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>                              <span class="at">KO =</span> <span class="st">"#4D6D93FF"</span>))</span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-53"><a href="#cb14-53" aria-hidden="true" tabindex="-1"></a>E2vsE3_pheatmap <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(mat,  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">7</span>, <span class="at">name =</span> <span class="st">"RdGy"</span>)))(<span class="dv">100</span>), </span>
<span id="cb14-54"><a href="#cb14-54" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cluster_rows=</span><span class="cn">TRUE</span>, <span class="at">cluster_cols=</span><span class="cn">FALSE</span>, <span class="at">fontsize_col =</span> <span class="dv">16</span>, </span>
<span id="cb14-55"><a href="#cb14-55" aria-hidden="true" tabindex="-1"></a>                        <span class="at">fontsize_row=</span><span class="dv">12</span>, <span class="at">labels_row =</span> <span class="fu">make_bold_names</span>(mat,rownames,overlap_genes),</span>
<span id="cb14-56"><a href="#cb14-56" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cellwidth =</span> <span class="dv">15</span>, <span class="at">cellheight=</span><span class="dv">10</span>, <span class="at">annotation_col =</span> group_col,</span>
<span id="cb14-57"><a href="#cb14-57" aria-hidden="true" tabindex="-1"></a>                        <span class="at">annotation_colors =</span> group_colour, <span class="at">labels_col =</span> <span class="st">""</span>, <span class="at">main =</span> <span class="st">"E2 vs E3"</span>)</span>
<span id="cb14-58"><a href="#cb14-58" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-59"><a href="#cb14-59" aria-hidden="true" tabindex="-1"></a>cols<span class="ot">=</span>mat_df[E2vsE3_pheatmap<span class="sc">$</span>tree_row<span class="sc">$</span>order,]<span class="sc">$</span>label_colour</span>
<span id="cb14-60"><a href="#cb14-60" aria-hidden="true" tabindex="-1"></a>E2vsE3_pheatmap<span class="sc">$</span>gtable<span class="sc">$</span>grobs[[<span class="dv">5</span>]]<span class="sc">$</span>gp<span class="ot">=</span><span class="fu">gpar</span>(<span class="at">col=</span>cols, <span class="at">fontsize =</span> <span class="dv">12</span>)</span>
<span id="cb14-61"><a href="#cb14-61" aria-hidden="true" tabindex="-1"></a>E2vsE3_pheatmap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_3a-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 3a) Heatmap showing differential chromatin accessibility of significant peaks (FDR &lt; 0.05) when comparing <em>APOE2</em> vs <em>APOE3</em>-expressing microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 3b: E4vsE3 DAR heatmap with RNA-seq genes ##</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>mat  <span class="ot">&lt;-</span> E4vsE3_counts</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> E4vsE3_atac_genes<span class="sc">$</span>SYMBOL</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a><span class="co"># overlapping genes between atac-seq and rna-seq </span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>overlap_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mat)[E4vsE3_atac_genes<span class="sc">$</span>SYMBOL <span class="sc">%in%</span> E4vsE3_rna_genes<span class="sc">$</span>hgnc_symbol]</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a><span class="co"># function to make selected rownames bold</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>make_bold_names <span class="ot">&lt;-</span> <span class="cf">function</span>(mat, rc_fun, rc_names) {</span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>  bold_names <span class="ot">&lt;-</span> <span class="fu">rc_fun</span>(mat)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>  ids <span class="ot">&lt;-</span> rc_names <span class="sc">%&gt;%</span> <span class="fu">match</span>(bold_names)</span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>  ids <span class="sc">%&gt;%</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>    purrr<span class="sc">::</span><span class="fu">walk</span>(</span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>      <span class="cf">function</span>(i)</span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>        bold_names[i] <span class="ot">&lt;&lt;-</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>        <span class="fu">bquote</span>(<span class="fu">bold</span>(.(<span class="fu">rc_fun</span>(mat)[i]))) <span class="sc">%&gt;%</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>        <span class="fu">as.expression</span>()</span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>  bold_names</span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>mat_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(mat)</span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>mat_df<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mat_df<span class="sc">$</span>E4_1<span class="sc">&gt;</span><span class="dv">0</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span>)</span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>mat_df<span class="sc">$</span>label_colour<span class="ot">=</span><span class="fu">ifelse</span>(mat_df<span class="sc">$</span>direction<span class="sc">==</span><span class="st">"Up"</span>,<span class="st">"#B2182B"</span>,<span class="st">"#4D4D4D"</span>)</span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a><span class="co"># group replicates </span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>group_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"E2"</span>,<span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"E3"</span>,<span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"E4"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"KO"</span>,<span class="dv">2</span>)))</span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(group_col) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(mat)</span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>group_colour <span class="ot">=</span> <span class="fu">list</span>(<span class="at">Group =</span> <span class="fu">c</span>(<span class="at">E2 =</span> <span class="st">"#92BBD9FF"</span>, </span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>                              <span class="at">E3 =</span> <span class="st">"#ECE28BFF"</span>,</span>
<span id="cb15-32"><a href="#cb15-32" aria-hidden="true" tabindex="-1"></a>                              <span class="at">E4 =</span> <span class="st">"#6FB382FF"</span>, </span>
<span id="cb15-33"><a href="#cb15-33" aria-hidden="true" tabindex="-1"></a>                              <span class="at">KO =</span> <span class="st">"#4D6D93FF"</span>))</span>
<span id="cb15-34"><a href="#cb15-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-35"><a href="#cb15-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-36"><a href="#cb15-36" aria-hidden="true" tabindex="-1"></a>E4vsE3_pheatmap <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(mat,  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">7</span>, <span class="at">name =</span> <span class="st">"RdGy"</span>)))(<span class="dv">100</span>), </span>
<span id="cb15-37"><a href="#cb15-37" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cluster_rows=</span><span class="cn">TRUE</span>, <span class="at">cluster_cols=</span><span class="cn">FALSE</span>, <span class="at">fontsize_col =</span> <span class="dv">16</span>, </span>
<span id="cb15-38"><a href="#cb15-38" aria-hidden="true" tabindex="-1"></a>                        <span class="at">fontsize_row=</span><span class="dv">12</span>, <span class="at">labels_row =</span> <span class="fu">make_bold_names</span>(mat, rownames, overlap_genes),</span>
<span id="cb15-39"><a href="#cb15-39" aria-hidden="true" tabindex="-1"></a>                        <span class="at">cellwidth =</span> <span class="dv">15</span>, <span class="at">cellheight=</span><span class="dv">10</span>, <span class="at">annotation_col =</span> group_col,</span>
<span id="cb15-40"><a href="#cb15-40" aria-hidden="true" tabindex="-1"></a>                        <span class="at">annotation_colors =</span> group_colour, <span class="at">labels_col =</span> <span class="st">""</span>, <span class="at">main =</span> <span class="st">"E4 vs E3"</span>)</span>
<span id="cb15-41"><a href="#cb15-41" aria-hidden="true" tabindex="-1"></a>cols<span class="ot">=</span>mat_df[E4vsE3_pheatmap<span class="sc">$</span>tree_row<span class="sc">$</span>order,]<span class="sc">$</span>label_colour</span>
<span id="cb15-42"><a href="#cb15-42" aria-hidden="true" tabindex="-1"></a>E4vsE3_pheatmap<span class="sc">$</span>gtable<span class="sc">$</span>grobs[[<span class="dv">5</span>]]<span class="sc">$</span>gp<span class="ot">=</span><span class="fu">gpar</span>(<span class="at">col=</span>cols, <span class="at">fontsize =</span> <span class="dv">12</span>)</span>
<span id="cb15-43"><a href="#cb15-43" aria-hidden="true" tabindex="-1"></a>E4vsE3_pheatmap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_3b-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 3b) Heatmap showing differential chromatin accessibility of significant peaks (FDR &lt; 0.05) when comparing <em>APOE4</em> vs <em>APOE3</em>-expressing microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 3d: CHCHD2 expression ## </span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>chchd2 <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">"51142"</span>, <span class="at">intgroup=</span><span class="st">"group"</span>, <span class="at">returnData =</span> <span class="cn">TRUE</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E3"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E4"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>,<span class="st">"KO"</span>))</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(chchd2, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>count, <span class="at">fill=</span>group)) <span class="sc">+</span> </span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Counts"</span>) <span class="sc">+</span> </span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"CHCHD2"</span>) <span class="sc">+</span> </span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">20</span>,<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face=</span><span class="fu">c</span>(<span class="st">"italic"</span>))) <span class="sc">+</span> <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons, <span class="at">method=</span><span class="st">"t.test"</span>) <span class="sc">+</span></span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill=</span><span class="cn">NULL</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_3c-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 3c) Boxplot of expression profiles of CHCHD2 shows reduced expression in the APOE4 and the APOE-KO microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 3d: chromatin accessibility around CHCHD2 ##</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>start <span class="ot">&lt;-</span> E4vsE2_atac_annot<span class="sc">$</span>geneStart[E4vsE2_atac_annot<span class="sc">$</span>SYMBOL<span class="sc">==</span><span class="st">"CHCHD2"</span>][<span class="dv">1</span>] <span class="sc">-</span><span class="dv">3000</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>end <span class="ot">&lt;-</span> E4vsE2_atac_annot<span class="sc">$</span>geneEnd[E4vsE2_atac_annot<span class="sc">$</span>SYMBOL<span class="sc">==</span><span class="st">"CHCHD2"</span>][<span class="dv">1</span>] <span class="sc">+</span> <span class="dv">3000</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="co"># plot data tracks </span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>track_list <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">KO =</span> <span class="st">"data/KO-1.bw"</span>,</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>                <span class="at">E4 =</span> <span class="st">"data/E4-1.bw"</span>, </span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>                <span class="at">E3 =</span> <span class="st">"data/E3-1.bw"</span>, </span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>                <span class="at">E2 =</span> <span class="st">"data/E2-1.bw"</span>)</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>chchd2_region <span class="ot">&lt;-</span> <span class="fu">toGRanges</span>(<span class="fu">data.frame</span>(<span class="at">chr=</span><span class="st">"chr7"</span>, <span class="at">start=</span>start, <span class="at">end=</span>end))</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a><span class="co"># set plotting parameters</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>pp <span class="ot">&lt;-</span> <span class="fu">getDefaultPlotParams</span>(<span class="at">plot.type=</span><span class="dv">1</span>)</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>leftmargin <span class="ot">&lt;-</span> <span class="fl">0.15</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>topmargin <span class="ot">&lt;-</span> <span class="dv">15</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>bottommargin <span class="ot">&lt;-</span> <span class="dv">15</span></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>ideogramheight <span class="ot">&lt;-</span> <span class="dv">5</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>pp<span class="sc">$</span>data1inmargin <span class="ot">&lt;-</span> <span class="dv">10</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">plotKaryotype</span>(<span class="at">zoom =</span> chchd2_region, <span class="at">cex=</span><span class="fl">1.7</span>, <span class="at">plot.params =</span> pp)</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAddBaseNumbers</span>(kp, <span class="at">tick.dist =</span> <span class="dv">3000</span>, <span class="at">minor.tick.dist =</span> <span class="dv">1500</span>,</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>                 <span class="at">add.units =</span> <span class="cn">TRUE</span>, <span class="at">cex=</span><span class="fl">1.5</span>, <span class="at">digits =</span> <span class="dv">6</span>)</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a>chchd2_data <span class="ot">&lt;-</span> <span class="fu">makeGenesDataFromTxDb</span>(TxDb.Hsapiens.UCSC.hg38.knownGene,</span>
<span id="cb17-31"><a href="#cb17-31" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">karyoplot=</span>kp,</span>
<span id="cb17-32"><a href="#cb17-32" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">plot.transcripts =</span> <span class="cn">TRUE</span>, </span>
<span id="cb17-33"><a href="#cb17-33" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">plot.transcripts.structure =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-34"><a href="#cb17-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-35"><a href="#cb17-35" aria-hidden="true" tabindex="-1"></a>chchd2_data<span class="sc">$</span>genes<span class="sc">$</span>name <span class="ot">&lt;-</span> <span class="st">"CHCHD2"</span></span>
<span id="cb17-36"><a href="#cb17-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-37"><a href="#cb17-37" aria-hidden="true" tabindex="-1"></a><span class="co"># KO</span></span>
<span id="cb17-38"><a href="#cb17-38" aria-hidden="true" tabindex="-1"></a>bigwig.file <span class="ot">&lt;-</span> track_list[[<span class="st">"KO"</span>]]</span>
<span id="cb17-39"><a href="#cb17-39" aria-hidden="true" tabindex="-1"></a>track_position <span class="ot">=</span> <span class="dv">1</span></span>
<span id="cb17-40"><a href="#cb17-40" aria-hidden="true" tabindex="-1"></a>track_colour <span class="ot">=</span> <span class="st">"#4D6D93FF"</span></span>
<span id="cb17-41"><a href="#cb17-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-42"><a href="#cb17-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjusting the margin for tracks</span></span>
<span id="cb17-43"><a href="#cb17-43" aria-hidden="true" tabindex="-1"></a>track_margin <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb17-44"><a href="#cb17-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-45"><a href="#cb17-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Autotrack for genes</span></span>
<span id="cb17-46"><a href="#cb17-46" aria-hidden="true" tabindex="-1"></a>genes_track_at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="at">r0=</span><span class="dv">0</span>, <span class="at">r1=</span>track_margin)</span>
<span id="cb17-47"><a href="#cb17-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-48"><a href="#cb17-48" aria-hidden="true" tabindex="-1"></a><span class="co"># Autotrack for the current track</span></span>
<span id="cb17-49"><a href="#cb17-49" aria-hidden="true" tabindex="-1"></a>at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(track_position, <span class="fu">length</span>(track_list), <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r1 <span class="sc">+</span> track_margin, <span class="at">margin =</span> <span class="fl">0.2</span>)</span>
<span id="cb17-50"><a href="#cb17-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting genes</span></span>
<span id="cb17-51"><a href="#cb17-51" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">kpPlotGenes</span>(kp, <span class="at">data=</span>chchd2_data, <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r0, <span class="at">r1=</span>genes_track_at<span class="sc">$</span>r1, <span class="at">gene.name.cex=</span><span class="dv">2</span>)</span>
<span id="cb17-52"><a href="#cb17-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-53"><a href="#cb17-53" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the current track</span></span>
<span id="cb17-54"><a href="#cb17-54" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">kpPlotBigWig</span>(kp, <span class="at">data=</span>bigwig.file, <span class="at">ymax=</span><span class="st">"visible.region"</span>, </span>
<span id="cb17-55"><a href="#cb17-55" aria-hidden="true" tabindex="-1"></a>                   <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span> <span class="fl">0.20466</span>, <span class="at">col =</span> track_colour)</span>
<span id="cb17-56"><a href="#cb17-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-57"><a href="#cb17-57" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjusting the y-axis limits and adding labels</span></span>
<span id="cb17-58"><a href="#cb17-58" aria-hidden="true" tabindex="-1"></a>computed.ymax <span class="ot">&lt;-</span> <span class="dv">31</span></span>
<span id="cb17-59"><a href="#cb17-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-60"><a href="#cb17-60" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAxis</span>(kp, <span class="at">ymin=</span><span class="dv">0</span>, <span class="at">ymax=</span>computed.ymax, <span class="at">numticks =</span> <span class="dv">2</span>, <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, <span class="at">cex =</span> <span class="fl">1.5</span>)</span>
<span id="cb17-61"><a href="#cb17-61" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAddLabels</span>(kp, <span class="at">labels =</span> <span class="fu">names</span>(track_list)[<span class="dv">1</span>], <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, </span>
<span id="cb17-62"><a href="#cb17-62" aria-hidden="true" tabindex="-1"></a>            <span class="at">cex=</span><span class="dv">2</span>, <span class="at">label.margin =</span> <span class="fl">0.035</span>, <span class="at">col =</span> track_colour)</span>
<span id="cb17-63"><a href="#cb17-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-64"><a href="#cb17-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-65"><a href="#cb17-65" aria-hidden="true" tabindex="-1"></a><span class="co"># E4</span></span>
<span id="cb17-66"><a href="#cb17-66" aria-hidden="true" tabindex="-1"></a>bigwig.file <span class="ot">&lt;-</span> track_list[[<span class="st">"E4"</span>]]</span>
<span id="cb17-67"><a href="#cb17-67" aria-hidden="true" tabindex="-1"></a>track_position <span class="ot">=</span> <span class="dv">2</span></span>
<span id="cb17-68"><a href="#cb17-68" aria-hidden="true" tabindex="-1"></a>track_colour <span class="ot">=</span> <span class="st">"#6FB382FF"</span></span>
<span id="cb17-69"><a href="#cb17-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-70"><a href="#cb17-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjusting the margin for tracks</span></span>
<span id="cb17-71"><a href="#cb17-71" aria-hidden="true" tabindex="-1"></a>track_margin <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb17-72"><a href="#cb17-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-73"><a href="#cb17-73" aria-hidden="true" tabindex="-1"></a><span class="co"># Autotrack for genes</span></span>
<span id="cb17-74"><a href="#cb17-74" aria-hidden="true" tabindex="-1"></a>genes_track_at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="at">r0=</span><span class="dv">0</span>, <span class="at">r1=</span>track_margin)</span>
<span id="cb17-75"><a href="#cb17-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-76"><a href="#cb17-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Autotrack for the current track</span></span>
<span id="cb17-77"><a href="#cb17-77" aria-hidden="true" tabindex="-1"></a>at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(track_position, <span class="fu">length</span>(track_list), <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r1 <span class="sc">+</span> track_margin, <span class="at">margin =</span> <span class="fl">0.2</span>)</span>
<span id="cb17-78"><a href="#cb17-78" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting genes</span></span>
<span id="cb17-79"><a href="#cb17-79" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">kpPlotGenes</span>(kp, <span class="at">data=</span>chchd2_data, <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r0, <span class="at">r1=</span>genes_track_at<span class="sc">$</span>r1, <span class="at">gene.name.cex=</span><span class="dv">2</span>)</span>
<span id="cb17-80"><a href="#cb17-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-81"><a href="#cb17-81" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the current track</span></span>
<span id="cb17-82"><a href="#cb17-82" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">kpPlotBigWig</span>(kp, <span class="at">data=</span>bigwig.file, <span class="at">ymax=</span><span class="st">"visible.region"</span>, </span>
<span id="cb17-83"><a href="#cb17-83" aria-hidden="true" tabindex="-1"></a>                   <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span> <span class="fl">0.41718</span>, <span class="at">col =</span> track_colour)</span>
<span id="cb17-84"><a href="#cb17-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-85"><a href="#cb17-85" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjusting the y-axis limits and adding labels</span></span>
<span id="cb17-86"><a href="#cb17-86" aria-hidden="true" tabindex="-1"></a>computed.ymax <span class="ot">&lt;-</span> <span class="dv">31</span></span>
<span id="cb17-87"><a href="#cb17-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-88"><a href="#cb17-88" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAxis</span>(kp, <span class="at">ymin=</span><span class="dv">0</span>, <span class="at">ymax=</span>computed.ymax, <span class="at">numticks =</span> <span class="dv">2</span>, <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, <span class="at">cex =</span> <span class="fl">1.5</span>)</span>
<span id="cb17-89"><a href="#cb17-89" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAddLabels</span>(kp, <span class="at">labels =</span> <span class="fu">names</span>(track_list)[<span class="dv">2</span>], <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, </span>
<span id="cb17-90"><a href="#cb17-90" aria-hidden="true" tabindex="-1"></a>            <span class="at">cex=</span><span class="dv">2</span>, <span class="at">label.margin =</span> <span class="fl">0.035</span>, <span class="at">col =</span> track_colour)</span>
<span id="cb17-91"><a href="#cb17-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-92"><a href="#cb17-92" aria-hidden="true" tabindex="-1"></a><span class="co"># E3</span></span>
<span id="cb17-93"><a href="#cb17-93" aria-hidden="true" tabindex="-1"></a>bigwig.file <span class="ot">&lt;-</span> track_list[[<span class="st">"E3"</span>]]</span>
<span id="cb17-94"><a href="#cb17-94" aria-hidden="true" tabindex="-1"></a>track_position <span class="ot">=</span> <span class="dv">3</span></span>
<span id="cb17-95"><a href="#cb17-95" aria-hidden="true" tabindex="-1"></a>track_colour <span class="ot">=</span> <span class="st">"#DCCA2CFF"</span></span>
<span id="cb17-96"><a href="#cb17-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-97"><a href="#cb17-97" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjusting the margin for tracks</span></span>
<span id="cb17-98"><a href="#cb17-98" aria-hidden="true" tabindex="-1"></a>track_margin <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb17-99"><a href="#cb17-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-100"><a href="#cb17-100" aria-hidden="true" tabindex="-1"></a><span class="co"># Autotrack for genes</span></span>
<span id="cb17-101"><a href="#cb17-101" aria-hidden="true" tabindex="-1"></a>genes_track_at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="at">r0=</span><span class="dv">0</span>, <span class="at">r1=</span>track_margin)</span>
<span id="cb17-102"><a href="#cb17-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-103"><a href="#cb17-103" aria-hidden="true" tabindex="-1"></a><span class="co"># Autotrack for the current track</span></span>
<span id="cb17-104"><a href="#cb17-104" aria-hidden="true" tabindex="-1"></a>at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(track_position, <span class="fu">length</span>(track_list), <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r1 <span class="sc">+</span> track_margin, <span class="at">margin =</span> <span class="fl">0.2</span>)</span>
<span id="cb17-105"><a href="#cb17-105" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting genes</span></span>
<span id="cb17-106"><a href="#cb17-106" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">kpPlotGenes</span>(kp, <span class="at">data=</span>chchd2_data, <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r0, <span class="at">r1=</span>genes_track_at<span class="sc">$</span>r1, <span class="at">gene.name.cex=</span><span class="dv">2</span>)</span>
<span id="cb17-107"><a href="#cb17-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-108"><a href="#cb17-108" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the current track</span></span>
<span id="cb17-109"><a href="#cb17-109" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">kpPlotBigWig</span>(kp, <span class="at">data=</span>bigwig.file, <span class="at">ymax=</span><span class="st">"visible.region"</span>, </span>
<span id="cb17-110"><a href="#cb17-110" aria-hidden="true" tabindex="-1"></a>                   <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span> at<span class="sc">$</span>r1, <span class="at">col =</span> track_colour)</span>
<span id="cb17-111"><a href="#cb17-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-112"><a href="#cb17-112" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjusting the y-axis limits and adding labels</span></span>
<span id="cb17-113"><a href="#cb17-113" aria-hidden="true" tabindex="-1"></a>computed.ymax <span class="ot">&lt;-</span> <span class="dv">31</span></span>
<span id="cb17-114"><a href="#cb17-114" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-115"><a href="#cb17-115" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAxis</span>(kp, <span class="at">ymin=</span><span class="dv">0</span>, <span class="at">ymax=</span>computed.ymax, <span class="at">numticks =</span> <span class="dv">2</span>, <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, <span class="at">cex =</span> <span class="fl">1.5</span>)</span>
<span id="cb17-116"><a href="#cb17-116" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAddLabels</span>(kp, <span class="at">labels =</span> <span class="fu">names</span>(track_list)[<span class="dv">3</span>], <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, </span>
<span id="cb17-117"><a href="#cb17-117" aria-hidden="true" tabindex="-1"></a>            <span class="at">cex=</span><span class="dv">2</span>, <span class="at">label.margin =</span> <span class="fl">0.035</span>, <span class="at">col =</span> track_colour)</span>
<span id="cb17-118"><a href="#cb17-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-119"><a href="#cb17-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-120"><a href="#cb17-120" aria-hidden="true" tabindex="-1"></a><span class="co"># E2</span></span>
<span id="cb17-121"><a href="#cb17-121" aria-hidden="true" tabindex="-1"></a>bigwig.file <span class="ot">&lt;-</span> track_list[[<span class="st">"E2"</span>]]</span>
<span id="cb17-122"><a href="#cb17-122" aria-hidden="true" tabindex="-1"></a>track_position <span class="ot">=</span> <span class="dv">4</span></span>
<span id="cb17-123"><a href="#cb17-123" aria-hidden="true" tabindex="-1"></a>track_colour <span class="ot">=</span> <span class="st">"#92BBD9FF"</span></span>
<span id="cb17-124"><a href="#cb17-124" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-125"><a href="#cb17-125" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjusting the margin for tracks</span></span>
<span id="cb17-126"><a href="#cb17-126" aria-hidden="true" tabindex="-1"></a>track_margin <span class="ot">&lt;-</span> <span class="fl">0.1</span></span>
<span id="cb17-127"><a href="#cb17-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-128"><a href="#cb17-128" aria-hidden="true" tabindex="-1"></a><span class="co"># Autotrack for genes</span></span>
<span id="cb17-129"><a href="#cb17-129" aria-hidden="true" tabindex="-1"></a>genes_track_at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(<span class="dv">1</span>, <span class="dv">1</span>, <span class="at">r0=</span><span class="dv">0</span>, <span class="at">r1=</span>track_margin)</span>
<span id="cb17-130"><a href="#cb17-130" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-131"><a href="#cb17-131" aria-hidden="true" tabindex="-1"></a><span class="co"># Autotrack for the current track</span></span>
<span id="cb17-132"><a href="#cb17-132" aria-hidden="true" tabindex="-1"></a>at <span class="ot">&lt;-</span> <span class="fu">autotrack</span>(track_position, <span class="fu">length</span>(track_list), <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r1 <span class="sc">+</span> track_margin, <span class="at">margin =</span> <span class="fl">0.2</span>)</span>
<span id="cb17-133"><a href="#cb17-133" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting genes</span></span>
<span id="cb17-134"><a href="#cb17-134" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">kpPlotGenes</span>(kp, <span class="at">data=</span>chchd2_data, <span class="at">r0=</span>genes_track_at<span class="sc">$</span>r0, <span class="at">r1=</span>genes_track_at<span class="sc">$</span>r1, <span class="at">gene.name.cex=</span><span class="dv">2</span>)</span>
<span id="cb17-135"><a href="#cb17-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-136"><a href="#cb17-136" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting the current track</span></span>
<span id="cb17-137"><a href="#cb17-137" aria-hidden="true" tabindex="-1"></a>kp <span class="ot">&lt;-</span> <span class="fu">kpPlotBigWig</span>(kp, <span class="at">data=</span>bigwig.file, <span class="at">ymax=</span><span class="st">"visible.region"</span>, </span>
<span id="cb17-138"><a href="#cb17-138" aria-hidden="true" tabindex="-1"></a>                   <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span> <span class="fl">0.91306</span>, <span class="at">col =</span> track_colour)</span>
<span id="cb17-139"><a href="#cb17-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-140"><a href="#cb17-140" aria-hidden="true" tabindex="-1"></a><span class="co"># Adjusting the y-axis limits and adding labels</span></span>
<span id="cb17-141"><a href="#cb17-141" aria-hidden="true" tabindex="-1"></a>computed.ymax <span class="ot">&lt;-</span> <span class="dv">31</span></span>
<span id="cb17-142"><a href="#cb17-142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-143"><a href="#cb17-143" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAxis</span>(kp, <span class="at">ymin=</span><span class="dv">0</span>, <span class="at">ymax=</span>computed.ymax, <span class="at">numticks =</span> <span class="dv">2</span>, <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, <span class="at">cex =</span> <span class="fl">1.5</span>)</span>
<span id="cb17-144"><a href="#cb17-144" aria-hidden="true" tabindex="-1"></a><span class="fu">kpAddLabels</span>(kp, <span class="at">labels =</span> <span class="fu">names</span>(track_list)[<span class="dv">4</span>], <span class="at">r0=</span>at<span class="sc">$</span>r0, <span class="at">r1=</span>at<span class="sc">$</span>r1, </span>
<span id="cb17-145"><a href="#cb17-145" aria-hidden="true" tabindex="-1"></a>            <span class="at">cex=</span><span class="dv">2</span>, <span class="at">label.margin =</span> <span class="fl">0.035</span>, <span class="at">col =</span> track_colour)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_3d-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="624"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 3d) Genome tracks of chromatin accessibility signals around the <em>CHCHD2</em> locus show a loss of the open chromatin peak at the <em>CHCHD2</em> promoter in <em>APOE4</em> and <em>APOE</em>-KO microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 3c: s-LDSC with AD, ALS, and ASD GWAS ##</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a><span class="co"># load in sLDSC res </span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>sLDSC_res <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/sLDSC_ATAC_peaks_res.csv"</span>)</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>enrichment_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sLDSC_res, <span class="fu">aes</span>(title, Enrichment, <span class="at">color=</span>group)) <span class="sc">+</span> </span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size=</span><span class="dv">4</span>) <span class="sc">+</span> </span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#C93312"</span>, <span class="st">"#899DA4"</span>, <span class="st">"#FAEFD1"</span>)) <span class="sc">+</span> </span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x=</span> <span class="fu">element_blank</span>(),</span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position=</span><span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb18-14"><a href="#cb18-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Enrichment"</span>) <span class="sc">+</span> </span>
<span id="cb18-15"><a href="#cb18-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_pointrange</span>(<span class="fu">aes</span>(<span class="at">ymin=</span>Enrichment<span class="sc">-</span>Enrichment_std_error, </span>
<span id="cb18-16"><a href="#cb18-16" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ymax=</span>Enrichment<span class="sc">+</span>Enrichment_std_error)) <span class="sc">+</span> </span>
<span id="cb18-17"><a href="#cb18-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">1</span>, <span class="at">linetype=</span><span class="st">"dotted"</span>, <span class="at">color=</span><span class="st">"slategrey"</span>) <span class="sc">+</span> </span>
<span id="cb18-18"><a href="#cb18-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>)) <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">color =</span> <span class="st">""</span>) <span class="sc">+</span> <span class="fu">facet_grid</span>(. <span class="sc">~</span> group)</span>
<span id="cb18-19"><a href="#cb18-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-20"><a href="#cb18-20" aria-hidden="true" tabindex="-1"></a>coefficient_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sLDSC_res, <span class="fu">aes</span>(title, <span class="sc">-</span><span class="fu">log10</span>(Coefficient_p_fdr), <span class="at">fill=</span>group)) <span class="sc">+</span> </span>
<span id="cb18-21"><a href="#cb18-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat=</span><span class="st">"identity"</span>) <span class="sc">+</span> </span>
<span id="cb18-22"><a href="#cb18-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb18-23"><a href="#cb18-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#C93312"</span>, <span class="st">"#899DA4"</span>, <span class="st">"#FAEFD1"</span>)) <span class="sc">+</span> </span>
<span id="cb18-24"><a href="#cb18-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb18-25"><a href="#cb18-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb18-26"><a href="#cb18-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">strip.text.x =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb18-27"><a href="#cb18-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position=</span><span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb18-28"><a href="#cb18-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"-log10(Coefficient p)"</span>)  <span class="sc">+</span> </span>
<span id="cb18-29"><a href="#cb18-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="fl">1.3</span>, <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"slategrey"</span>) <span class="sc">+</span></span>
<span id="cb18-30"><a href="#cb18-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>)) <span class="sc">+</span> <span class="fu">facet_grid</span>(. <span class="sc">~</span> group) <span class="sc">+</span> </span>
<span id="cb18-31"><a href="#cb18-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Open chromatin regions in </span><span class="sc">\n</span><span class="st"> xenograted human microglia"</span>)</span>
<span id="cb18-32"><a href="#cb18-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb18-33"><a href="#cb18-33" aria-hidden="true" tabindex="-1"></a><span class="fu">ggarrange</span>(enrichment_plot, coefficient_plot, <span class="at">ncol=</span><span class="dv">1</span>, <span class="at">nrow=</span><span class="dv">2</span>, <span class="at">common.legend=</span><span class="cn">FALSE</span>, <span class="at">align=</span><span class="st">"v"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_3e-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 3e) s-LDSC analysis using all open chromatin regions from the xenotransplanted human microglia with GWAS summary statistics for AD, ALS, and ASD shows a microglia-specific enrichment for AD.</strong></p>
<p><strong>Figure 4: Pro-inflammatory cytokines are upregulated in <em>APOE4</em>-expressing microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 4a: Microglia cluster heatmap ## </span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>sheet_names <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"HM"</span>, <span class="st">"tCRM"</span>, <span class="st">"CRM-1"</span>, <span class="st">"CRM-2"</span>, <span class="st">"RM"</span>, <span class="st">"DAM"</span>, <span class="st">"HLA"</span>, <span class="st">"IRM"</span>)</span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>microglial_states <span class="ot">&lt;-</span> <span class="fu">lapply</span>(sheet_names, <span class="cf">function</span>(sheet_name) {</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_excel</span>(<span class="st">"data/scRNAseq_microglia_states_Mancuso2024.xlsx"</span>, <span class="at">sheet =</span> sheet_name)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="co"># get top 100 upregulated genes in each microglia cluster</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>microglial_states_top100 <span class="ot">&lt;-</span> <span class="fu">lapply</span>(microglial_states, <span class="cf">function</span>(df) {</span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(avg_log2FC <span class="sc">&gt;</span> <span class="dv">0</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(p_val_adj) <span class="sc">%&gt;%</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">head</span>(<span class="dv">100</span>)</span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co"># function to run hypergeometric test using DEGs and top 100 genes from microglia scRNA-seq clusters</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a>run_hypergeo_test <span class="ot">&lt;-</span> <span class="cf">function</span>(df, enrichment_data_frames, comparison) {</span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a>  results_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter by fdr &lt; 0.05</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a>  filtered_df <span class="ot">&lt;-</span> df[df<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a>  filtered_df <span class="ot">&lt;-</span> filtered_df[<span class="fu">complete.cases</span>(filtered_df<span class="sc">$</span>baseMean), ]</span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset rows with log2FoldChange &gt; 0</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a>  upregulated <span class="ot">&lt;-</span> filtered_df[filtered_df<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset rows with log2FoldChange &lt; 0</span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a>  downregulated <span class="ot">&lt;-</span> filtered_df[filtered_df<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>,]</span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Now, check for enrichment in each data frame in the list</span></span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (enrichment_df <span class="cf">in</span> enrichment_data_frames) {</span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a>    total_genes <span class="ot">&lt;-</span> <span class="fu">nrow</span>(df)</span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a>    upregulated_ol <span class="ot">&lt;-</span> <span class="fu">intersect</span>(upregulated<span class="sc">$</span>hgnc_symbol, enrichment_df<span class="sc">$</span>gene)</span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>    GeneRatio_upregulated <span class="ot">&lt;-</span> <span class="fu">length</span>(upregulated_ol)<span class="sc">/</span><span class="fu">nrow</span>(enrichment_df)</span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>    Count_upregulated <span class="ot">&lt;-</span> <span class="fu">length</span>(upregulated_ol)</span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>    downregulated_ol <span class="ot">&lt;-</span> <span class="fu">intersect</span>(downregulated<span class="sc">$</span>hgnc_symbol, enrichment_df<span class="sc">$</span>gene)</span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>    GeneRatio_downregulated <span class="ot">&lt;-</span> <span class="fu">length</span>(downregulated_ol)<span class="sc">/</span><span class="fu">nrow</span>(enrichment_df)</span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>    Count_downregulated <span class="ot">&lt;-</span> <span class="fu">length</span>(downregulated_ol)</span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if the overlap is greater than 0 for upregulated genes</span></span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(upregulated_ol) <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>      p_value_upregulated <span class="ot">&lt;-</span> <span class="fu">phyper</span>((<span class="fu">length</span>(upregulated_ol) <span class="sc">-</span> <span class="dv">1</span>), <span class="fu">nrow</span>(enrichment_df), total_genes <span class="sc">-</span> <span class="fu">nrow</span>(enrichment_df), <span class="fu">nrow</span>(upregulated), <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a>      p_value_upregulated <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if the overlap is greater than 0 for downregulated genes</span></span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">length</span>(downregulated_ol) <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a>      p_value_downregulated <span class="ot">&lt;-</span> <span class="fu">phyper</span>((<span class="fu">length</span>(downregulated_ol) <span class="sc">-</span> <span class="dv">1</span>), <span class="fu">nrow</span>(enrichment_df), total_genes <span class="sc">-</span> <span class="fu">nrow</span>(enrichment_df), <span class="fu">nrow</span>(downregulated), <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb19-52"><a href="#cb19-52" aria-hidden="true" tabindex="-1"></a>      p_value_downregulated <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb19-53"><a href="#cb19-53" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb19-54"><a href="#cb19-54" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-55"><a href="#cb19-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create a data frame for the results and append it to the main results data frame</span></span>
<span id="cb19-56"><a href="#cb19-56" aria-hidden="true" tabindex="-1"></a>    result_upregulated <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Comparison =</span> comparison, <span class="at">Direction =</span> <span class="st">"Upregulated"</span>, </span>
<span id="cb19-57"><a href="#cb19-57" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">p =</span> p_value_upregulated, <span class="at">Cluster =</span> <span class="fu">unique</span>(enrichment_df<span class="sc">$</span>cluster),</span>
<span id="cb19-58"><a href="#cb19-58" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">GeneRatio =</span> GeneRatio_upregulated, <span class="at">Count =</span> Count_upregulated)</span>
<span id="cb19-59"><a href="#cb19-59" aria-hidden="true" tabindex="-1"></a>    result_downregulated <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Comparison =</span> comparison, <span class="at">Direction =</span> <span class="st">"Downregulated"</span>, </span>
<span id="cb19-60"><a href="#cb19-60" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">p =</span> p_value_downregulated, <span class="at">Cluster =</span> <span class="fu">unique</span>(enrichment_df<span class="sc">$</span>cluster),</span>
<span id="cb19-61"><a href="#cb19-61" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">GeneRatio =</span> GeneRatio_downregulated, <span class="at">Count =</span> Count_downregulated)</span>
<span id="cb19-62"><a href="#cb19-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb19-63"><a href="#cb19-63" aria-hidden="true" tabindex="-1"></a>    results_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results_df, result_upregulated, result_downregulated)</span>
<span id="cb19-64"><a href="#cb19-64" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb19-65"><a href="#cb19-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb19-66"><a href="#cb19-66" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return the results if needed</span></span>
<span id="cb19-67"><a href="#cb19-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(results_df)</span>
<span id="cb19-68"><a href="#cb19-68" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb19-69"><a href="#cb19-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-70"><a href="#cb19-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply the function to each contrast</span></span>
<span id="cb19-71"><a href="#cb19-71" aria-hidden="true" tabindex="-1"></a>E2vsE3_mglcluster_ol <span class="ot">&lt;-</span> <span class="fu">run_hypergeo_test</span>(E2vsE3_rna, microglial_states_top100, <span class="st">"E2vsE3"</span>)</span>
<span id="cb19-72"><a href="#cb19-72" aria-hidden="true" tabindex="-1"></a>E4vsE3_mglcluster_ol <span class="ot">&lt;-</span> <span class="fu">run_hypergeo_test</span>(E4vsE3_rna, microglial_states_top100, <span class="st">"E4vsE3"</span>)</span>
<span id="cb19-73"><a href="#cb19-73" aria-hidden="true" tabindex="-1"></a>E4vsE2_mglcluster_ol <span class="ot">&lt;-</span> <span class="fu">run_hypergeo_test</span>(E4vsE2_rna, microglial_states_top100, <span class="st">"E4vsE2"</span>)</span>
<span id="cb19-74"><a href="#cb19-74" aria-hidden="true" tabindex="-1"></a>KOvsE3_mglcluster_ol <span class="ot">&lt;-</span> <span class="fu">run_hypergeo_test</span>(KOvsE3_rna, microglial_states_top100, <span class="st">"KOvsE3"</span>)</span>
<span id="cb19-75"><a href="#cb19-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-76"><a href="#cb19-76" aria-hidden="true" tabindex="-1"></a><span class="co"># get all results and perform multiple testing correction</span></span>
<span id="cb19-77"><a href="#cb19-77" aria-hidden="true" tabindex="-1"></a>all_mglcluster_ol <span class="ot">&lt;-</span> <span class="fu">rbind</span>(E2vsE3_mglcluster_ol, E4vsE3_mglcluster_ol, E4vsE2_mglcluster_ol, KOvsE3_mglcluster_ol)</span>
<span id="cb19-78"><a href="#cb19-78" aria-hidden="true" tabindex="-1"></a>all_mglcluster_ol<span class="sc">$</span>bonf <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(all_mglcluster_ol<span class="sc">$</span>p, <span class="at">method=</span><span class="st">"bonferroni"</span>)</span>
<span id="cb19-79"><a href="#cb19-79" aria-hidden="true" tabindex="-1"></a>all_mglcluster_ol<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">paste</span>(all_mglcluster_ol<span class="sc">$</span>Direction, <span class="st">"in"</span>, all_mglcluster_ol<span class="sc">$</span>Comparison)</span>
<span id="cb19-80"><a href="#cb19-80" aria-hidden="true" tabindex="-1"></a>all_mglcluster_ol<span class="sc">$</span>p_scientific <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(<span class="fu">format</span>(all_mglcluster_ol<span class="sc">$</span>bonf, <span class="at">scientific =</span> <span class="cn">TRUE</span>))</span>
<span id="cb19-81"><a href="#cb19-81" aria-hidden="true" tabindex="-1"></a>all_mglcluster_ol<span class="sc">$</span>signif[all_mglcluster_ol<span class="sc">$</span>bonf <span class="sc">&lt;</span> <span class="fl">0.05</span>] <span class="ot">&lt;-</span> <span class="st">"*"</span></span>
<span id="cb19-82"><a href="#cb19-82" aria-hidden="true" tabindex="-1"></a>all_mglcluster_ol<span class="sc">$</span>signif[all_mglcluster_ol<span class="sc">$</span>bonf <span class="sc">&lt;</span> <span class="fl">0.01</span>] <span class="ot">&lt;-</span> <span class="st">"**"</span></span>
<span id="cb19-83"><a href="#cb19-83" aria-hidden="true" tabindex="-1"></a>all_mglcluster_ol<span class="sc">$</span>signif[all_mglcluster_ol<span class="sc">$</span>bonf <span class="sc">&lt;</span> <span class="fl">0.001</span>] <span class="ot">&lt;-</span> <span class="st">"***"</span></span>
<span id="cb19-84"><a href="#cb19-84" aria-hidden="true" tabindex="-1"></a>all_mglcluster_ol<span class="sc">$</span>signif[all_mglcluster_ol<span class="sc">$</span>bonf <span class="sc">&lt;</span> <span class="fl">0.0001</span>] <span class="ot">&lt;-</span> <span class="st">"****"</span></span>
<span id="cb19-85"><a href="#cb19-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-86"><a href="#cb19-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-87"><a href="#cb19-87" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(all_mglcluster_ol, <span class="fu">aes</span>(<span class="at">x =</span> Comparison, <span class="at">y =</span> Cluster)) <span class="sc">+</span></span>
<span id="cb19-88"><a href="#cb19-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> GeneRatio), <span class="at">color =</span> <span class="st">"black"</span>) <span class="sc">+</span></span>
<span id="cb19-89"><a href="#cb19-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">ifelse</span>(bonf <span class="sc">&lt;</span> <span class="fl">0.05</span>, <span class="fu">sprintf</span>(<span class="st">"%.1e"</span>, p_scientific), <span class="st">""</span>)), <span class="at">vjust =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb19-90"><a href="#cb19-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"#B2182B"</span>) <span class="sc">+</span></span>
<span id="cb19-91"><a href="#cb19-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb19-92"><a href="#cb19-92" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>Direction) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Microglia cluster"</span>) <span class="sc">+</span> </span>
<span id="cb19-93"><a href="#cb19-93" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">16</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_4a-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="960"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 4a) Heatmap showing enrichment of genes differentially expressed across the <em>APOE</em> groups amongst microglia clusters defined by scRNA-seq.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 4b: GO enrichment using E4 genes overlapping DAM ## </span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co"># get dam microglia cluster </span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>dam <span class="ot">&lt;-</span> microglial_states_top100[[<span class="dv">6</span>]]</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>upregulated_ol <span class="ot">&lt;-</span> <span class="fu">intersect</span>(E4vsE2_rna_sig<span class="sc">$</span>hgnc_symbol[E4vsE2_rna_sig<span class="sc">$</span>log2FoldChange<span class="sc">&gt;</span><span class="dv">0</span>], dam<span class="sc">$</span>gene)</span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>downregulated_ol <span class="ot">&lt;-</span> <span class="fu">intersect</span>(E4vsE2_rna_sig<span class="sc">$</span>hgnc_symbol[E4vsE2_rna_sig<span class="sc">$</span>log2FoldChange<span class="sc">&lt;</span><span class="dv">0</span>], dam<span class="sc">$</span>gene)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a>E4vsE2_down_dam_bp <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">enrichGO</span>(downregulated_ol,</span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">OrgDb =</span> org.Hs.eg.db, </span>
<span id="cb20-13"><a href="#cb20-13" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb20-14"><a href="#cb20-14" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">ont =</span> <span class="st">"BP"</span>, </span>
<span id="cb20-15"><a href="#cb20-15" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>, </span>
<span id="cb20-16"><a href="#cb20-16" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb20-17"><a href="#cb20-17" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">universe =</span> <span class="fu">unique</span>(E4vsE2_rna<span class="sc">$</span>hgnc_symbol),</span>
<span id="cb20-18"><a href="#cb20-18" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb20-19"><a href="#cb20-19" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">minGSSize =</span> <span class="dv">10</span>,</span>
<span id="cb20-20"><a href="#cb20-20" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">maxGSSize =</span> <span class="dv">650</span>, </span>
<span id="cb20-21"><a href="#cb20-21" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">readable =</span> <span class="cn">FALSE</span>, </span>
<span id="cb20-22"><a href="#cb20-22" aria-hidden="true" tabindex="-1"></a>                                           <span class="at">pool =</span> <span class="cn">FALSE</span>))</span>
<span id="cb20-23"><a href="#cb20-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-24"><a href="#cb20-24" aria-hidden="true" tabindex="-1"></a>E4vsE2_down_dam_bp<span class="sc">$</span>GeneRatio_calc <span class="ot">&lt;-</span> E4vsE2_down_dam_bp<span class="sc">$</span>Count<span class="sc">/</span><span class="fu">length</span>(downregulated_ol)</span>
<span id="cb20-25"><a href="#cb20-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-26"><a href="#cb20-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Dotplot of GO BP processes enriched for genes down in E4 and up in DAM</span></span>
<span id="cb20-27"><a href="#cb20-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(E4vsE2_down_dam_bp[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,], </span>
<span id="cb20-28"><a href="#cb20-28" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>GeneRatio_calc, <span class="at">y=</span><span class="fu">reorder</span>(Description, GeneRatio_calc), </span>
<span id="cb20-29"><a href="#cb20-29" aria-hidden="true" tabindex="-1"></a>           <span class="at">colour =</span> p.adjust, <span class="at">size =</span> Count)) <span class="sc">+</span> </span>
<span id="cb20-30"><a href="#cb20-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb20-31"><a href="#cb20-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Gene ratio"</span>) <span class="sc">+</span> </span>
<span id="cb20-32"><a href="#cb20-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Biological process"</span>) <span class="sc">+</span> </span>
<span id="cb20-33"><a href="#cb20-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb20-34"><a href="#cb20-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Genes down in E4 enriched in DAM"</span>) <span class="sc">+</span> </span>
<span id="cb20-35"><a href="#cb20-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_gradient</span>(<span class="at">low =</span> <span class="st">"#B1D5BBFF"</span>, <span class="at">high =</span> <span class="st">"#6FB382FF"</span>) <span class="sc">+</span> </span>
<span id="cb20-36"><a href="#cb20-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">18</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">18</span>, <span class="at">hjust=</span><span class="fl">0.5</span>)) <span class="sc">+</span> </span>
<span id="cb20-37"><a href="#cb20-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">"Gene count"</span>), <span class="at">colour =</span> <span class="fu">guide_legend</span>(<span class="at">title =</span> <span class="st">"FDR"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_4b-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="1152"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 4b) Dotplot of pathway enrichment analysis using genes downregulated in <em>APOE4</em> microglia that are enriched in the DAM cluster.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 4c: CCL3 expression ##</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>ccl3 <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">"6348"</span>, <span class="at">intgroup=</span><span class="st">"group"</span>, <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E3"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E4"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>,<span class="st">"KO"</span>))</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Boxplot of CCL3 expression</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ccl3, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>count, <span class="at">fill=</span>group)) <span class="sc">+</span> </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Count"</span>) <span class="sc">+</span> </span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"CCL3"</span>) <span class="sc">+</span> </span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face=</span><span class="st">"italic"</span>)) <span class="sc">+</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons, <span class="at">method=</span><span class="st">"t.test"</span>) <span class="sc">+</span> </span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_4c-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 4c) Boxplot of gene expression profiles of <em>CCL3</em>.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 4d: CCL3L1 expression ## </span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>ccl3l1 <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">"414062"</span>, <span class="at">intgroup=</span><span class="st">"group"</span>, <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E3"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E4"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>,<span class="st">"KO"</span>))</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ccl3l1, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>count, <span class="at">fill=</span>group)) <span class="sc">+</span> </span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Count"</span>) <span class="sc">+</span> </span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"CCL3L1"</span>) <span class="sc">+</span> </span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face=</span><span class="st">"italic"</span>)) <span class="sc">+</span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons, <span class="at">method=</span><span class="st">"t.test"</span>) <span class="sc">+</span> </span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_4d-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 4d) Boxplot of gene expression profiles of <em>CCL3L1</em>.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 4e: CCL4L2 expression ##</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>ccl4l2 <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">"9650"</span>, <span class="at">intgroup=</span><span class="st">"group"</span>, <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E3"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E4"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>,<span class="st">"KO"</span>))</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ccl4l2, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>count, <span class="at">fill=</span>group)) <span class="sc">+</span> </span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Count"</span>) <span class="sc">+</span> </span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"CCL4L2"</span>) <span class="sc">+</span> </span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face=</span><span class="st">"italic"</span>)) <span class="sc">+</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons, <span class="at">method=</span><span class="st">"t.test"</span>) <span class="sc">+</span> <span class="fu">geom_jitter</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_4e-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 4e) Boxplot of gene expression profiles of <em>CCL4L2</em>.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 4f: CXCL16 expression ## </span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>cxcl16 <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">"58191"</span>, <span class="at">intgroup=</span><span class="st">"group"</span>, <span class="at">returnData=</span><span class="cn">TRUE</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E3"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E4"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>,<span class="st">"KO"</span>))</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(cxcl16, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>count, <span class="at">fill=</span>group)) <span class="sc">+</span> </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Count"</span>) <span class="sc">+</span> </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"CXCL16"</span>) <span class="sc">+</span> </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face=</span><span class="st">"italic"</span>)) <span class="sc">+</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons, <span class="at">method=</span><span class="st">"t.test"</span>) <span class="sc">+</span> <span class="fu">geom_jitter</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_4f-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 4f) Boxplot of expression profiles of <em>CXCL16</em>.</strong></p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>dds_norm <span class="ot">&lt;-</span> <span class="fu">vst</span>(ddsTxi_subset)</span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>dds_norm_t <span class="ot">&lt;-</span> <span class="fu">assay</span>(dds_norm) <span class="sc">%&gt;%</span> <span class="fu">t</span>() </span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Pick soft threshold parameter</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>sft <span class="ot">&lt;-</span> <span class="fu">pickSoftThreshold</span>(dds_norm_t,</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>                         <span class="at">dataIsExpr =</span> <span class="cn">TRUE</span>,</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>                         <span class="at">corFnc =</span> cor,</span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>                         <span class="at">networkType =</span> <span class="st">"signed"</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a>sft_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(sft<span class="sc">$</span>fitIndices) <span class="sc">%&gt;%</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">model_fit =</span> <span class="sc">-</span><span class="fu">sign</span>(slope) <span class="sc">*</span> SFT.R.sq)</span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a>sft_threshold <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(sft_df, <span class="fu">aes</span>(<span class="at">x =</span> Power, <span class="at">y =</span> model_fit, <span class="at">label =</span> Power)) <span class="sc">+</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_text</span>(<span class="at">nudge_y =</span> <span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">0.80</span>, <span class="at">col =</span> <span class="st">"red"</span>) <span class="sc">+</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="fu">min</span>(sft_df<span class="sc">$</span>model_fit), <span class="fl">1.05</span>)) <span class="sc">+</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Soft Threshold (power)"</span>) <span class="sc">+</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Scale Free Topology Model Fit, signed R^2"</span>) <span class="sc">+</span></span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Scale independence"</span>) <span class="sc">+</span></span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>()</span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Run WGCNA</span></span>
<span id="cb25-27"><a href="#cb25-27" aria-hidden="true" tabindex="-1"></a>network <span class="ot">&lt;-</span> <span class="fu">blockwiseModules</span>(dds_norm_t, </span>
<span id="cb25-28"><a href="#cb25-28" aria-hidden="true" tabindex="-1"></a>                            <span class="at">power =</span> <span class="dv">20</span>,</span>
<span id="cb25-29"><a href="#cb25-29" aria-hidden="true" tabindex="-1"></a>                            <span class="at">TOMType =</span> <span class="st">"signed"</span>, </span>
<span id="cb25-30"><a href="#cb25-30" aria-hidden="true" tabindex="-1"></a>                            <span class="at">minModuleSize =</span> <span class="dv">30</span>, </span>
<span id="cb25-31"><a href="#cb25-31" aria-hidden="true" tabindex="-1"></a>                            <span class="at">mergeCutHeight =</span> <span class="fl">0.25</span>,</span>
<span id="cb25-32"><a href="#cb25-32" aria-hidden="true" tabindex="-1"></a>                            <span class="at">numericLabels =</span> <span class="cn">TRUE</span>, </span>
<span id="cb25-33"><a href="#cb25-33" aria-hidden="true" tabindex="-1"></a>                            <span class="at">saveTOMs =</span> <span class="cn">TRUE</span>, </span>
<span id="cb25-34"><a href="#cb25-34" aria-hidden="true" tabindex="-1"></a>                            <span class="at">saveTOMFileBase =</span> <span class="st">"TOM"</span>, </span>
<span id="cb25-35"><a href="#cb25-35" aria-hidden="true" tabindex="-1"></a>                            <span class="at">randomSeed =</span> <span class="dv">123</span>)</span>
<span id="cb25-36"><a href="#cb25-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-37"><a href="#cb25-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Identify modules and assign colors</span></span>
<span id="cb25-38"><a href="#cb25-38" aria-hidden="true" tabindex="-1"></a>module_colours <span class="ot">&lt;-</span> <span class="fu">labels2colors</span>(network<span class="sc">$</span>colors)</span>
<span id="cb25-39"><a href="#cb25-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-40"><a href="#cb25-40" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract module membership and module eigengene values</span></span>
<span id="cb25-41"><a href="#cb25-41" aria-hidden="true" tabindex="-1"></a>module_membership <span class="ot">&lt;-</span> <span class="fu">moduleEigengenes</span>(dds_norm_t, network<span class="sc">$</span>colors)</span>
<span id="cb25-42"><a href="#cb25-42" aria-hidden="true" tabindex="-1"></a>module_eigengenes <span class="ot">&lt;-</span> module_membership<span class="sc">$</span>eigengenes</span>
<span id="cb25-43"><a href="#cb25-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-44"><a href="#cb25-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-45"><a href="#cb25-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Run differential expression analysis </span></span>
<span id="cb25-46"><a href="#cb25-46" aria-hidden="true" tabindex="-1"></a>samples <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">paste</span>(<span class="st">"E2-"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">sep=</span><span class="st">""</span>), <span class="fu">paste</span>(<span class="st">"E3-"</span>, <span class="dv">2</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">sep=</span><span class="st">""</span>), <span class="fu">paste</span>(<span class="st">"E4-"</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">sep=</span><span class="st">""</span>), <span class="fu">paste</span>(<span class="st">"KO-"</span>, <span class="dv">3</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">sep=</span><span class="st">""</span>))</span>
<span id="cb25-47"><a href="#cb25-47" aria-hidden="true" tabindex="-1"></a>group <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"E2"</span>, <span class="at">times =</span> <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"E3"</span>, <span class="at">times =</span> <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"E4"</span>, <span class="at">times =</span> <span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"KO"</span>, <span class="at">times =</span> <span class="dv">3</span>))</span>
<span id="cb25-48"><a href="#cb25-48" aria-hidden="true" tabindex="-1"></a>colData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">samples=</span>samples, <span class="at">group=</span>group)</span>
<span id="cb25-49"><a href="#cb25-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-50"><a href="#cb25-50" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> colData<span class="sc">$</span>group)</span>
<span id="cb25-51"><a href="#cb25-51" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(design) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E3"</span>, <span class="st">"E4"</span>, <span class="st">"KO"</span>)</span>
<span id="cb25-52"><a href="#cb25-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-53"><a href="#cb25-53" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">lmFit</span>(<span class="fu">t</span>(module_eigengenes), <span class="at">design =</span> design)</span>
<span id="cb25-54"><a href="#cb25-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-55"><a href="#cb25-55" aria-hidden="true" tabindex="-1"></a>contrasts <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">makeContrasts</span>(</span>
<span id="cb25-56"><a href="#cb25-56" aria-hidden="true" tabindex="-1"></a>  <span class="at">E4vsE3 =</span> E4 <span class="sc">-</span> E3,</span>
<span id="cb25-57"><a href="#cb25-57" aria-hidden="true" tabindex="-1"></a>  <span class="at">E4vsE2 =</span> E4 <span class="sc">-</span> E2,</span>
<span id="cb25-58"><a href="#cb25-58" aria-hidden="true" tabindex="-1"></a>  <span class="at">E2vsE3 =</span> E2 <span class="sc">-</span> E3, </span>
<span id="cb25-59"><a href="#cb25-59" aria-hidden="true" tabindex="-1"></a>  <span class="at">KOvsE3 =</span> KO <span class="sc">-</span> E3, </span>
<span id="cb25-60"><a href="#cb25-60" aria-hidden="true" tabindex="-1"></a>  <span class="at">levels =</span> <span class="fu">colnames</span>(fit<span class="sc">$</span>coefficients)</span>
<span id="cb25-61"><a href="#cb25-61" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb25-62"><a href="#cb25-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-63"><a href="#cb25-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Apply contrasts to the linear model fit</span></span>
<span id="cb25-64"><a href="#cb25-64" aria-hidden="true" tabindex="-1"></a>contrasts_fit <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">contrasts.fit</span>(fit, contrasts)</span>
<span id="cb25-65"><a href="#cb25-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-66"><a href="#cb25-66" aria-hidden="true" tabindex="-1"></a>contrasts_fit <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">eBayes</span>(contrasts_fit)</span>
<span id="cb25-67"><a href="#cb25-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-68"><a href="#cb25-68" aria-hidden="true" tabindex="-1"></a>E4vsE3_limma <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">topTable</span>(contrasts_fit, <span class="at">coef =</span> <span class="st">"E4vsE3"</span>, <span class="at">number =</span> <span class="cn">Inf</span>)</span>
<span id="cb25-69"><a href="#cb25-69" aria-hidden="true" tabindex="-1"></a>E4vsE2_limma <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">topTable</span>(contrasts_fit, <span class="at">coef =</span> <span class="st">"E4vsE2"</span>, <span class="at">number =</span> <span class="cn">Inf</span>)</span>
<span id="cb25-70"><a href="#cb25-70" aria-hidden="true" tabindex="-1"></a>E2vsE3_limma <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">topTable</span>(contrasts_fit, <span class="at">coef =</span> <span class="st">"E2vsE3"</span>, <span class="at">number =</span> <span class="cn">Inf</span>)</span>
<span id="cb25-71"><a href="#cb25-71" aria-hidden="true" tabindex="-1"></a>KOvsE3_limma <span class="ot">&lt;-</span> limma<span class="sc">::</span><span class="fu">topTable</span>(contrasts_fit, <span class="at">coef =</span> <span class="st">"KOvsE3"</span>, <span class="at">number =</span> <span class="cn">Inf</span>)</span>
<span id="cb25-72"><a href="#cb25-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-73"><a href="#cb25-73" aria-hidden="true" tabindex="-1"></a><span class="co"># enrichment analysis </span></span>
<span id="cb25-74"><a href="#cb25-74" aria-hidden="true" tabindex="-1"></a>gene_module_key <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">enframe</span>(network<span class="sc">$</span>colors, <span class="at">name =</span> <span class="st">"gene"</span>, <span class="at">value =</span> <span class="st">"module"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb25-75"><a href="#cb25-75" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">mutate</span>(<span class="at">module =</span> <span class="fu">paste0</span>(<span class="st">"ME"</span>, module))</span>
<span id="cb25-76"><a href="#cb25-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-77"><a href="#cb25-77" aria-hidden="true" tabindex="-1"></a><span class="co"># get gene ids </span></span>
<span id="cb25-78"><a href="#cb25-78" aria-hidden="true" tabindex="-1"></a>gene_symbols <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb25-79"><a href="#cb25-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">'hgnc_symbol'</span>, <span class="st">'entrezgene_id'</span>),</span>
<span id="cb25-80"><a href="#cb25-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> gene_module_key<span class="sc">$</span>gene,</span>
<span id="cb25-81"><a href="#cb25-81" aria-hidden="true" tabindex="-1"></a>    <span class="at">mart =</span> ensembl</span>
<span id="cb25-82"><a href="#cb25-82" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb25-83"><a href="#cb25-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-84"><a href="#cb25-84" aria-hidden="true" tabindex="-1"></a>gene_module_key <span class="ot">&lt;-</span> <span class="fu">merge</span>(gene_module_key, gene_symbols, <span class="at">by.x=</span><span class="st">"gene"</span>, <span class="at">by.y=</span><span class="st">"entrezgene_id"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><strong>Figure 5: Gene networks upregulated in APOE2 microglia are associated with cellular migration and immune responses.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Proliferation and migration module </span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>module_eigengenes_exp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(module_eigengenes, colData)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>module_eigengenes_exp<span class="sc">$</span>title <span class="ot">&lt;-</span> <span class="st">"Proliferation and migration module"</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(module_eigengenes_exp, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> ME1, <span class="at">fill =</span> group)) <span class="sc">+</span></span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.3</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a>  ggforce<span class="sc">::</span><span class="fu">geom_sina</span>(<span class="at">maxwidth =</span> <span class="fl">0.3</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"Module eigengenes"</span>) <span class="sc">+</span> </span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> title) <span class="sc">+</span> </span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">18</span>)) <span class="sc">+</span> </span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"#92BBD9FF"</span>)) <span class="sc">+</span> </span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_5a-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="432"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 5a) Barplot of eigengene expression of the WGCNA module associated with proliferation and migration.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Run pathway enrichment analysis </span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>all_GO_res <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="co"># List of APOE contrast data frames with names</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>limma_dfs <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">E2vsE3 =</span> E2vsE3_limma, <span class="at">E4vsE3 =</span> E4vsE3_limma, <span class="at">E4vsE2 =</span> E4vsE2_limma, <span class="at">KOvsE3 =</span> KOvsE3_limma)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Loop through each data frame</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (contrast <span class="cf">in</span> <span class="fu">names</span>(limma_dfs)) {</span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(contrast)</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  current_df <span class="ot">&lt;-</span> limma_dfs[[contrast]]</span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>  subset_df <span class="ot">&lt;-</span> current_df <span class="sc">%&gt;%</span> <span class="fu">filter</span>(adj.P.Val <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb27-14"><a href="#cb27-14" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-15"><a href="#cb27-15" aria-hidden="true" tabindex="-1"></a>  subset_df<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(subset_df<span class="sc">$</span>logFC <span class="sc">&gt;</span> <span class="dv">0</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span>)</span>
<span id="cb27-16"><a href="#cb27-16" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-17"><a href="#cb27-17" aria-hidden="true" tabindex="-1"></a>  module_names <span class="ot">&lt;-</span> <span class="fu">rownames</span>(subset_df)</span>
<span id="cb27-18"><a href="#cb27-18" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb27-19"><a href="#cb27-19" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Iterate through each module name</span></span>
<span id="cb27-20"><a href="#cb27-20" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (module_name <span class="cf">in</span> module_names) {</span>
<span id="cb27-21"><a href="#cb27-21" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb27-22"><a href="#cb27-22" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Filter gene_module_key for the specific module</span></span>
<span id="cb27-23"><a href="#cb27-23" aria-hidden="true" tabindex="-1"></a>    ME <span class="ot">&lt;-</span> gene_module_key <span class="sc">%&gt;%</span> <span class="fu">filter</span>(module <span class="sc">==</span> module_name)</span>
<span id="cb27-24"><a href="#cb27-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">print</span>(module_name)</span>
<span id="cb27-25"><a href="#cb27-25" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb27-26"><a href="#cb27-26" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Perform pathway enrichment analysis</span></span>
<span id="cb27-27"><a href="#cb27-27" aria-hidden="true" tabindex="-1"></a>    GO_BP <span class="ot">&lt;-</span> <span class="fu">enrichGO</span>(ME<span class="sc">$</span>hgnc_symbol,</span>
<span id="cb27-28"><a href="#cb27-28" aria-hidden="true" tabindex="-1"></a>                      <span class="at">OrgDb =</span> org.Hs.eg.db, <span class="at">keyType =</span> <span class="st">"SYMBOL"</span>,</span>
<span id="cb27-29"><a href="#cb27-29" aria-hidden="true" tabindex="-1"></a>                      <span class="at">ont =</span> <span class="st">"BP"</span>, <span class="at">pvalueCutoff =</span> <span class="fl">0.05</span>, <span class="at">pAdjustMethod =</span> <span class="st">"fdr"</span>,</span>
<span id="cb27-30"><a href="#cb27-30" aria-hidden="true" tabindex="-1"></a>                      <span class="at">universe =</span> <span class="fu">unique</span>(gene_module_key<span class="sc">$</span>SYMBOL),</span>
<span id="cb27-31"><a href="#cb27-31" aria-hidden="true" tabindex="-1"></a>                      <span class="at">qvalueCutoff =</span> <span class="fl">0.05</span>,</span>
<span id="cb27-32"><a href="#cb27-32" aria-hidden="true" tabindex="-1"></a>                      <span class="at">minGSSize =</span> <span class="dv">10</span>, <span class="at">maxGSSize =</span> <span class="dv">600</span>, <span class="at">readable =</span> <span class="cn">FALSE</span>, <span class="at">pool =</span> <span class="cn">FALSE</span>)</span>
<span id="cb27-33"><a href="#cb27-33" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb27-34"><a href="#cb27-34" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Check if there are results</span></span>
<span id="cb27-35"><a href="#cb27-35" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span> (<span class="fu">nrow</span>(GO_BP) <span class="sc">&gt;</span> <span class="dv">0</span>) {</span>
<span id="cb27-36"><a href="#cb27-36" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Add information to the result list</span></span>
<span id="cb27-37"><a href="#cb27-37" aria-hidden="true" tabindex="-1"></a>      all_GO_res[[contrast]][[module_name]]<span class="sc">$</span>GO_Biological_Process <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(GO_BP)</span>
<span id="cb27-38"><a href="#cb27-38" aria-hidden="true" tabindex="-1"></a>      all_GO_res[[contrast]][[module_name]]<span class="sc">$</span>GO_Biological_Process<span class="sc">$</span>direction <span class="ot">&lt;-</span> subset_df<span class="sc">$</span>direction[<span class="fu">rownames</span>(subset_df) <span class="sc">==</span> module_name]</span>
<span id="cb27-39"><a href="#cb27-39" aria-hidden="true" tabindex="-1"></a>      all_GO_res[[contrast]][[module_name]]<span class="sc">$</span>GO_Biological_Process<span class="sc">$</span>contrast <span class="ot">&lt;-</span> <span class="fu">paste0</span>(all_GO_res[[contrast]][[module_name]]<span class="sc">$</span>GO_Biological_Process<span class="sc">$</span>direction, <span class="st">" in "</span>, contrast)</span>
<span id="cb27-40"><a href="#cb27-40" aria-hidden="true" tabindex="-1"></a>      all_GO_res[[contrast]][[module_name]]<span class="sc">$</span>GO_Biological_Process<span class="sc">$</span>category <span class="ot">&lt;-</span> <span class="st">"Biological process"</span></span>
<span id="cb27-41"><a href="#cb27-41" aria-hidden="true" tabindex="-1"></a>      all_GO_res[[contrast]][[module_name]]<span class="sc">$</span>GO_Biological_Process<span class="sc">$</span>module <span class="ot">&lt;-</span> module_name</span>
<span id="cb27-42"><a href="#cb27-42" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb27-43"><a href="#cb27-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb27-44"><a href="#cb27-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb27-45"><a href="#cb27-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-46"><a href="#cb27-46" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-47"><a href="#cb27-47" aria-hidden="true" tabindex="-1"></a>all_GO_res_df <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(<span class="fu">unlist</span>(<span class="fu">unlist</span>(all_GO_res, <span class="at">recursive =</span> <span class="cn">FALSE</span>), <span class="at">recursive =</span> <span class="cn">FALSE</span>), as.data.frame))</span>
<span id="cb27-48"><a href="#cb27-48" aria-hidden="true" tabindex="-1"></a>all_GO_res_df<span class="sc">$</span>module_name[all_GO_res_df<span class="sc">$</span>module <span class="sc">==</span> <span class="st">"ME1"</span>] <span class="ot">&lt;-</span> <span class="st">"Proliferation and migration module"</span></span>
<span id="cb27-49"><a href="#cb27-49" aria-hidden="true" tabindex="-1"></a>all_GO_res_df<span class="sc">$</span>module_name[all_GO_res_df<span class="sc">$</span>module <span class="sc">==</span> <span class="st">"ME22"</span>] <span class="ot">&lt;-</span> <span class="st">"Immune response module"</span></span>
<span id="cb27-50"><a href="#cb27-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-51"><a href="#cb27-51" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate gene ratio </span></span>
<span id="cb27-52"><a href="#cb27-52" aria-hidden="true" tabindex="-1"></a>all_GO_res_df <span class="ot">&lt;-</span> all_GO_res_df <span class="sc">%&gt;%</span></span>
<span id="cb27-53"><a href="#cb27-53" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rowwise</span>() <span class="sc">%&gt;%</span></span>
<span id="cb27-54"><a href="#cb27-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">GeneRatio_calc =</span> <span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(GeneRatio, <span class="st">"/"</span>)[[<span class="dv">1</span>]])[<span class="dv">1</span>] <span class="sc">/</span></span>
<span id="cb27-55"><a href="#cb27-55" aria-hidden="true" tabindex="-1"></a>           <span class="fu">as.numeric</span>(<span class="fu">strsplit</span>(GeneRatio, <span class="st">"/"</span>)[[<span class="dv">1</span>]])[<span class="dv">2</span>])</span>
<span id="cb27-56"><a href="#cb27-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-57"><a href="#cb27-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-58"><a href="#cb27-58" aria-hidden="true" tabindex="-1"></a><span class="co"># get top 30 pathways </span></span>
<span id="cb27-59"><a href="#cb27-59" aria-hidden="true" tabindex="-1"></a>ME1_module_top30 <span class="ot">&lt;-</span> <span class="fu">head</span>(all_GO_res_df[<span class="fu">order</span>(all_GO_res_df<span class="sc">$</span>GeneRatio_calc, <span class="at">decreasing =</span> <span class="cn">TRUE</span>) <span class="sc">&amp;</span> all_GO_res_df<span class="sc">$</span>module <span class="sc">==</span> <span class="st">"ME1"</span>, ], <span class="dv">30</span>)</span>
<span id="cb27-60"><a href="#cb27-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-61"><a href="#cb27-61" aria-hidden="true" tabindex="-1"></a><span class="co"># create colours vector for y-axis labels associated with proliferation and migration</span></span>
<span id="cb27-62"><a href="#cb27-62" aria-hidden="true" tabindex="-1"></a>colours_vector <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb27-63"><a href="#cb27-63" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grepl</span>(<span class="st">"migration|proliferation|chemokine|motility|locomotion"</span>, ME1_module_top30<span class="sc">$</span>Description),</span>
<span id="cb27-64"><a href="#cb27-64" aria-hidden="true" tabindex="-1"></a>  <span class="st">"#92BBD9FF"</span>,</span>
<span id="cb27-65"><a href="#cb27-65" aria-hidden="true" tabindex="-1"></a>  <span class="st">"black"</span></span>
<span id="cb27-66"><a href="#cb27-66" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb27-67"><a href="#cb27-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-68"><a href="#cb27-68" aria-hidden="true" tabindex="-1"></a>colours_vector <span class="ot">&lt;-</span> colours_vector[<span class="fu">order</span>(ME1_module_top30<span class="sc">$</span>GeneRatio_calc, ME1_module_top30<span class="sc">$</span>Description)]</span>
<span id="cb27-69"><a href="#cb27-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-70"><a href="#cb27-70" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot top 30 GO BP enrichments for the ME1 (proliferation and migration) module</span></span>
<span id="cb27-71"><a href="#cb27-71" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ME1_module_top30, </span>
<span id="cb27-72"><a href="#cb27-72" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>GeneRatio_calc, <span class="at">y=</span><span class="fu">reorder</span>(Description, GeneRatio_calc), <span class="at">colour=</span>p.adjust, <span class="at">size=</span>Count, <span class="at">shape=</span>category)) <span class="sc">+</span> </span>
<span id="cb27-73"><a href="#cb27-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb27-74"><a href="#cb27-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>() <span class="sc">+</span> </span>
<span id="cb27-75"><a href="#cb27-75" aria-hidden="true" tabindex="-1"></a>  DOSE<span class="sc">::</span><span class="fu">theme_dose</span>() <span class="sc">+</span> </span>
<span id="cb27-76"><a href="#cb27-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_gradient</span>(<span class="at">low =</span> <span class="st">"#C3DAEAFF"</span>, <span class="at">high =</span> <span class="st">"#92BBD9FF"</span>) <span class="sc">+</span>  <span class="co"># Use the colors you prefer</span></span>
<span id="cb27-77"><a href="#cb27-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Gene ratio"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"GO Biological Process"</span>) <span class="sc">+</span> </span>
<span id="cb27-78"><a href="#cb27-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size</span>(<span class="at">range=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">7</span>)) <span class="sc">+</span> </span>
<span id="cb27-79"><a href="#cb27-79" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">"Gene count"</span>), <span class="at">shape=</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb27-80"><a href="#cb27-80" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">vjust=</span><span class="dv">1</span>), </span>
<span id="cb27-81"><a href="#cb27-81" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size=</span><span class="dv">18</span>),</span>
<span id="cb27-82"><a href="#cb27-82" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">colour =</span> colours_vector),</span>
<span id="cb27-83"><a href="#cb27-83" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">20</span>),</span>
<span id="cb27-84"><a href="#cb27-84" aria-hidden="true" tabindex="-1"></a>        <span class="at">text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">20</span>))  <span class="sc">+</span> </span>
<span id="cb27-85"><a href="#cb27-85" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>module_name) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"Up in APOE2"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_5b-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="1056"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 5b) GO biological processes enriched for genes within the WGCNA module associated with proliferation and migration that is upregulated in <em>APOE2</em> microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Immune response module </span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>module_eigengenes_exp <span class="ot">&lt;-</span> <span class="fu">cbind</span>(module_eigengenes, colData)</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>module_eigengenes_exp<span class="sc">$</span>title <span class="ot">&lt;-</span> <span class="st">"Immune response module"</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(module_eigengenes_exp, <span class="fu">aes</span>(<span class="at">x =</span> group, <span class="at">y =</span> ME22, <span class="at">fill =</span> group)) <span class="sc">+</span></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">width =</span> <span class="fl">0.3</span>, <span class="at">outlier.shape =</span> <span class="cn">NA</span>) <span class="sc">+</span> </span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  ggforce<span class="sc">::</span><span class="fu">geom_sina</span>(<span class="at">maxwidth =</span> <span class="fl">0.3</span>) <span class="sc">+</span> </span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Module eigengenes"</span>) <span class="sc">+</span> </span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(. <span class="sc">~</span> title) <span class="sc">+</span> </span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">18</span>)) <span class="sc">+</span> </span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"#92BBD9FF"</span>)) <span class="sc">+</span> </span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_5c-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="432"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 5c) Barplot of eigengene expression of the WGCNA module associated with immune responses.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get top 30 pathways </span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>ME22_module_top30 <span class="ot">&lt;-</span> <span class="fu">head</span>(all_GO_res_df[<span class="fu">order</span>(all_GO_res_df<span class="sc">$</span>GeneRatio_calc, <span class="at">decreasing =</span> <span class="cn">TRUE</span>) <span class="sc">&amp;</span> all_GO_res_df<span class="sc">$</span>module <span class="sc">==</span> <span class="st">"ME22"</span>, ], <span class="dv">30</span>)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="co"># create colours vector for y-axis labels associated with proliferation and migration</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>colours_vector <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">grepl</span>(<span class="st">"immune|immunity|wounding|complement|wound"</span>, ME22_module_top30<span class="sc">$</span>Description),</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"#92BBD9FF"</span>,</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"black"</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>colours_vector <span class="ot">&lt;-</span> colours_vector[<span class="fu">order</span>(ME22_module_top30<span class="sc">$</span>GeneRatio_calc, ME22_module_top30<span class="sc">$</span>Description)]</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a><span class="co"># plot top 30 GO BP enrichments for M22 (immune response) module </span></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ME22_module_top30, </span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>       <span class="fu">aes</span>(<span class="at">x=</span>GeneRatio_calc, <span class="at">y=</span><span class="fu">reorder</span>(Description, GeneRatio_calc), <span class="at">colour=</span>p.adjust, <span class="at">size=</span>Count, <span class="at">shape=</span>category)) <span class="sc">+</span> </span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>() <span class="sc">+</span> </span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>  DOSE<span class="sc">::</span><span class="fu">theme_dose</span>() <span class="sc">+</span> </span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_gradient</span>(<span class="at">low =</span> <span class="st">"#C3DAEAFF"</span>, <span class="at">high =</span> <span class="st">"#92BBD9FF"</span>) <span class="sc">+</span>  <span class="co"># Use the colors you prefer</span></span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"Gene ratio"</span>) <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"GO Biological Process"</span>) <span class="sc">+</span> </span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_size</span>(<span class="at">range=</span><span class="fu">c</span>(<span class="dv">3</span>,<span class="dv">7</span>)) <span class="sc">+</span> </span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">size=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">"Gene count"</span>), <span class="at">shape=</span> <span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle=</span><span class="dv">45</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">vjust=</span><span class="dv">1</span>), </span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size=</span><span class="dv">20</span>),</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.text.y =</span> <span class="fu">element_text</span>(<span class="at">colour =</span> colours_vector),</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">20</span>),</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>        <span class="at">text=</span><span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">18</span>)) <span class="sc">+</span> </span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Up in APOE2"</span>) <span class="sc">+</span></span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"adaptive immune response based on somatic recombination of immune receptors built from immunoglobulin superfamily domains"</span> <span class="ot">=</span> <span class="st">"adaptive immune response based on somatic recombination of </span><span class="sc">\n</span><span class="st"> immune receptors built from immunoglobulin superfamily domains"</span>)) <span class="sc">+</span> </span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>module_name)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_5d-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="1056"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 5d) GO biological processes enriched for genes within the WGCNA module associated with immune responses, which is upregulated in <em>APOE2</em> microglia.</strong></p>
<p><strong>Figure 6: Regions with increased chromatin accessibility in <em>APOE2</em> microglia are enriched for the binding of vitamin D receptor.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 6a: motif enrichment heatmap ## </span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Read in motif enrichment results for top 100 upregulated peaks in E2 vs E3</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>homer_res <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/HOMER_motif_enrich_E2vsE3_top100_res.csv"</span>)</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>homer_res<span class="sc">$</span>Percentage.of.Targets <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"%"</span>, <span class="st">""</span>, homer_res<span class="sc">$</span>Percentage.of.Targets)</span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>homer_res<span class="sc">$</span>Percentage.of.Background <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">"%"</span>, <span class="st">""</span>, homer_res<span class="sc">$</span>Percentage.of.Background)</span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a><span class="co"># delete rows where percentage in background is 0 </span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>homer_res <span class="ot">&lt;-</span> homer_res[<span class="fu">as.numeric</span>(homer_res<span class="sc">$</span>Percentage.of.Background) <span class="sc">!=</span> <span class="fl">0.00</span>, ]</span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-12"><a href="#cb30-12" aria-hidden="true" tabindex="-1"></a><span class="co"># calculate log2enr </span></span>
<span id="cb30-13"><a href="#cb30-13" aria-hidden="true" tabindex="-1"></a>homer_res<span class="sc">$</span>logFC <span class="ot">&lt;-</span> <span class="fu">log2</span>(<span class="fu">as.numeric</span>(homer_res<span class="sc">$</span>Percentage.of.Targets) <span class="sc">/</span> <span class="fu">as.numeric</span>(homer_res<span class="sc">$</span>Percentage.of.Background))</span>
<span id="cb30-14"><a href="#cb30-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-15"><a href="#cb30-15" aria-hidden="true" tabindex="-1"></a><span class="co"># multiple testing correction </span></span>
<span id="cb30-16"><a href="#cb30-16" aria-hidden="true" tabindex="-1"></a>homer_res<span class="sc">$</span>fdr <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(homer_res<span class="sc">$</span>P.value, <span class="at">method=</span><span class="st">"fdr"</span>)</span>
<span id="cb30-17"><a href="#cb30-17" aria-hidden="true" tabindex="-1"></a>homer_res<span class="sc">$</span>log10fdr <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fu">log10</span>(homer_res<span class="sc">$</span>fdr)</span>
<span id="cb30-18"><a href="#cb30-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-19"><a href="#cb30-19" aria-hidden="true" tabindex="-1"></a>homer_res<span class="sc">$</span>motif_name <span class="ot">&lt;-</span> <span class="fu">sub</span>(<span class="st">"^([^</span><span class="sc">\\</span><span class="st">(/]+).*"</span>, <span class="st">"</span><span class="sc">\\</span><span class="st">1"</span>, homer_res<span class="sc">$</span>Best.Match.Details)</span>
<span id="cb30-20"><a href="#cb30-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-21"><a href="#cb30-21" aria-hidden="true" tabindex="-1"></a>plot_fc <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(homer_res, <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> motif_name, <span class="at">fill =</span> logFC)) <span class="sc">+</span></span>
<span id="cb30-22"><a href="#cb30-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb30-23"><a href="#cb30-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span> </span>
<span id="cb30-24"><a href="#cb30-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"#B2182B"</span>) <span class="sc">+</span></span>
<span id="cb30-25"><a href="#cb30-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">""</span>, <span class="at">fill =</span> <span class="st">"LogFC"</span>) <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb30-26"><a href="#cb30-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb30-27"><a href="#cb30-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-28"><a href="#cb30-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-29"><a href="#cb30-29" aria-hidden="true" tabindex="-1"></a>plot_fdr <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(homer_res, <span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span>, <span class="at">y =</span> motif_name, <span class="at">fill =</span> log10fdr)) <span class="sc">+</span></span>
<span id="cb30-30"><a href="#cb30-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_tile</span>() <span class="sc">+</span></span>
<span id="cb30-31"><a href="#cb30-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_gradient2</span>(<span class="at">low =</span> <span class="st">"white"</span>, <span class="at">high =</span> <span class="st">"#4D4D4D"</span>) <span class="sc">+</span></span>
<span id="cb30-32"><a href="#cb30-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb30-33"><a href="#cb30-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">""</span>, <span class="at">fill =</span> <span class="st">"Log10(FDR)"</span>) <span class="sc">+</span> <span class="fu">coord_fixed</span>() <span class="sc">+</span></span>
<span id="cb30-34"><a href="#cb30-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span>
<span id="cb30-35"><a href="#cb30-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-36"><a href="#cb30-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb30-37"><a href="#cb30-37" aria-hidden="true" tabindex="-1"></a>motif_heatmaps <span class="ot">&lt;-</span> plot_fc <span class="sc">+</span> plot_fdr <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">guides =</span> <span class="st">"collect"</span>) <span class="sc">&amp;</span> <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><img src="figures/motif_enrich_heatmap.png" class="img-fluid"></p>
<p><strong>Figure 6a) Heatmap showing enrichment of motifs in regions with increased chromatin accessibility in <em>APOE2</em> microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Figure 6b: VDR target gene overlap ## </span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>vdr_target_genes <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/VDR_target_genes_24h_monocytes_Warwick2021.csv"</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a>vdr_target_genes_pos <span class="ot">&lt;-</span> vdr_target_genes[vdr_target_genes<span class="sc">$</span>log2FoldChange<span class="sc">&gt;</span><span class="dv">0</span> <span class="sc">&amp;</span> vdr_target_genes<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>vdr_target_genes_pos <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(vdr_target_genes_pos)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a>vdr_target_genes_pos<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="st">"Upregulated"</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>vdr_target_genes_neg <span class="ot">&lt;-</span> vdr_target_genes[vdr_target_genes<span class="sc">$</span>log2FoldChange<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> vdr_target_genes<span class="sc">$</span>padj<span class="sc">&lt;</span><span class="fl">0.05</span>,]</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>vdr_target_genes_neg <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(vdr_target_genes_neg)</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>vdr_target_genes_neg<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="st">"Downregulated"</span></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a>vdr_target_genes_all <span class="ot">&lt;-</span> <span class="fu">list</span>(<span class="at">Upregulated =</span> vdr_target_genes_pos, <span class="at">Downregulated =</span> vdr_target_genes_neg)</span>
<span id="cb31-13"><a href="#cb31-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-14"><a href="#cb31-14" aria-hidden="true" tabindex="-1"></a>run_hypergeo_test_vdr <span class="ot">&lt;-</span> <span class="cf">function</span>(df, enrichment_data_frames, comparison) {</span>
<span id="cb31-15"><a href="#cb31-15" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Initialize an empty results dataframe</span></span>
<span id="cb31-16"><a href="#cb31-16" aria-hidden="true" tabindex="-1"></a>  results_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span>
<span id="cb31-17"><a href="#cb31-17" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-18"><a href="#cb31-18" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Filter by FDR &lt; 0.05</span></span>
<span id="cb31-19"><a href="#cb31-19" aria-hidden="true" tabindex="-1"></a>  filtered_df <span class="ot">&lt;-</span> df[df<span class="sc">$</span>fdr <span class="sc">&lt;</span> <span class="fl">0.05</span>,]</span>
<span id="cb31-20"><a href="#cb31-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-21"><a href="#cb31-21" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset rows with log2FoldChange &gt; 0</span></span>
<span id="cb31-22"><a href="#cb31-22" aria-hidden="true" tabindex="-1"></a>  upregulated <span class="ot">&lt;-</span> filtered_df[filtered_df<span class="sc">$</span>log2FoldChange <span class="sc">&gt;</span> <span class="dv">0</span>,]</span>
<span id="cb31-23"><a href="#cb31-23" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-24"><a href="#cb31-24" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Subset rows with log2FoldChange &lt; 0</span></span>
<span id="cb31-25"><a href="#cb31-25" aria-hidden="true" tabindex="-1"></a>  downregulated <span class="ot">&lt;-</span> filtered_df[filtered_df<span class="sc">$</span>log2FoldChange <span class="sc">&lt;</span> <span class="dv">0</span>,]</span>
<span id="cb31-26"><a href="#cb31-26" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-27"><a href="#cb31-27" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Iterate over each enrichment dataframe</span></span>
<span id="cb31-28"><a href="#cb31-28" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (enrichment_df <span class="cf">in</span> enrichment_data_frames) {</span>
<span id="cb31-29"><a href="#cb31-29" aria-hidden="true" tabindex="-1"></a>    total_genes <span class="ot">&lt;-</span> <span class="fu">nrow</span>(df)</span>
<span id="cb31-30"><a href="#cb31-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb31-31"><a href="#cb31-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate overlap with upregulated genes</span></span>
<span id="cb31-32"><a href="#cb31-32" aria-hidden="true" tabindex="-1"></a>    upregulated_ol <span class="ot">&lt;-</span> <span class="fu">intersect</span>(upregulated<span class="sc">$</span>hgnc_symbol, enrichment_df<span class="sc">$</span>GeneSymbol)</span>
<span id="cb31-33"><a href="#cb31-33" aria-hidden="true" tabindex="-1"></a>    GeneRatio_upregulated <span class="ot">&lt;-</span> <span class="fu">length</span>(upregulated_ol) <span class="sc">/</span> <span class="fu">nrow</span>(enrichment_df)</span>
<span id="cb31-34"><a href="#cb31-34" aria-hidden="true" tabindex="-1"></a>    Count_upregulated <span class="ot">&lt;-</span> <span class="fu">length</span>(upregulated_ol)</span>
<span id="cb31-35"><a href="#cb31-35" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb31-36"><a href="#cb31-36" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate overlap with downregulated genes</span></span>
<span id="cb31-37"><a href="#cb31-37" aria-hidden="true" tabindex="-1"></a>    downregulated_ol <span class="ot">&lt;-</span> <span class="fu">intersect</span>(downregulated<span class="sc">$</span>hgnc_symbol, enrichment_df<span class="sc">$</span>GeneSymbol)</span>
<span id="cb31-38"><a href="#cb31-38" aria-hidden="true" tabindex="-1"></a>    GeneRatio_downregulated <span class="ot">&lt;-</span> <span class="fu">length</span>(downregulated_ol) <span class="sc">/</span> <span class="fu">nrow</span>(enrichment_df)</span>
<span id="cb31-39"><a href="#cb31-39" aria-hidden="true" tabindex="-1"></a>    Count_downregulated <span class="ot">&lt;-</span> <span class="fu">length</span>(downregulated_ol)</span>
<span id="cb31-40"><a href="#cb31-40" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb31-41"><a href="#cb31-41" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate p-values for upregulated genes</span></span>
<span id="cb31-42"><a href="#cb31-42" aria-hidden="true" tabindex="-1"></a>    p_value_upregulated <span class="ot">&lt;-</span> <span class="cf">if</span> (<span class="fu">length</span>(upregulated_ol) <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb31-43"><a href="#cb31-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">phyper</span>((<span class="fu">length</span>(upregulated_ol) <span class="sc">-</span> <span class="dv">1</span>), <span class="fu">nrow</span>(enrichment_df), total_genes <span class="sc">-</span> <span class="fu">nrow</span>(enrichment_df), <span class="fu">nrow</span>(upregulated), <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-44"><a href="#cb31-44" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb31-45"><a href="#cb31-45" aria-hidden="true" tabindex="-1"></a>      <span class="dv">1</span></span>
<span id="cb31-46"><a href="#cb31-46" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb31-47"><a href="#cb31-47" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb31-48"><a href="#cb31-48" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate p-values for downregulated genes</span></span>
<span id="cb31-49"><a href="#cb31-49" aria-hidden="true" tabindex="-1"></a>    p_value_downregulated <span class="ot">&lt;-</span> <span class="cf">if</span> (<span class="fu">length</span>(downregulated_ol) <span class="sc">&gt;</span> <span class="dv">1</span>) {</span>
<span id="cb31-50"><a href="#cb31-50" aria-hidden="true" tabindex="-1"></a>      <span class="fu">phyper</span>((<span class="fu">length</span>(downregulated_ol) <span class="sc">-</span> <span class="dv">1</span>), <span class="fu">nrow</span>(enrichment_df), total_genes <span class="sc">-</span> <span class="fu">nrow</span>(enrichment_df), <span class="fu">nrow</span>(downregulated), <span class="at">lower.tail =</span> <span class="cn">FALSE</span>)</span>
<span id="cb31-51"><a href="#cb31-51" aria-hidden="true" tabindex="-1"></a>    } <span class="cf">else</span> {</span>
<span id="cb31-52"><a href="#cb31-52" aria-hidden="true" tabindex="-1"></a>      <span class="dv">1</span></span>
<span id="cb31-53"><a href="#cb31-53" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb31-54"><a href="#cb31-54" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb31-55"><a href="#cb31-55" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create data frames for the results and append them to the main results dataframe</span></span>
<span id="cb31-56"><a href="#cb31-56" aria-hidden="true" tabindex="-1"></a>    result_upregulated <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Comparison =</span> comparison, <span class="at">Direction_apoe =</span> <span class="st">"Upregulated"</span>, </span>
<span id="cb31-57"><a href="#cb31-57" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">p =</span> p_value_upregulated, <span class="at">Direction_vdr =</span> <span class="fu">unique</span>(enrichment_df<span class="sc">$</span>direction),</span>
<span id="cb31-58"><a href="#cb31-58" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">GeneRatio =</span> GeneRatio_upregulated, <span class="at">Count =</span> Count_upregulated)</span>
<span id="cb31-59"><a href="#cb31-59" aria-hidden="true" tabindex="-1"></a>    result_downregulated <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Comparison =</span> comparison, <span class="at">Direction_apoe =</span> <span class="st">"Downregulated"</span>, </span>
<span id="cb31-60"><a href="#cb31-60" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">p =</span> p_value_downregulated, <span class="at">Direction_vdr =</span> <span class="fu">unique</span>(enrichment_df<span class="sc">$</span>direction),</span>
<span id="cb31-61"><a href="#cb31-61" aria-hidden="true" tabindex="-1"></a>                                       <span class="at">GeneRatio =</span> GeneRatio_downregulated, <span class="at">Count =</span> Count_downregulated)</span>
<span id="cb31-62"><a href="#cb31-62" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb31-63"><a href="#cb31-63" aria-hidden="true" tabindex="-1"></a>    results_df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(results_df, result_upregulated, result_downregulated)</span>
<span id="cb31-64"><a href="#cb31-64" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb31-65"><a href="#cb31-65" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb31-66"><a href="#cb31-66" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Return the results dataframe</span></span>
<span id="cb31-67"><a href="#cb31-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(results_df)</span>
<span id="cb31-68"><a href="#cb31-68" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb31-69"><a href="#cb31-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-70"><a href="#cb31-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-71"><a href="#cb31-71" aria-hidden="true" tabindex="-1"></a>E2vsE3_vdr_ol <span class="ot">&lt;-</span> <span class="fu">run_hypergeo_test_vdr</span>(E2vsE3_rna, vdr_target_genes_all, <span class="st">"E2vsE3"</span>)</span>
<span id="cb31-72"><a href="#cb31-72" aria-hidden="true" tabindex="-1"></a>E4vsE3_vdr_ol <span class="ot">&lt;-</span> <span class="fu">run_hypergeo_test_vdr</span>(E4vsE3_rna, vdr_target_genes_all, <span class="st">"E4vsE3"</span>)</span>
<span id="cb31-73"><a href="#cb31-73" aria-hidden="true" tabindex="-1"></a>E4vsE2_vdr_ol <span class="ot">&lt;-</span> <span class="fu">run_hypergeo_test_vdr</span>(E4vsE2_rna, vdr_target_genes_all, <span class="st">"E4vsE2"</span>)</span>
<span id="cb31-74"><a href="#cb31-74" aria-hidden="true" tabindex="-1"></a>KOvsE3_vdr_ol <span class="ot">&lt;-</span> <span class="fu">run_hypergeo_test_vdr</span>(KOvsE3_rna, vdr_target_genes_all, <span class="st">"KOvsE3"</span>)</span>
<span id="cb31-75"><a href="#cb31-75" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-76"><a href="#cb31-76" aria-hidden="true" tabindex="-1"></a>all_vdr_ol <span class="ot">&lt;-</span> <span class="fu">rbind</span>(E2vsE3_vdr_ol, E4vsE3_vdr_ol, E4vsE2_vdr_ol, KOvsE3_vdr_ol)</span>
<span id="cb31-77"><a href="#cb31-77" aria-hidden="true" tabindex="-1"></a>all_vdr_ol<span class="sc">$</span>p.adjust <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(all_vdr_ol<span class="sc">$</span>p, <span class="at">method=</span><span class="st">"bonf"</span>)</span>
<span id="cb31-78"><a href="#cb31-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-79"><a href="#cb31-79" aria-hidden="true" tabindex="-1"></a>all_vdr_ol<span class="sc">$</span>label_vdr <span class="ot">&lt;-</span> <span class="fu">paste</span>(all_vdr_ol<span class="sc">$</span>Direction_vdr, <span class="st">"in"</span>, all_vdr_ol<span class="sc">$</span>Comparison)</span>
<span id="cb31-80"><a href="#cb31-80" aria-hidden="true" tabindex="-1"></a>all_vdr_ol<span class="sc">$</span>label_apoe <span class="ot">&lt;-</span> <span class="fu">paste</span>(all_vdr_ol<span class="sc">$</span>Direction_apoe, <span class="st">"in"</span>, all_vdr_ol<span class="sc">$</span>Comparison)</span>
<span id="cb31-81"><a href="#cb31-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-82"><a href="#cb31-82" aria-hidden="true" tabindex="-1"></a>vdr_ol_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(all_vdr_ol[all_vdr_ol<span class="sc">$</span>Direction_vdr<span class="sc">==</span><span class="st">"Upregulated"</span>,],</span>
<span id="cb31-83"><a href="#cb31-83" aria-hidden="true" tabindex="-1"></a>                          <span class="fu">aes</span>(<span class="at">x=</span><span class="sc">-</span><span class="fu">log10</span>(p), <span class="at">y=</span><span class="fu">reorder</span>(label_apoe, p))) <span class="sc">+</span> </span>
<span id="cb31-84"><a href="#cb31-84" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">fill=</span>label_apoe)) <span class="sc">+</span></span>
<span id="cb31-85"><a href="#cb31-85" aria-hidden="true" tabindex="-1"></a>  DOSE<span class="sc">::</span><span class="fu">theme_dose</span>() <span class="sc">+</span> </span>
<span id="cb31-86"><a href="#cb31-86" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span><span class="st">"none"</span>) <span class="sc">+</span> </span>
<span id="cb31-87"><a href="#cb31-87" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">"-log10(p)"</span>) <span class="sc">+</span> </span>
<span id="cb31-88"><a href="#cb31-88" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"Up VDR target genes in monocytes (24 hours)"</span>) <span class="sc">+</span></span>
<span id="cb31-89"><a href="#cb31-89" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb31-90"><a href="#cb31-90" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept=</span><span class="sc">-</span><span class="fu">log10</span>(<span class="fl">0.003125</span>), <span class="at">linetype=</span><span class="st">"dashed"</span>, <span class="at">color=</span><span class="st">"darkgrey"</span>) <span class="sc">+</span></span>
<span id="cb31-91"><a href="#cb31-91" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"#B1D5BBFF"</span>, <span class="st">"#92BBD9FF"</span>, <span class="st">"#4D6D93FF"</span>, <span class="st">"#9DAFC3FF"</span>, <span class="st">"#ECE28BFF"</span>, <span class="st">"#DCCA2CFF"</span>, <span class="st">"#C3DAEAFF"</span>, <span class="st">"#DCCA2CFF"</span>))</span>
<span id="cb31-92"><a href="#cb31-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-93"><a href="#cb31-93" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(vdr_ol_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_6b-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p><strong>Figure 6b) Barplot showing the overrepresentation of genes upregulated in <em>APOE2</em> microglia in a list of VDR target genes in monocytes.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="do">## C: IL-1OR receptor ##</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>il10r <span class="ot">&lt;-</span> <span class="fu">plotCounts</span>(dds, <span class="at">gene=</span><span class="st">"3587"</span>, <span class="at">intgroup=</span><span class="st">"group"</span>, <span class="at">returnData =</span> <span class="cn">TRUE</span>)</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>my_comparisons <span class="ot">&lt;-</span> <span class="fu">list</span>( <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E3"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>, <span class="st">"E4"</span>), <span class="fu">c</span>(<span class="st">"E2"</span>,<span class="st">"KO"</span>))</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(il10r, <span class="fu">aes</span>(<span class="at">x=</span>group, <span class="at">y=</span>count, <span class="at">fill=</span>group)) <span class="sc">+</span> </span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span> </span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span> </span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">+</span> </span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylab</span>(<span class="st">"Counts"</span>) <span class="sc">+</span> </span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">"IL-10RA"</span>) <span class="sc">+</span> </span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlab</span>(<span class="st">""</span>) <span class="sc">+</span> </span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">20</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">face=</span><span class="fu">c</span>(<span class="st">"italic"</span>))) <span class="sc">+</span> </span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>(<span class="at">comparisons =</span> my_comparisons, <span class="at">method=</span><span class="st">"t.test"</span>, <span class="at">size=</span><span class="dv">4</span>) <span class="sc">+</span> </span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_jitter</span>() <span class="sc">+</span> </span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">fill=</span><span class="cn">NULL</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/figure_6c-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<pre><code></code></pre>
<p><strong>Figure 6c) Boxplot of gene expression profiles of <em>IL-10RA</em>.</strong></p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/vitD_il10_microglia.png" class="img-fluid quarto-figure quarto-figure-center figure-img"></p>
</figure>
</div>
<p><strong>Figure 6d) Graphic adapted from Boontanrart et al.&nbsp;(2016), showing a mechanism of anti-inflammatory microglia activation mediated through vitamin D and IL-10 signalling.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(vsd) <span class="ot">&lt;-</span> vsd<span class="sc">$</span>samples</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vsd)))</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>pca_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pca<span class="sc">$</span>x)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>pca_data<span class="sc">$</span>group <span class="ot">&lt;-</span> vsd<span class="sc">$</span>group</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>pca_plot_data <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"group"</span>, <span class="at">returnData =</span> <span class="cn">TRUE</span>)</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>apoe_palette <span class="ot">&lt;-</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA plot</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a>rna_pca <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_plot_data, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> group)) <span class="sc">+</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> apoe_palette) <span class="sc">+</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Gene expression"</span>, <span class="at">x =</span> <span class="st">"PC1"</span>, <span class="at">y =</span> <span class="st">"PC2"</span>) <span class="sc">+</span> </span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),  </span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>),  </span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color =</span> <span class="st">"slategrey"</span>, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">size =</span> <span class="dv">1</span>), </span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>)</span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">colour =</span> <span class="st">"slategrey"</span>) <span class="sc">+</span> </span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">colour =</span> <span class="st">"slategrey"</span>)</span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a><span class="co"># outliers: check which samples are 3 standard deviations from the mean </span></span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a>find_group_outliers <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span></span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(group) <span class="sc">%&gt;%</span></span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a>      <span class="at">outliers_PC1 =</span> <span class="fu">list</span>(<span class="fu">which</span>(<span class="fu">abs</span>(PC1 <span class="sc">-</span> <span class="fu">mean</span>(PC1)) <span class="sc">&gt;</span> (<span class="dv">3</span> <span class="sc">*</span> <span class="fu">sd</span>(PC1)))),</span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a>      <span class="at">outliers_PC2 =</span> <span class="fu">list</span>(<span class="fu">which</span>(<span class="fu">abs</span>(PC2 <span class="sc">-</span> <span class="fu">mean</span>(PC2)) <span class="sc">&gt;</span> (<span class="dv">3</span> <span class="sc">*</span> <span class="fu">sd</span>(PC2))))</span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a><span class="fu">find_group_outliers</span>(pca_data)</span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(rna_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_1a-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 1a) PCA on gene expression across the <em>APOE</em> isoforms and the <em>APOE</em>-KO. PCA was generated using the variance stabilizing transformation (VST)-normalised expression counts matrix.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample clustering </span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>rna_cluster <span class="ot">&lt;-</span> <span class="fu">plot_sample_clustering</span>(vsd, <span class="at">anno_vars =</span> <span class="fu">c</span>(<span class="st">"samples"</span>, <span class="st">"group"</span>), <span class="at">distance =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(rna_cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_1b-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 1b) Hierarchical clustering using euclidean distance of VST-normalised expression counts of the <em>APOE</em> isoforms and the <em>APOE</em>-KO.</strong></p>
<div class="cell" data-layout-align="center" data-warninig="false">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(atac_dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(vsd) <span class="ot">&lt;-</span> vsd<span class="sc">$</span>samples</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>pca <span class="ot">&lt;-</span> <span class="fu">prcomp</span>(<span class="fu">t</span>(<span class="fu">assay</span>(vsd)))</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>pca_data <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(pca<span class="sc">$</span>x)</span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>pca_data<span class="sc">$</span>group <span class="ot">&lt;-</span> vsd<span class="sc">$</span>group</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>pca_plot_data <span class="ot">&lt;-</span> <span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"group"</span>, <span class="at">returnData =</span> <span class="cn">TRUE</span>)</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>apoe_palette <span class="ot">&lt;-</span> <span class="fu">ghibli_palette</span>(<span class="st">"YesterdayMedium"</span>, <span class="at">direction =</span> <span class="sc">-</span><span class="dv">1</span>)</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a><span class="co"># PCA plot</span></span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>atac_pca <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(pca_plot_data, <span class="fu">aes</span>(<span class="at">x =</span> PC1, <span class="at">y =</span> PC2, <span class="at">color =</span> group)) <span class="sc">+</span></span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> apoe_palette) <span class="sc">+</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Chromatin accessibility"</span>, <span class="at">x =</span> <span class="st">"PC1"</span>, <span class="at">y =</span> <span class="st">"PC2"</span>) <span class="sc">+</span> </span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb36-18"><a href="#cb36-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_blank</span>(),  </span>
<span id="cb36-19"><a href="#cb36-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>),  </span>
<span id="cb36-20"><a href="#cb36-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.border =</span> <span class="fu">element_rect</span>(<span class="at">color =</span> <span class="st">"slategrey"</span>, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">size =</span> <span class="dv">1</span>), </span>
<span id="cb36-21"><a href="#cb36-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">14</span>)</span>
<span id="cb36-22"><a href="#cb36-22" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb36-23"><a href="#cb36-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">colour =</span> <span class="st">"slategrey"</span>) <span class="sc">+</span> </span>
<span id="cb36-24"><a href="#cb36-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">colour =</span> <span class="st">"slategrey"</span>)</span>
<span id="cb36-25"><a href="#cb36-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-26"><a href="#cb36-26" aria-hidden="true" tabindex="-1"></a><span class="co"># outliers: check which samples are 3 standard deviations from the mean </span></span>
<span id="cb36-27"><a href="#cb36-27" aria-hidden="true" tabindex="-1"></a>find_group_outliers <span class="ot">&lt;-</span> <span class="cf">function</span>(df) {</span>
<span id="cb36-28"><a href="#cb36-28" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span></span>
<span id="cb36-29"><a href="#cb36-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(group) <span class="sc">%&gt;%</span></span>
<span id="cb36-30"><a href="#cb36-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(</span>
<span id="cb36-31"><a href="#cb36-31" aria-hidden="true" tabindex="-1"></a>      <span class="at">outliers_PC1 =</span> <span class="fu">list</span>(<span class="fu">which</span>(<span class="fu">abs</span>(PC1 <span class="sc">-</span> <span class="fu">mean</span>(PC1)) <span class="sc">&gt;</span> (<span class="dv">3</span> <span class="sc">*</span> <span class="fu">sd</span>(PC1)))),</span>
<span id="cb36-32"><a href="#cb36-32" aria-hidden="true" tabindex="-1"></a>      <span class="at">outliers_PC2 =</span> <span class="fu">list</span>(<span class="fu">which</span>(<span class="fu">abs</span>(PC2 <span class="sc">-</span> <span class="fu">mean</span>(PC2)) <span class="sc">&gt;</span> (<span class="dv">3</span> <span class="sc">*</span> <span class="fu">sd</span>(PC2))))</span>
<span id="cb36-33"><a href="#cb36-33" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb36-34"><a href="#cb36-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-35"><a href="#cb36-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-36"><a href="#cb36-36" aria-hidden="true" tabindex="-1"></a><span class="fu">find_group_outliers</span>(pca_data)</span>
<span id="cb36-37"><a href="#cb36-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-38"><a href="#cb36-38" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(atac_pca)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_1c-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 1c) PCA on genome-wide chromatin accessibility across the <em>APOE</em> isoforms and the <em>APOE</em>-KO. PCA was generated using the variance stabilizing transformation (VST)-normalised expression counts matrix.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Sample clustering </span></span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>atac_cluster <span class="ot">&lt;-</span> <span class="fu">plot_sample_clustering</span>(vsd, <span class="at">anno_vars =</span> <span class="fu">c</span>(<span class="st">"samples"</span>, <span class="st">"group"</span>), <span class="at">distance =</span> <span class="st">"euclidean"</span>)</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(atac_cluster)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_1d-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 1d) Hierarchical clustering using euclidean distance of VST-normalised ATAC-seq read counts of the <em>APOE</em> isoforms and the <em>APOE</em>-KO.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>peaks_annot2 <span class="ot">&lt;-</span> <span class="fu">annotatePeak</span>(</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">makeGRangesFromDataFrame</span>(atac_peaks[, <span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>]), </span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">TxDb =</span> txdb, </span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">tssRegion =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">3000</span>, <span class="dv">3000</span>), </span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">annoDb =</span> <span class="st">"org.Hs.eg.db"</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plotAnnoPie</span>(peaks_annot2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_1e-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 1e) Pie chart of genomic annotations of the consensus set of chromatin accessibility peaks.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="do">## TSS enrichments ## </span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>peaks_gr <span class="ot">&lt;-</span> <span class="fu">as.GRanges</span>(peaks_annot2)</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>promoter <span class="ot">&lt;-</span> <span class="fu">getPromoters</span>(<span class="at">TxDb=</span>txdb, <span class="at">upstream=</span><span class="dv">3000</span>, <span class="at">downstream=</span><span class="dv">3000</span>)</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>tagMatrix <span class="ot">&lt;-</span> <span class="fu">getTagMatrix</span>(peaks_gr, <span class="at">windows=</span>promoter)</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="fu">tagHeatmap</span>(tagMatrix)</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a><span class="do">## Average Profile of ATAC peaks binding to TSS region ## </span></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>tss_enrich <span class="ot">&lt;-</span> <span class="fu">plotAvgProf</span>(tagMatrix, <span class="at">xlim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">3000</span>, <span class="dv">3000</span>),</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>            <span class="at">xlab=</span><span class="st">"Genomic Region (5' -&gt; 3')"</span>, <span class="at">ylab =</span> <span class="st">"Read Count Frequency"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<p><img src="images/tss_enrich.png" class="img-fluid"></p>
<p><strong>Supplementary Figure 1f) TSS enrichment profiles of ATAC-seq peaks.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># APOE-KO vs APOE3 </span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">head</span>(comparisons[comparisons<span class="sc">$</span>comparison <span class="sc">==</span> <span class="st">"KOvsE3"</span>, ], <span class="dv">10</span>)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="co"># need to add labels for the top 20 genes</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>KOvsE3_volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(comparisons[comparisons<span class="sc">$</span>comparison<span class="sc">==</span><span class="st">"KOvsE3"</span>,], <span class="fu">aes</span>(<span class="at">x=</span>log2FoldChange, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">colour=</span>diffExpressed)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="fl">1.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="dv">10</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#4D6D93FF"</span>, <span class="st">"#4D6D93FF"</span>, <span class="st">"#9DAFC3FF"</span>)) <span class="sc">+</span> <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"-log10(FDR)"</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">colour=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">""</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"KOvsE3"</span>) <span class="sc">+</span> <span class="fu">geom_text_repel</span>(<span class="at">data =</span> top_genes, <span class="fu">aes</span>(<span class="at">label =</span> hgnc_symbol), <span class="at">box.padding =</span> <span class="fl">0.5</span>, <span class="at">colour=</span><span class="st">"#2A3C50FF"</span>, <span class="at">size=</span><span class="dv">4</span>)</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(KOvsE3_volcano)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_2a-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 2a) Differentially expressed genes in the <em>APOE</em>-KO vs <em>APOE3</em>.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb41"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co"># APOE-KO vs APOE3 excluding APOE </span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">head</span>(comparisons[comparisons<span class="sc">$</span>comparison <span class="sc">==</span> <span class="st">"KOvsE3"</span>, ], <span class="dv">21</span>)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">subset</span>(top_genes, hgnc_symbol <span class="sc">!=</span> <span class="st">"APOE"</span>)</span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>KOvsE3_exc_apoe_volcano <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(comparisons[comparisons<span class="sc">$</span>comparison<span class="sc">==</span><span class="st">"KOvsE3"</span>,], <span class="fu">aes</span>(<span class="at">x=</span>log2FoldChange, <span class="at">y=</span><span class="sc">-</span><span class="fu">log10</span>(fdr), <span class="at">colour=</span>diffExpressed)) <span class="sc">+</span> <span class="fu">geom_point</span>(<span class="at">size=</span><span class="fl">1.5</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="dv">10</span>) <span class="sc">+</span> <span class="fu">scale_colour_manual</span>(<span class="at">values=</span><span class="fu">c</span>(<span class="st">"#4D6D93FF"</span>, <span class="st">"#4D6D93FF"</span>, <span class="st">"#9DAFC3FF"</span>)) <span class="sc">+</span> <span class="fu">theme_classic</span>() <span class="sc">+</span> <span class="fu">ylab</span>(<span class="st">"-log10(FDR)"</span>) <span class="sc">+</span> <span class="fu">guides</span>(<span class="at">colour=</span><span class="fu">guide_legend</span>(<span class="at">title=</span><span class="st">""</span>)) <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>), <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>)) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">"KOvsE3 (excluding APOE)"</span>) <span class="sc">+</span> <span class="fu">geom_text_repel</span>(<span class="at">data =</span> top_genes, <span class="fu">aes</span>(<span class="at">label =</span> hgnc_symbol), <span class="at">box.padding =</span> <span class="fl">0.5</span>, <span class="at">colour=</span><span class="st">"#2A3C50FF"</span>, <span class="at">size=</span><span class="dv">4</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="dv">0</span>,<span class="dv">50</span>)</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(KOvsE3_exc_apoe_volcano)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_2b-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 2b) Differentially expressed genes in the <em>APOE</em>-KO vs <em>APOE3</em> (excluding <em>APOE</em>).</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="do">## B: E4vsE2 DAR heatmap with RNA-seq genes ## </span></span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a>mat  <span class="ot">&lt;-</span> E4vsE2_counts</span>
<span id="cb42-3"><a href="#cb42-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> E4vsE2_atac_genes<span class="sc">$</span>SYMBOL</span>
<span id="cb42-4"><a href="#cb42-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-5"><a href="#cb42-5" aria-hidden="true" tabindex="-1"></a><span class="co"># overlapping genes between atac-seq and rna-seq </span></span>
<span id="cb42-6"><a href="#cb42-6" aria-hidden="true" tabindex="-1"></a>overlap_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mat)[E4vsE2_atac_genes<span class="sc">$</span>SYMBOL <span class="sc">%in%</span> E4vsE2_rna_genes<span class="sc">$</span>hgnc_symbol]</span>
<span id="cb42-7"><a href="#cb42-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-8"><a href="#cb42-8" aria-hidden="true" tabindex="-1"></a><span class="co"># function to make selected rownames bold</span></span>
<span id="cb42-9"><a href="#cb42-9" aria-hidden="true" tabindex="-1"></a>make_bold_names <span class="ot">&lt;-</span> <span class="cf">function</span>(mat, rc_fun, rc_names) {</span>
<span id="cb42-10"><a href="#cb42-10" aria-hidden="true" tabindex="-1"></a>  bold_names <span class="ot">&lt;-</span> <span class="fu">rc_fun</span>(mat)[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>]</span>
<span id="cb42-11"><a href="#cb42-11" aria-hidden="true" tabindex="-1"></a>  ids <span class="ot">&lt;-</span> rc_names <span class="sc">%&gt;%</span> <span class="fu">match</span>(bold_names)</span>
<span id="cb42-12"><a href="#cb42-12" aria-hidden="true" tabindex="-1"></a>  ids <span class="sc">%&gt;%</span></span>
<span id="cb42-13"><a href="#cb42-13" aria-hidden="true" tabindex="-1"></a>    purrr<span class="sc">::</span><span class="fu">walk</span>(</span>
<span id="cb42-14"><a href="#cb42-14" aria-hidden="true" tabindex="-1"></a>      <span class="cf">function</span>(i)</span>
<span id="cb42-15"><a href="#cb42-15" aria-hidden="true" tabindex="-1"></a>        bold_names[i] <span class="ot">&lt;&lt;-</span></span>
<span id="cb42-16"><a href="#cb42-16" aria-hidden="true" tabindex="-1"></a>        <span class="fu">bquote</span>(<span class="fu">bold</span>(.(<span class="fu">rc_fun</span>(mat)[i]))) <span class="sc">%&gt;%</span></span>
<span id="cb42-17"><a href="#cb42-17" aria-hidden="true" tabindex="-1"></a>        <span class="fu">as.expression</span>()</span>
<span id="cb42-18"><a href="#cb42-18" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb42-19"><a href="#cb42-19" aria-hidden="true" tabindex="-1"></a>  bold_names</span>
<span id="cb42-20"><a href="#cb42-20" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb42-21"><a href="#cb42-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-22"><a href="#cb42-22" aria-hidden="true" tabindex="-1"></a>mat_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(mat)</span>
<span id="cb42-23"><a href="#cb42-23" aria-hidden="true" tabindex="-1"></a>mat_df<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mat_df<span class="sc">$</span>E4_1<span class="sc">&gt;</span><span class="dv">0</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span>)</span>
<span id="cb42-24"><a href="#cb42-24" aria-hidden="true" tabindex="-1"></a>mat_df<span class="sc">$</span>label_colour<span class="ot">=</span><span class="fu">ifelse</span>(mat_df<span class="sc">$</span>direction<span class="sc">==</span><span class="st">"Up"</span>,<span class="st">"#B2182B"</span>,<span class="st">"#4D4D4D"</span>)</span>
<span id="cb42-25"><a href="#cb42-25" aria-hidden="true" tabindex="-1"></a><span class="co"># group replicates </span></span>
<span id="cb42-26"><a href="#cb42-26" aria-hidden="true" tabindex="-1"></a>group_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"E2"</span>,<span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"E3"</span>,<span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"E4"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"KO"</span>,<span class="dv">2</span>)))</span>
<span id="cb42-27"><a href="#cb42-27" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(group_col) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(mat)</span>
<span id="cb42-28"><a href="#cb42-28" aria-hidden="true" tabindex="-1"></a>group_colour <span class="ot">=</span> <span class="fu">list</span>(<span class="at">Group =</span> <span class="fu">c</span>(<span class="at">E2 =</span> <span class="st">"#92BBD9FF"</span>, </span>
<span id="cb42-29"><a href="#cb42-29" aria-hidden="true" tabindex="-1"></a>                              <span class="at">E3 =</span> <span class="st">"#ECE28BFF"</span>,</span>
<span id="cb42-30"><a href="#cb42-30" aria-hidden="true" tabindex="-1"></a>                              <span class="at">E4 =</span> <span class="st">"#6FB382FF"</span>, </span>
<span id="cb42-31"><a href="#cb42-31" aria-hidden="true" tabindex="-1"></a>                              <span class="at">KO =</span> <span class="st">"#4D6D93FF"</span>))</span>
<span id="cb42-32"><a href="#cb42-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-33"><a href="#cb42-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-34"><a href="#cb42-34" aria-hidden="true" tabindex="-1"></a>E4vsE2_pheatmap <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(mat,  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">7</span>, <span class="at">name =</span> <span class="st">"RdGy"</span>)))(<span class="dv">100</span>), </span>
<span id="cb42-35"><a href="#cb42-35" aria-hidden="true" tabindex="-1"></a>                            <span class="at">cluster_rows=</span><span class="cn">TRUE</span>, <span class="at">cluster_cols=</span><span class="cn">FALSE</span>, <span class="at">fontsize_col =</span> <span class="dv">16</span>, </span>
<span id="cb42-36"><a href="#cb42-36" aria-hidden="true" tabindex="-1"></a>                            <span class="at">fontsize_row=</span><span class="dv">12</span>, <span class="at">labels_row =</span> <span class="fu">make_bold_names</span>(mat,rownames,overlap_genes),</span>
<span id="cb42-37"><a href="#cb42-37" aria-hidden="true" tabindex="-1"></a>                            <span class="at">cellwidth =</span> <span class="dv">15</span>, <span class="at">cellheight=</span><span class="dv">10</span>, <span class="at">annotation_col =</span> group_col,</span>
<span id="cb42-38"><a href="#cb42-38" aria-hidden="true" tabindex="-1"></a>                            <span class="at">annotation_colors =</span> group_colour, <span class="at">labels_col =</span> <span class="st">""</span>, <span class="at">main =</span> <span class="st">"E4 vs E2"</span>)</span>
<span id="cb42-39"><a href="#cb42-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb42-40"><a href="#cb42-40" aria-hidden="true" tabindex="-1"></a>cols<span class="ot">=</span>mat_df[E4vsE2_pheatmap<span class="sc">$</span>tree_row<span class="sc">$</span>order,]<span class="sc">$</span>label_colour</span>
<span id="cb42-41"><a href="#cb42-41" aria-hidden="true" tabindex="-1"></a>E4vsE2_pheatmap<span class="sc">$</span>gtable<span class="sc">$</span>grobs[[<span class="dv">5</span>]]<span class="sc">$</span>gp<span class="ot">=</span><span class="fu">gpar</span>(<span class="at">col=</span>cols, <span class="at">fontsize =</span> <span class="dv">12</span>)</span>
<span id="cb42-42"><a href="#cb42-42" aria-hidden="true" tabindex="-1"></a>E4vsE2_pheatmap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_3a-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 2a) Heatmap showing differential accessibility of significant peaks (FDR &lt; 0.05) when comparing APOE4 vs APOE3-expressing microglia. Shown are the genes annotated to the top 20 peaks, genes marked in bold were also significantly differentially expressed in the RNA-seq analysis.</strong></p>
<div class="cell">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a><span class="do">## B: KOvsE3 DAR heatmap with RNA-seq genes ## </span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>mat  <span class="ot">&lt;-</span> KOvsE3_counts</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> KOvsE3_atac_genes<span class="sc">$</span>SYMBOL</span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a><span class="co"># overlapping genes between atac-seq and rna-seq </span></span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a>overlap_genes <span class="ot">&lt;-</span> <span class="fu">rownames</span>(mat)[KOvsE3_atac_genes<span class="sc">$</span>SYMBOL <span class="sc">%in%</span> KOvsE3_rna_genes<span class="sc">$</span>hgnc_symbol]</span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>mat_df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(mat)</span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>mat_df<span class="sc">$</span>direction <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(mat_df<span class="sc">$</span>KO_3<span class="sc">&gt;</span><span class="dv">0</span>, <span class="st">"Up"</span>, <span class="st">"Down"</span>)</span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>mat_df<span class="sc">$</span>label_colour<span class="ot">=</span><span class="fu">ifelse</span>(mat_df<span class="sc">$</span>direction<span class="sc">==</span><span class="st">"Up"</span>,<span class="st">"#B2182B"</span>,<span class="st">"#4D4D4D"</span>)</span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a><span class="co"># group replicates </span></span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>group_col <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Group =</span> <span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"E2"</span>,<span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"E3"</span>,<span class="dv">5</span>), <span class="fu">rep</span>(<span class="st">"E4"</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">"KO"</span>,<span class="dv">2</span>)))</span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(group_col) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(mat)</span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>group_colour <span class="ot">=</span> <span class="fu">list</span>(<span class="at">Group =</span> <span class="fu">c</span>(<span class="at">E2 =</span> <span class="st">"#92BBD9FF"</span>, </span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>                              <span class="at">E3 =</span> <span class="st">"#ECE28BFF"</span>,</span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a>                              <span class="at">E4 =</span> <span class="st">"#6FB382FF"</span>, </span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>                              <span class="at">KO =</span> <span class="st">"#4D6D93FF"</span>))</span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a>KOvsE3_pheatmap <span class="ot">&lt;-</span> <span class="fu">pheatmap</span>(</span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>  mat,  </span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">color =</span> <span class="fu">colorRampPalette</span>(<span class="fu">rev</span>(<span class="fu">brewer.pal</span>(<span class="at">n =</span> <span class="dv">7</span>, <span class="at">name =</span> <span class="st">"RdGy"</span>)))(<span class="dv">100</span>), </span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_rows =</span> <span class="cn">TRUE</span>, </span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a>  <span class="at">cluster_cols =</span> <span class="cn">FALSE</span>, </span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_col =</span> <span class="dv">16</span>, </span>
<span id="cb43-27"><a href="#cb43-27" aria-hidden="true" tabindex="-1"></a>  <span class="at">fontsize_row =</span> <span class="dv">12</span>, </span>
<span id="cb43-28"><a href="#cb43-28" aria-hidden="true" tabindex="-1"></a>  <span class="at">labels_row =</span> <span class="fu">make_bold_names</span>(mat, rownames, overlap_genes),</span>
<span id="cb43-29"><a href="#cb43-29" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellwidth =</span> <span class="dv">15</span>, </span>
<span id="cb43-30"><a href="#cb43-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">cellheight =</span> <span class="dv">10</span>, </span>
<span id="cb43-31"><a href="#cb43-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_col =</span> group_col,</span>
<span id="cb43-32"><a href="#cb43-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_colors =</span> group_colour, </span>
<span id="cb43-33"><a href="#cb43-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">labels_col =</span> <span class="st">""</span>, </span>
<span id="cb43-34"><a href="#cb43-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">main =</span> <span class="st">"KO vs E3"</span></span>
<span id="cb43-35"><a href="#cb43-35" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb43-36"><a href="#cb43-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-37"><a href="#cb43-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-38"><a href="#cb43-38" aria-hidden="true" tabindex="-1"></a>cols<span class="ot">=</span>mat_df[KOvsE3_pheatmap<span class="sc">$</span>tree_row<span class="sc">$</span>order,]<span class="sc">$</span>label_colour</span>
<span id="cb43-39"><a href="#cb43-39" aria-hidden="true" tabindex="-1"></a>KOvsE3_pheatmap<span class="sc">$</span>gtable<span class="sc">$</span>grobs[[<span class="dv">5</span>]]<span class="sc">$</span>gp<span class="ot">=</span><span class="fu">gpar</span>(<span class="at">col=</span>cols, <span class="at">fontsize =</span> <span class="dv">12</span>)</span>
<span id="cb43-40"><a href="#cb43-40" aria-hidden="true" tabindex="-1"></a>KOvsE3_pheatmap</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_3b-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 2b) Heatmap showing differential accessibility of significant peaks (FDR &lt; 0.05) when comparing the APOE-KO vs APOE3. Shown are the genes annotated to the top 20 peaks, genes marked in bold were also significantly differentially expressed in the RNA-seq analysis.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># corelate with ATAC-seq promoter peaks</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>promoter_peaks <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"data/APOE_microglia_ATACseq_promoter_peaks.csv"</span>)</span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a><span class="co"># get median logFC for each comparison</span></span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a>get_median_logfc <span class="ot">&lt;-</span> <span class="cf">function</span>(peaks, comparison) {</span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a>  peaks <span class="sc">%&gt;%</span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(comparison <span class="sc">==</span> comparison) <span class="sc">%&gt;%</span></span>
<span id="cb44-8"><a href="#cb44-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(geneId) <span class="sc">%&gt;%</span></span>
<span id="cb44-9"><a href="#cb44-9" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">summarise</span>(<span class="at">median_log2FoldChange =</span> <span class="fu">median</span>(log2FoldChange, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb44-10"><a href="#cb44-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb44-11"><a href="#cb44-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-12"><a href="#cb44-12" aria-hidden="true" tabindex="-1"></a><span class="co"># E2vsE3 </span></span>
<span id="cb44-13"><a href="#cb44-13" aria-hidden="true" tabindex="-1"></a>E2vsE3_atac_logFC <span class="ot">&lt;-</span> <span class="fu">get_median_logfc</span>(promoter_peaks, <span class="st">"E2vsE3"</span>)</span>
<span id="cb44-14"><a href="#cb44-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-15"><a href="#cb44-15" aria-hidden="true" tabindex="-1"></a>E2vsE3_cor <span class="ot">&lt;-</span> <span class="fu">merge</span>(E2vsE3_rna_sig, E2vsE3_atac_logFC, <span class="at">by.x=</span><span class="st">"entrezgene_id"</span>, <span class="at">by.y=</span><span class="st">"geneId"</span>)</span>
<span id="cb44-16"><a href="#cb44-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-17"><a href="#cb44-17" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(E2vsE3_cor, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> median_log2FoldChange)) <span class="sc">+</span></span>
<span id="cb44-18"><a href="#cb44-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">colour=</span><span class="st">"#92BBD9FF"</span>, <span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb44-19"><a href="#cb44-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">colour=</span><span class="st">"#92BBD9FF"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add a linear regression line</span></span>
<span id="cb44-20"><a href="#cb44-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span>  <span class="co"># Set a minimal theme with no background</span></span>
<span id="cb44-21"><a href="#cb44-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"E2 vs E3"</span>,</span>
<span id="cb44-22"><a href="#cb44-22" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Expression logFC"</span>,</span>
<span id="cb44-23"><a href="#cb44-23" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Accessibility logFC"</span>) <span class="sc">+</span> </span>
<span id="cb44-24"><a href="#cb44-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">size=</span><span class="dv">4</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">8</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">8</span>) <span class="sc">+</span> </span>
<span id="cb44-25"><a href="#cb44-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_4a-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 4a) RNA-seq logFC and ATAC-seq logFC correlate across DEGs and their promoter peaks in APOE2 vs APOE3-expressing microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co"># E4vsE2 </span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>E4vsE2_atac_logFC <span class="ot">&lt;-</span> <span class="fu">get_median_logfc</span>(promoter_peaks, <span class="st">"E4vsE2"</span>)</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>E4vsE2_cor <span class="ot">&lt;-</span> <span class="fu">merge</span>(E4vsE2_rna_sig, E4vsE2_atac_logFC, <span class="at">by.x=</span><span class="st">"entrezgene_id"</span>, <span class="at">by.y=</span><span class="st">"geneId"</span>)</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>E4vsE2_cor_res <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(E4vsE2_cor<span class="sc">$</span>log2FoldChange, E4vsE2_cor<span class="sc">$</span>median_log2FoldChange, </span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(E4vsE2_cor, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> median_log2FoldChange)) <span class="sc">+</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">colour=</span><span class="st">"#6FB382FF"</span>, <span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">colour=</span><span class="st">"#6FB382FF"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add a linear regression line</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span>  <span class="co"># Set a minimal theme with no background</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"E4 vs E2"</span>,</span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Expression logFC"</span>,</span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Accessibility logFC"</span>) <span class="sc">+</span> </span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">size=</span><span class="dv">4</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">8</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">8</span>) <span class="sc">+</span> </span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_4b-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 4b) RNA-seq logFC and ATAC-seq logFC correlate across DEGs and their promoter peaks in <em>APOE4</em> vs <em>APOE2</em>-expressing microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># E4vsE3 </span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>E4vsE3_atac_logFC <span class="ot">&lt;-</span> <span class="fu">get_median_logfc</span>(promoter_peaks, <span class="st">"E4vsE3"</span>)</span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a>E4vsE3_cor <span class="ot">&lt;-</span> <span class="fu">merge</span>(E4vsE3_rna_sig, E4vsE3_atac_logFC, <span class="at">by.x=</span><span class="st">"entrezgene_id"</span>, <span class="at">by.y=</span><span class="st">"geneId"</span>)</span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-6"><a href="#cb46-6" aria-hidden="true" tabindex="-1"></a>E4vsE3_cor_res <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(E4vsE3_cor<span class="sc">$</span>log2FoldChange, E4vsE3_cor<span class="sc">$</span>median_log2FoldChange, </span>
<span id="cb46-7"><a href="#cb46-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb46-8"><a href="#cb46-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-9"><a href="#cb46-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(E4vsE3_cor, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> median_log2FoldChange)) <span class="sc">+</span></span>
<span id="cb46-10"><a href="#cb46-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">colour=</span><span class="st">"#DCCA2CFF"</span>, <span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb46-11"><a href="#cb46-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">colour=</span><span class="st">"#DCCA2CFF"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add a linear regression line</span></span>
<span id="cb46-12"><a href="#cb46-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span>  <span class="co"># Set a minimal theme with no background</span></span>
<span id="cb46-13"><a href="#cb46-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"E4 vs E3"</span>,</span>
<span id="cb46-14"><a href="#cb46-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Expression logFC"</span>,</span>
<span id="cb46-15"><a href="#cb46-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Accessibility logFC"</span>) <span class="sc">+</span> </span>
<span id="cb46-16"><a href="#cb46-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">size=</span><span class="dv">4</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">8</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">8</span>) <span class="sc">+</span> </span>
<span id="cb46-17"><a href="#cb46-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_4c-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 3c) RNA-seq logFC and ATAC-seq logFC correlate across DEGs and their promoter peaks in APOE4 vs APOE3-expressing microglia.</strong></p>
<div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="co"># KOvsE3 </span></span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>KOvsE3_atac_logFC <span class="ot">&lt;-</span> <span class="fu">get_median_logfc</span>(promoter_peaks, <span class="st">"KOvsE3"</span>)</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a>KOvsE3_cor <span class="ot">&lt;-</span> <span class="fu">merge</span>(KOvsE3_rna_sig, KOvsE3_atac_logFC, <span class="at">by.x=</span><span class="st">"entrezgene_id"</span>, <span class="at">by.y=</span><span class="st">"geneId"</span>)</span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>KOvsE3_cor_res <span class="ot">&lt;-</span> <span class="fu">cor.test</span>(KOvsE3_cor<span class="sc">$</span>log2FoldChange, KOvsE3_cor<span class="sc">$</span>median_log2FoldChange, </span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>                           <span class="at">method =</span> <span class="st">"pearson"</span>)</span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(KOvsE3_cor, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> median_log2FoldChange)) <span class="sc">+</span></span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">colour=</span><span class="st">"#4D6D93FF"</span>, <span class="at">size=</span><span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">"lm"</span>, <span class="at">se =</span> <span class="cn">FALSE</span>, <span class="at">colour=</span><span class="st">"#4D6D93FF"</span>, <span class="at">alpha=</span><span class="fl">0.1</span>) <span class="sc">+</span>  <span class="co"># Add a linear regression line</span></span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span>  <span class="co"># Set a minimal theme with no background</span></span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"KO vs E3"</span>,</span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Expression logFC"</span>,</span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Accessibility logFC"</span>) <span class="sc">+</span> </span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">size=</span><span class="dv">4</span>) <span class="sc">+</span> <span class="fu">xlim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">8</span>) <span class="sc">+</span> <span class="fu">ylim</span>(<span class="sc">-</span><span class="dv">8</span>,<span class="dv">8</span>) <span class="sc">+</span> </span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size=</span><span class="dv">16</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="manuscript_figures_files/figure-html/supp_figure_4d-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="480"></p>
</figure>
</div>
</div>
</div>
<p><strong>Supplementary Figure 3d) RNA-seq logFC and ATAC-seq logFC correlate across DEGs and their promoter peaks in the <em>APOE</em>-KO vs <em>APOE2</em>-expressing microglia.</strong></p>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>