---
title: "APOE_microglia_figures"
author: "Kitty Murphy"
date: "`r Sys.Date()`"
output: html_document
---

Code for generating results and figures.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library(ggplot2)
library(ggrepel)
library(RRHO2)
library(wesanderson)
library(readxl)
```


```{r volcano_plots}

setwd("/Volumes/cm1118/projects/mrc_bdr/live/APOE_microglia")

## RNAseq 

# differential expression analysis res
E2vsE3_DEGs <- read_excel("RNAseq/results/APOE_microglia_DeSeq2_DEGs.xlsx", sheet = "E2vsE3")
E4vsE3_DEGs <- read_excel("RNAseq/results/APOE_microglia_DeSeq2_DEGs.xlsx", sheet = "E4vsE3")
E4vsE2_DEGs <- read_excel("RNAseq/results/APOE_microglia_DeSeq2_DEGs.xlsx", sheet = "E4vsE2")
KOvsE3_DEGs <- read_excel("RNAseq/results/APOE_microglia_DeSeq2_DEGs.xlsx", sheet = "KOvsE3")

# E2vsE3
E2vsE3_DEG_plot <- ggplot(E2vsE3_DEGs, aes(x=log2FoldChange, y=-log10(P_fdr))) + 
  geom_point(aes(color=diffExpressed),size=1) + theme_minimal() + xlim(-20,20) + 
  scale_colour_manual(values = wes_palette("FantasticFox1")) + 
  geom_text_repel(data=E2vsE3_DEGs[1:5,],aes(x = log2FoldChange, y = -log10(P_fdr),label=hgnc_symbol, color=diffExpressed, size=14), show.legend=FALSE) + 
  xlab("log2(fold change)") + ylab("-log10(FDR)") + ggtitle("E2 vs E3 DEGs") + 
  guides(colour = guide_legend(title="",override.aes = list(shape=19, size=2))) + 
  theme(text = element_text(size = 14)) 

# E4vsE3 
E4vsE3_DEG_plot <- ggplot(E4vsE3_DEGs, aes(x=log2FoldChange, y=-log10(P_fdr))) + 
  geom_point(aes(color=diffExpressed),size=1) + theme_minimal() + xlim(-10,10) + 
  scale_colour_manual(values = wes_palette("FantasticFox1")) + 
  geom_text_repel(data=E4vsE3_DEGs[1:5,],aes(x = log2FoldChange, y = -log10(P_fdr),label=hgnc_symbol, color=diffExpressed, size=14), show.legend=FALSE) + 
  xlab("log2(fold change)") + ylab("-log10(FDR)") + ggtitle("E4 vs E3 DEGs") + 
  guides(colour = guide_legend(title="",override.aes = list(shape=19, size=2))) + 
  theme(text = element_text(size = 14)) 

# E4vsE2
E4vsE2_DEG_plot <- ggplot(E4vsE2_DEGs, aes(x=log2FoldChange, y=-log10(P_fdr))) + 
  geom_point(aes(color=diffExpressed),size=1) + theme_minimal() + xlim(-20,20) + 
  scale_colour_manual(values = wes_palette("FantasticFox1")) + 
  geom_text_repel(data=E4vsE2_DEGs[1:5,],aes(x = log2FoldChange, y = -log10(P_fdr),label=hgnc_symbol, color=diffExpressed, size=14), show.legend=FALSE) + 
  xlab("log2(fold change)") + ylab("-log10(FDR)") + ggtitle("E4 vs E2 DEGs") + 
  guides(colour = guide_legend(title="",override.aes = list(shape=19, size=2))) + 
  theme(text = element_text(size = 14)) 

# KOvsE3
KOvsE3_DEG_plot <- ggplot(KOvsE3_DEGs, aes(x=log2FoldChange, y=-log10(P_fdr))) + 
  geom_point(aes(color=diffExpressed),size=1) + theme_minimal() + xlim(-20,20) + 
  scale_colour_manual(values = wes_palette("FantasticFox1")) + 
  geom_text_repel(data=KOvsE3_DEGs[1:5,],aes(x = log2FoldChange, y = -log10(P_fdr),label=hgnc_symbol, color=diffExpressed, size=14), show.legend=FALSE) + 
  xlab("log2(fold change)") + ylab("-log10(FDR)") + ggtitle("KO vs E3 DEGs") + 
  guides(colour = guide_legend(title="",override.aes = list(shape=19, size=2))) + 
  theme(text = element_text(size = 14)) 

## ATACseq

# differential expression analysis res
E2vsE3_DARs <- read_excel("ATACseq/results/APOE_microglia_edgeR_DARs.xlsx", sheet = "E2vsE3")
E4vsE3_DARs <- read_excel("ATACseq/results/APOE_microglia_edgeR_DARs.xlsx", sheet = "E4vsE3")
E4vsE2_DARs <- read_excel("ATACseq/results/APOE_microglia_edgeR_DARs.xlsx", sheet = "E4vsE2")
KOvsE3_DARs <- read_excel("ATACseq/results/APOE_microglia_edgeR_DARs.xlsx", sheet = "KOvsE3")

# E2vsE3
E2vsE3_DAR_plot <- ggplot(E2vsE3_DARs, aes(x=logFC, y=-log10(P_fdr))) + 
  geom_point(aes(color=diffAccessible),size=1) + theme_minimal() + xlim(-10,10) + 
  scale_colour_manual(values = wes_palette("FantasticFox1"))  + 
  xlab("log(fold change)") + ylab("-log10(FDR)") + ggtitle("E2 vs E3 DARs") + 
  guides(colour = guide_legend(title="",override.aes = list(shape=19, size=2))) + 
  theme(text = element_text(size = 14)) 

# E4vsE4
E4vsE3_DAR_plot <- ggplot(E4vsE3_DARs, aes(x=logFC, y=-log10(P_fdr))) + 
  geom_point(aes(color=diffAccessible),size=1) + theme_minimal() + xlim(-10,10) + 
  scale_colour_manual(values = wes_palette("FantasticFox1"))  + 
  xlab("log(fold change)") + ylab("-log10(FDR)") + ggtitle("E4 vs E3 DARs") + 
  guides(colour = guide_legend(title="",override.aes = list(shape=19, size=2))) + 
  theme(text = element_text(size = 14))

# E4vsE2
E4vsE2_DAR_plot <- ggplot(E4vsE2_DARs, aes(x=logFC, y=-log10(P_fdr))) + 
  geom_point(aes(color=diffAccessible),size=1) + theme_minimal() + xlim(-10,10) + 
  scale_colour_manual(values = wes_palette("FantasticFox1"))  + 
  xlab("log(fold change)") + ylab("-log10(FDR)") + ggtitle("E4 vs E2 DARs") + 
  guides(colour = guide_legend(title="",override.aes = list(shape=19, size=2))) + 
  theme(text = element_text(size = 14))

# KOvsE3
KOvsE3_DAR_plot <- ggplot(KOvsE3_DARs, aes(x=logFC, y=-log10(P_fdr))) + 
  geom_point(aes(color=diffAccessible),size=1) + theme_minimal() + xlim(-10,10) + 
  scale_colour_manual(values = wes_palette("FantasticFox1"))  + 
  xlab("log(fold change)") + ylab("-log10(FDR)") + ggtitle("KO vs E3 DARs") + 
  guides(colour = guide_legend(title="",override.aes = list(shape=19, size=2))) + 
  theme(text = element_text(size = 14))

```

```{r rrho_plots}

# E2
E2vsE3_DEGs_pos <- E2vsE3_DEGs[E2vsE3_DEGs$log2FoldChange>0,]
E2vsE3_DEGs_neg <- E2vsE3_DEGs[E2vsE3_DEGs$log2FoldChange<0,]
E2vsE3_DEGs_pos_rrho <- data.frame(gene=E2vsE3_DEGs_pos$hgnc_symbol, score=-log10(E2vsE3_DEGs_pos$pvalue))
E2vsE3_DEGs_pos_rrho <- E2vsE3_DEGs_pos_rrho[!grepl("NA", E2vsE3_DEGs_pos_rrho$gene),]
E2vsE3_DEGs_neg_rrho <- data.frame(gene=E2vsE3_DEGs_neg$hgnc_symbol, score=-log10(E2vsE3_DEGs_neg$pvalue)*-1)
E2vsE3_DEGs_neg_rrho <- E2vsE3_DEGs_neg_rrho[!grepl("NA", E2vsE3_DEGs_neg_rrho$gene),]
E2vsE3_DEGs_rrho <- rbind(E2vsE3_DEGs_pos_rrho, E2vsE3_DEGs_neg_rrho)
E2vsE3_DEGs_rrho <- E2vsE3_DEGs_rrho[!duplicated(E2vsE3_DEGs_rrho$gene),]
E2vsE3_DEGs_rrho <- na.omit(E2vsE3_DEGs_rrho)

# E4
E4vsE3_DEGs_pos <- E4vsE3_DEGs[E4vsE3_DEGs$log2FoldChange>0,]
E4vsE3_DEGs_neg <- E4vsE3_DEGs[E4vsE3_DEGs$log2FoldChange<0,]
E4vsE3_DEGs_pos_rrho <- data.frame(gene=E4vsE3_DEGs_pos$hgnc_symbol, score=-log10(E4vsE3_DEGs_pos$pvalue))
E4vsE3_DEGs_pos_rrho <- E4vsE3_DEGs_pos_rrho[!grepl("NA", E4vsE3_DEGs_pos_rrho$gene),]
E4vsE3_DEGs_neg_rrho <- data.frame(gene=E4vsE3_DEGs_neg$hgnc_symbol, score=-log10(E4vsE3_DEGs_neg$pvalue)*-1)
E4vsE3_DEGs_neg_rrho <- E4vsE3_DEGs_neg_rrho[!grepl("NA", E4vsE3_DEGs_neg_rrho$gene),]
E4vsE3_DEGs_rrho <- rbind(E4vsE3_DEGs_pos_rrho, E4vsE3_DEGs_neg_rrho)
E4vsE3_DEGs_rrho <- E4vsE3_DEGs_rrho[!duplicated(E4vsE3_DEGs_rrho$gene),]
E4vsE3_DEGs_rrho <- na.omit(E4vsE3_DEGs_rrho)

# KO
KOvsE3_DEGs_pos <- KOvsE3_DEGs[KOvsE3_DEGs$log2FoldChange>0,]
KOvsE3_DEGs_neg <- KOvsE3_DEGs[KOvsE3_DEGs$log2FoldChange<0,]
KOvsE3_DEGs_pos_rrho <- data.frame(gene=KOvsE3_DEGs_pos$hgnc_symbol, score=-log10(KOvsE3_DEGs_pos$pvalue))
KOvsE3_DEGs_pos_rrho <- KOvsE3_DEGs_pos_rrho[!grepl("NA", KOvsE3_DEGs_pos_rrho$gene),]
KOvsE3_DEGs_neg_rrho <- data.frame(gene=KOvsE3_DEGs_neg$hgnc_symbol, score=-log10(KOvsE3_DEGs_neg$pvalue)*-1)
KOvsE3_DEGs_neg_rrho <- KOvsE3_DEGs_neg_rrho[!grepl("NA", KOvsE3_DEGs_neg_rrho$gene),]
KOvsE3_DEGs_rrho <- rbind(KOvsE3_DEGs_pos_rrho, KOvsE3_DEGs_neg_rrho)
KOvsE3_DEGs_rrho <- KOvsE3_DEGs_rrho[!duplicated(KOvsE3_DEGs_rrho$gene),]
KOvsE3_DEGs_rrho <- na.omit(KOvsE3_DEGs_rrho)

## E2 vs KO 
E2vsKO_RRHO_obj <-  RRHO2_initialize(E2vsE3_DEGs_rrho, KOvsE3_DEGs_rrho, labels = c("E2 vs E3", "KO vs E3"), log10.ind=TRUE)
E2vsKO_RRHO_heatmap <- RRHO2_heatmap(E2vsKO_RRHO_obj)

# E4 vs KO 
E4vsKO_RRHO_obj <-  RRHO2_initialize(E4vsE3_DEGs_rrho, KOvsE3_DEGs_rrho, labels = c("E4 vs E3", "KO vs E3"), log10.ind=TRUE)
E4vsKO_RRHO_heatmap <- RRHO2_heatmap(E4vsKO_RRHO_obj)

# E4 vs E2
E4vsE2_RRHO_obj <-  RRHO2_initialize(E4vsE3_DEGs_rrho, E2vsE3_DEGs_rrho, labels = c("E4 vs E3", "E2 vs E3"), log10.ind=TRUE)
E4vsE2_RRHO_heatmap <- RRHO2_heatmap(E4vsE2_RRHO_obj)

## correlations
names(E2vsE3_DEGs_rrho) <- c("gene", "E2score")
names(KOvsE3_DEGs_rrho) <- c("gene", "KOscore")
names(E4vsE3_DEGs_rrho) <- c("gene", "E4score")

E2vsKO_corr <- merge(E2vsE3_DEGs_rrho, KOvsE3_DEGs_rrho, by="gene")
E4vsKO_corr <- merge(E4vsE3_DEGs_rrho, KOvsE3_DEGs_rrho, by="gene")
E4vsE2_corr <- merge(E4vsE3_DEGs_rrho, E2vsE3_DEGs_rrho, by="gene")
``` 

```{r expression_lvl_plots}
# read in dds 
dds <- readRDS("RNAseq/data/DeSeq2_dds.rds")
dds$Group <- factor(dds$Group, levels=c("E2", "E3", "E4", "KO"))

# CHCHD2 
chchd2 <- plotCounts(dds, gene="ENSG00000106153", intgroup="Group", returnData = TRUE)
my_comparisons <- list( c("E4", "E3"), c("E4", "E2"))
chchd2_plot <- ggplot(chchd2, aes(x=Group, y=count, fill=Group)) + geom_boxplot() + 
  theme_classic() + scale_fill_manual(values = wes_palette("Darjeeling1")) + 
  ylab("Count") + ggtitle("CHCHD2") + 
  theme(text = element_text(size = 14), plot.title = element_text(hjust = 0.5, face="italic")) + 
  stat_compare_means(comparisons = my_comparisons, method="t.test") + geom_jitter()

# ZNF248 
znf248 <- plotCounts(dds, gene="ENSG00000198105", intgroup="Group", returnData=TRUE)
my_comparisons <- list( c("E4", "KO"), c("E4", "E3"), c("E4","E2"))
znf248_plot <- ggplot(znf248, aes(x=Group, y=count, fill=Group)) + geom_boxplot() + 
  theme_classic() + scale_fill_manual(values = wes_palette("Darjeeling1")) + 
  ylab("Count") + ggtitle("ZNF248") + 
  theme(text = element_text(size = 14), plot.title = element_text(hjust = 0.5, face="italic")) + 
  stat_compare_means(comparisons = my_comparisons, method="t.test") + geom_jitter()
``` 

```{r figure_1}
cowplot::plot_grid(E2vsE3_DEG_plot, E4vsE3_DEG_plot, E4vsE2_DEG_plot, KOvsE3_DEG_plot, E2vsE3_DAR_plot, E4vsE3_DAR_plot, E4vsE2_DAR_plot, KOvsE3_DAR_plot, nrow = 2, ncol=1, labels=c("a","b","c", "d", "e", "f", "g", "h"))
``` 